<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>DESEQ2_POST_PROCESS &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="BISKMARK" href="bismark.html" />
    <link rel="prev" title="BCFTOOLS ANNOTATE" href="bcftools_annotate.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bcftools_annotate.html">BCFTOOLS ANNOTATE</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">DESEQ2_POST_PROCESS</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="bismark.html">BISKMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="bismark_multisample.html">BISKMARK_MULTISAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bowtie2.html">BOWTIE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_fixmate.html">BWA_FIXMATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_mapping.html">BWA_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="clusterprofiler.html">CLUSTERP PROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="compress_index_vcf.html">COMPRESS_INDEX_VCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="coverage_variant_region.html">COVERAGE_VARIANT_REGION</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_pe.html">DADA2-PE</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_se.html">DADA2-SE</a></li>
<li class="toctree-l2"><a class="reference internal" href="deseq2_post_process.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="download_references.html">DOWNLOAD REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_cytoscan.html">EACON CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_oncoscan.html">EACON ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_post_process.html">EACON POST PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastq_screen_indexer.html">FASTQ_SCREEN_INDEXER</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk_bqsr.html">GATK_BQSR</a></li>
<li class="toctree-l2"><a class="reference internal" href="index_datasets.html">INDEX_DATASETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="maftools.html">MAFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller.html">META_CALLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller_germline.html">META_CALLER_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="msi_sensor_pro.html">MSISENSOR_PRO</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_germline.html">MUTECT2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_somatic.html">MUTECT2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_annotate.html">SNPEFF_ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="somatic_tmb.html">SOMATIC_TMB</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_arriba.html">STAR-ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_mapping.html">STAR_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="tximport_deseq2.html">TXIMPORT_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_calling.html">VARSCAN2_CALLING</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_germline.html">VARSCAN2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_somatic.html">VARSCAN2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="vcf_post_process.html">VCF_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="wisecondorx.html">WISECONDORX</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../meta-wrappers.html">Meta-Wrappers</a> &raquo;</li>
      <li>DESEQ2_POST_PROCESS</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/meta-wrappers/bioinfokit.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="deseq2-post-process">
<span id="id1"></span><h1>DESEQ2_POST_PROCESS<a class="headerlink" href="#deseq2-post-process" title="Permalink to this heading"></a></h1>
<p>Plot various information based on DESeq2 results</p>
<section id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this heading"></a></h2>
<p>This meta-wrapper can be used by integrating the following into your workflow:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">os.path</span>

<span class="kn">from</span> <span class="nn">typing</span> <span class="kn">import</span> <span class="n">Optional</span>

<span class="n">default_config</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;thresholds&quot;</span><span class="p">:{</span>
        <span class="s2">&quot;alpha&quot;</span><span class="p">:</span> <span class="mf">0.05</span><span class="p">,</span>
        <span class="s2">&quot;fc_threshold&quot;</span><span class="p">:</span> <span class="mf">0.01</span>
    <span class="p">},</span>
    <span class="s2">&quot;bioinfokit&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;maplot_extra&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;lfc&quot;</span><span class="p">:</span> <span class="s2">&quot;log2FoldChange&quot;</span><span class="p">,</span>
            <span class="s2">&quot;ct_count&quot;</span><span class="p">:</span> <span class="s2">&quot;a.chr21&quot;</span><span class="p">,</span>
            <span class="s2">&quot;st_count&quot;</span><span class="p">:</span> <span class="s2">&quot;e.chr21&quot;</span>
        <span class="p">},</span>
        <span class="s2">&quot;heatmap_extra&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;rowclus&quot;</span><span class="p">:</span> <span class="kc">True</span><span class="p">,</span>
            <span class="s2">&quot;colclus&quot;</span><span class="p">:</span> <span class="kc">True</span>
        <span class="p">},</span>
        <span class="s2">&quot;volcanoplot_extra&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;lfc&quot;</span><span class="p">:</span> <span class="s2">&quot;log2FoldChange&quot;</span><span class="p">,</span>
            <span class="s2">&quot;pv&quot;</span><span class="p">:</span> <span class="s2">&quot;padj&quot;</span><span class="p">,</span>
            <span class="s2">&quot;geneid&quot;</span><span class="p">:</span> <span class="s2">&quot;Unnamed: 0&quot;</span><span class="p">,</span>
            <span class="s2">&quot;lfc_thr&quot;</span><span class="p">:</span> <span class="p">(</span><span class="mf">0.001</span><span class="p">,</span> <span class="mf">0.001</span><span class="p">),</span>
            <span class="s2">&quot;pv_thr&quot;</span><span class="p">:</span> <span class="p">(</span><span class="mf">0.05</span><span class="p">,</span> <span class="mf">0.05</span><span class="p">),</span>
            <span class="s2">&quot;gstyle&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
            <span class="s2">&quot;sign_line&quot;</span><span class="p">:</span> <span class="kc">True</span><span class="p">,</span>
            <span class="s2">&quot;plotlegend&quot;</span><span class="p">:</span> <span class="kc">True</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>

<span class="k">try</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">config</span> <span class="o">==</span> <span class="nb">dict</span><span class="p">():</span>
        <span class="n">config</span> <span class="o">=</span> <span class="n">default_config</span>
<span class="k">except</span> <span class="ne">NameError</span><span class="p">:</span>
    <span class="n">config</span> <span class="o">=</span> <span class="n">default_config</span>

<span class="k">def</span> <span class="nf">rst_exists</span><span class="p">(</span><span class="n">path</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">]:</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    If a provided rst path exists, then it is returned,</span>
<span class="sd">    otherwise &quot;None&quot; is returned.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">return</span> <span class="n">path</span> <span class="k">if</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">path</span><span class="p">)</span> <span class="k">else</span> <span class="kc">None</span>

<span class="n">volcanoplot_rst</span> <span class="o">=</span> <span class="n">rst_exists</span><span class="p">(</span><span class="s2">&quot;reports/bioinfokit_volcanoplot.rst&quot;</span><span class="p">)</span>
<span class="n">bioinfokit_heatmap_rst</span> <span class="o">=</span> <span class="n">rst_exists</span><span class="p">(</span><span class="s2">&quot;reports/bioinfokit_heatmap.rst&quot;</span><span class="p">)</span>
<span class="n">bioinfokit_maplot_rst</span> <span class="o">=</span> <span class="n">rst_exists</span><span class="p">(</span><span class="s2">&quot;reports/bioinfokit_maplot.rst&quot;</span><span class="p">)</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule builds several PCA and their scree plot</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">bioinfokit_pcaplots</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">counts</span><span class="o">=</span><span class="s2">&quot;deseq2/filtered/filtered_counts.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">loadings_correlation_heatmap</span><span class="o">=</span><span class="s2">&quot;bioinfokit/figures/loadings.png&quot;</span><span class="p">,</span>
        <span class="n">pca_scree</span><span class="o">=</span><span class="s2">&quot;bioinfokit/figures/screeplot.png&quot;</span><span class="p">,</span>
        <span class="n">pca_2d</span><span class="o">=</span><span class="s2">&quot;bioinfokit/figures/pca2d.png&quot;</span><span class="p">,</span>
        <span class="n">pca_3d</span><span class="o">=</span><span class="s2">&quot;bioinfokit/figures/pca3d.png&quot;</span><span class="p">,</span>
        <span class="n">biplot_2d</span><span class="o">=</span><span class="s2">&quot;bioinfokit/figures/biplot2d.png&quot;</span><span class="p">,</span>
        <span class="n">biplot_3d</span><span class="o">=</span><span class="s2">&quot;bioinfokit/figures/biplot3d.png&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">read_csv</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;header&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> <span class="s2">&quot;index_col&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">}</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/pca.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bioinfokit/pca&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule builds a Volcanoplot from DESeq2 results</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">bioinfokit_volcanoplot</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;deseq2/wald/Cond_compairing_B_vs_A.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">report</span><span class="p">(</span>
            <span class="s2">&quot;bioinfokit/figures/volcanoplot.png&quot;</span><span class="p">,</span>
            <span class="n">caption</span><span class="o">=</span><span class="n">volcanoplot_rst</span><span class="p">,</span>
            <span class="n">category</span><span class="o">=</span><span class="s2">&quot;Volcano-Plot&quot;</span>
        <span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Plotting Volcanoplot with Bioinfokit&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">2048</span><span class="p">,</span> <span class="mi">10240</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">20</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">read_csv</span><span class="o">=</span><span class="p">{</span>
            <span class="s2">&quot;header&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span>
            <span class="s2">&quot;index_col&quot;</span><span class="p">:</span> <span class="kc">None</span>
        <span class="p">},</span>
        <span class="n">volcano</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bioinfokit&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;volcanoplot_extra&quot;</span><span class="p">,</span> <span class="p">{</span>
            <span class="s2">&quot;lfc&quot;</span><span class="p">:</span> <span class="s2">&quot;log2FoldChange&quot;</span><span class="p">,</span>
            <span class="s2">&quot;pv&quot;</span><span class="p">:</span> <span class="s2">&quot;padj&quot;</span><span class="p">,</span>
            <span class="s2">&quot;geneid&quot;</span><span class="p">:</span> <span class="s2">&quot;Unnamed: 0&quot;</span><span class="p">,</span>
            <span class="s2">&quot;lfc_thr&quot;</span><span class="p">:</span> <span class="p">(</span><span class="mf">0.001</span><span class="p">,</span> <span class="mf">0.001</span><span class="p">),</span>
            <span class="s2">&quot;pv_thr&quot;</span><span class="p">:</span> <span class="p">(</span><span class="mf">0.05</span><span class="p">,</span> <span class="mf">0.05</span><span class="p">),</span>
            <span class="s2">&quot;gstyle&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
            <span class="s2">&quot;sign_line&quot;</span><span class="p">:</span> <span class="kc">True</span><span class="p">,</span>
            <span class="s2">&quot;plotlegend&quot;</span><span class="p">:</span> <span class="kc">True</span>
        <span class="p">})</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bioinfokit/volcanoplot.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bioinfokit/volcanoplot&quot;</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule creates a sample clustered heatmap from the filtered-counts table</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">bioinfokit_sample_heatmap</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;deseq2/filtered/filtered_counts.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">report</span><span class="p">(</span>
            <span class="s2">&quot;bioinfokit/figures/sample_heatmap.png&quot;</span><span class="p">,</span>
            <span class="n">caption</span><span class="o">=</span><span class="n">bioinfokit_heatmap_rst</span><span class="p">,</span>
            <span class="n">category</span><span class="o">=</span><span class="s2">&quot;Clustered Heatmap&quot;</span>
        <span class="p">)</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">2048</span><span class="p">,</span> <span class="mi">10240</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">20</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">read_csv</span><span class="o">=</span><span class="p">{</span>
            <span class="s2">&quot;header&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span>
            <span class="s2">&quot;index_col&quot;</span><span class="p">:</span> <span class="mi">0</span>
        <span class="p">},</span>
        <span class="n">hmap</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bioinfokit&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;heatmap_extra&quot;</span><span class="p">,</span> <span class="p">{</span>
            <span class="s2">&quot;rowclus&quot;</span><span class="p">:</span> <span class="kc">True</span><span class="p">,</span>
            <span class="s2">&quot;colclus&quot;</span><span class="p">:</span> <span class="kc">True</span>
        <span class="p">})</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bioinfokit/sample_heatmap.png&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bioinfokit/heatmap&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule creates a MA-plot from DESeq2 merged results</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">bioinfokit_maplot</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;deseq2/filtered/merged.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">report</span><span class="p">(</span>
            <span class="s2">&quot;bioinfokit/figures/maplot.png&quot;</span><span class="p">,</span>
            <span class="n">caption</span><span class="o">=</span><span class="n">bioinfokit_maplot_rst</span><span class="p">,</span>
            <span class="n">category</span><span class="o">=</span><span class="s2">&quot;MA-Plot&quot;</span>
        <span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Building MA-plot&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">2048</span><span class="p">,</span> <span class="mi">10240</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">20</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">read_csv</span><span class="o">=</span><span class="p">{</span>
            <span class="s2">&quot;header&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span>
            <span class="s2">&quot;index_col&quot;</span><span class="p">:</span> <span class="mi">0</span>
        <span class="p">},</span>
        <span class="n">maplot</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bioinfokit&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;maplot_extra&quot;</span><span class="p">,</span> <span class="p">{</span>
            <span class="s2">&quot;lfc&quot;</span><span class="p">:</span> <span class="s2">&quot;log2FoldChange&quot;</span><span class="p">,</span>
            <span class="s2">&quot;ct_count&quot;</span><span class="p">:</span> <span class="s2">&quot;a.chr21&quot;</span><span class="p">,</span>
            <span class="s2">&quot;st_count&quot;</span><span class="p">:</span> <span class="s2">&quot;e.chr21&quot;</span>
        <span class="p">})</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bioinfokit/maplot.png&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bioinfokit/maplot&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule merges and filters both DESeq2 counts and results for further graphs</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">filter_deseq2</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">wald_tsv</span> <span class="o">=</span> <span class="s2">&quot;deseq2/wald/Cond_compairing_B_vs_A.tsv&quot;</span><span class="p">,</span>
        <span class="n">dst_tsv</span> <span class="o">=</span> <span class="s2">&quot;deseq2/dst/Cond_compairing_B_vs_A.tsv&quot;</span><span class="p">,</span>
        <span class="n">gene2gene</span> <span class="o">=</span> <span class="s2">&quot;tximport/gene2gene.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">filtered_counts</span><span class="o">=</span><span class="s2">&quot;deseq2/filtered/filtered_counts.tsv&quot;</span><span class="p">,</span>
        <span class="n">filtered_deseq2</span><span class="o">=</span><span class="s2">&quot;deseq2/filtered/filtered_deseq2.tsv&quot;</span><span class="p">,</span>
        <span class="n">merged_table</span><span class="o">=</span><span class="s2">&quot;deseq2/filtered/merged.tsv&quot;</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Filtering and merging DESeq2 results&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">4096</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">20</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">alpha</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;thresholds&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;deseq2_alpha&quot;</span><span class="p">,</span> <span class="mf">0.05</span><span class="p">),</span>
        <span class="n">fc_threshold</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;thresholds&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;fc_threshold&quot;</span><span class="p">,</span> <span class="mf">0.01</span><span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/deseq2/filter.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/pandas/deseq2_merge&quot;</span>



<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule build the conversion table from transcript to genes and their names.</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">gene_to_gene</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">gtf</span><span class="o">=</span><span class="s2">&quot;refs/ensembl/chr21.gtf&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">gene2gene_large</span><span class="o">=</span><span class="s2">&quot;tximport/gene2gene.tsv&quot;</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Building transcripts/genes conversion table&quot;</span>
    <span class="n">cache</span><span class="p">:</span> <span class="kc">True</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1536</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">gencode</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
        <span class="n">header</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
        <span class="n">positions</span> <span class="o">=</span> <span class="kc">True</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/tximport/tx2gene.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gtf/tx2gene&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely, as long as the dependencies between the rules remain as listed here.
For additional parameters in each individual wrapper, please refer to their corresponding documentation (see links below).</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</section>
<section id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this heading"></a></h2>
<p>The following individual wrappers are used in this meta-wrapper:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/bioinfokit/pca.html#bio-bioinfokit-pca"><span class="std std-ref">PCA</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/bioinfokit/volcanoplot.html#bio-bioinfokit-volcanoplot"><span class="std std-ref">BIOINFOKIT VOLCANOPLOT</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/bioinfokit/heatmap.html#bio-bioinfokit-heatmap"><span class="std std-ref">BIOINFOKIT HEATMAP</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/bioinfokit/maplot.html#bio-bioinfokit-maplot"><span class="std std-ref">BIOINFOKIT MA PLOT</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/pandas/deseq2_merge.html#bio-pandas-deseq2-merge"><span class="std std-ref">DESEQ2 MERGE</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/gtf/tx2gene.html#bio-gtf-tx2gene"><span class="std std-ref">TX2GENE</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this heading"></a></h2>
<p>MultiQC report is based on configurations from in-house scripts.</p>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="bcftools_annotate.html" class="btn btn-neutral float-left" title="BCFTOOLS ANNOTATE" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="bismark.html" class="btn btn-neutral float-right" title="BISKMARK" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>