<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>BISKMARK &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="BISKMARK_MULTISAMPLE" href="bismark_multisample.html" />
    <link rel="prev" title="DESEQ2_POST_PROCESS" href="bioinfokit.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bcftools_annotate.html">BCFTOOLS ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bioinfokit.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">BISKMARK</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="bismark_multisample.html">BISKMARK_MULTISAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bowtie2.html">BOWTIE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_fixmate.html">BWA_FIXMATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_mapping.html">BWA_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="clusterprofiler.html">CLUSTERP PROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="compress_index_vcf.html">COMPRESS_INDEX_VCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="coverage_variant_region.html">COVERAGE_VARIANT_REGION</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_pe.html">DADA2-PE</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_se.html">DADA2-SE</a></li>
<li class="toctree-l2"><a class="reference internal" href="deseq2_post_process.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="download_references.html">DOWNLOAD REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_cytoscan.html">EACON CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_oncoscan.html">EACON ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_post_process.html">EACON POST PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastq_screen_indexer.html">FASTQ_SCREEN_INDEXER</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk_bqsr.html">GATK_BQSR</a></li>
<li class="toctree-l2"><a class="reference internal" href="index_datasets.html">INDEX_DATASETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="maftools.html">MAFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller.html">META_CALLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller_germline.html">META_CALLER_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="msi_sensor_pro.html">MSISENSOR_PRO</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_germline.html">MUTECT2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_somatic.html">MUTECT2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_annotate.html">SNPEFF_ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="somatic_tmb.html">SOMATIC_TMB</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_arriba.html">STAR-ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_mapping.html">STAR_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="tximport_deseq2.html">TXIMPORT_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_calling.html">VARSCAN2_CALLING</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_germline.html">VARSCAN2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_somatic.html">VARSCAN2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="vcf_post_process.html">VCF_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="wisecondorx.html">WISECONDORX</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../meta-wrappers.html">Meta-Wrappers</a> &raquo;</li>
      <li>BISKMARK</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/meta-wrappers/bismark.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="biskmark">
<span id="id1"></span><h1>BISKMARK<a class="headerlink" href="#biskmark" title="Permalink to this headline"></a></h1>
<p>A pipeline to map bisulfite converted sequence reads and determine cytosine methylation states</p>
<section id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this headline"></a></h2>
<p>This meta-wrapper can be used by integrating the following into your workflow:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">default_config_bismark</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;genome&quot;</span><span class="p">:</span> <span class="s2">&quot;/path/to/fasta.fa&quot;</span><span class="p">,</span>
    <span class="s2">&quot;paired&quot;</span><span class="p">:</span> <span class="kc">True</span>
<span class="p">}</span>

<span class="k">try</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">config</span> <span class="o">==</span> <span class="nb">dict</span><span class="p">():</span>
        <span class="n">config</span> <span class="o">=</span> <span class="n">default_config_bismark</span>
<span class="k">except</span> <span class="ne">NameError</span><span class="p">:</span>
    <span class="n">config</span> <span class="o">=</span> <span class="n">default_config_bismark</span>


<span class="n">rule</span> <span class="n">bismark_reports_pe</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">alignment_report</span><span class="o">=</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">_PE_report.txt&quot;</span><span class="p">,</span>
        <span class="n">nucleotide_report</span><span class="o">=</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">.PE.nucleotide_stats.txt&quot;</span><span class="p">,</span>
        <span class="n">dedup_report</span><span class="o">=</span><span class="s2">&quot;bismark/deduplicated/</span><span class="si">{sample}</span><span class="s2">.deduplication_report.txt&quot;</span><span class="p">,</span>
        <span class="n">mbias_report</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">.PE.M-bias.txt&quot;</span><span class="p">,</span>
        <span class="n">splitting_report</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">_PE_splitting_report.txt&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">html</span><span class="o">=</span><span class="s2">&quot;bismark/report/</span><span class="si">{sample}</span><span class="s2">.PE.html&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Building HTML report for </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">15</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">skip_optional_reports</span><span class="o">=</span><span class="kc">False</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bismark/report/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bismark/bismark2report&quot;</span>


<span class="n">use</span> <span class="n">rule</span> <span class="n">bismark_reports_pe</span> <span class="k">as</span> <span class="n">bismark_reports_se</span> <span class="k">with</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">alignment_report</span><span class="o">=</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">_SE_report.txt&quot;</span><span class="p">,</span>
        <span class="n">nucleotide_report</span><span class="o">=</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">.SE.nucleotide_stats.txt&quot;</span><span class="p">,</span>
        <span class="n">dedup_report</span><span class="o">=</span><span class="s2">&quot;bismark/deduplicated/</span><span class="si">{sample}</span><span class="s2">.SE.deduplication_report.txt&quot;</span><span class="p">,</span>
        <span class="n">mbias_report</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">.SE.M-bias.txt&quot;</span><span class="p">,</span>
        <span class="n">splitting_report</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">_SE_splitting_report.txt&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">html</span><span class="o">=</span><span class="s2">&quot;bismark/report/</span><span class="si">{sample}</span><span class="s2">.SE.html&quot;</span>


<span class="n">rule</span> <span class="n">bismark_methylation_extractor_pe</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;bismark/deduplicated/</span><span class="si">{sample}</span><span class="s2">.PE.bam&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">mbias_r1</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">.M-bias_R1.png&quot;</span><span class="p">,</span>
        <span class="n">mbias_r2</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">.M-bias_R2.png&quot;</span><span class="p">,</span>
        <span class="n">mbias_report</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">.PE.M-bias.txt&quot;</span><span class="p">,</span>
        <span class="n">splitting_report</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">_PE_splitting_report.txt&quot;</span><span class="p">,</span>
        <span class="n">methylome_CpG_cov</span><span class="o">=</span><span class="s2">&quot;bismark/meth_cpg/</span><span class="si">{sample}</span><span class="s2">.bismark.cov.gz&quot;</span><span class="p">,</span>
        <span class="n">methylome_CpG_mlevel_bedGraph</span><span class="o">=</span><span class="s2">&quot;bismark/meth_cpg/</span><span class="si">{sample}</span><span class="s2">.bedGraph.gz&quot;</span><span class="p">,</span>
        <span class="n">read_base_meth_state_cpg</span><span class="o">=</span><span class="s2">&quot;bismark/meth/CpG_context_</span><span class="si">{sample}</span><span class="s2">.txt.gz&quot;</span><span class="p">,</span>
        <span class="n">read_base_meth_state_chg</span><span class="o">=</span><span class="s2">&quot;bismark/meth/CHG_context_</span><span class="si">{sample}</span><span class="s2">.txt.gz&quot;</span><span class="p">,</span>
        <span class="n">read_base_meth_state_chh</span><span class="o">=</span><span class="s2">&quot;bismark/meth/CHH_context_</span><span class="si">{sample}</span><span class="s2">.txt.gz&quot;</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Extracting methylation on </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">35</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bismark/meth/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">output_dir</span><span class="o">=</span><span class="s2">&quot;bismark/meth&quot;</span><span class="p">,</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;--gzip --comprehensive --bedGraph&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bismark/bismark_methylation_extractor&quot;</span>


<span class="n">use</span> <span class="n">rule</span> <span class="n">bismark_methylation_extractor_pe</span> <span class="k">as</span> <span class="n">bismark_methylation_extractor_se</span> <span class="k">with</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;bismark/deduplicated/</span><span class="si">{sample}</span><span class="s2">.SE.bam&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">mbias_r1</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">.M-bias_R1.png&quot;</span><span class="p">,</span>
        <span class="n">mbias_report</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">.SE.M-bias.txt&quot;</span><span class="p">,</span>
        <span class="n">splitting_report</span><span class="o">=</span><span class="s2">&quot;bismark/meth/</span><span class="si">{sample}</span><span class="s2">_SE_splitting_report.txt&quot;</span><span class="p">,</span>
        <span class="n">methylome_CpG_cov</span><span class="o">=</span><span class="s2">&quot;bismark/meth_cpg/</span><span class="si">{sample}</span><span class="s2">.bismark.cov.gz&quot;</span><span class="p">,</span>
        <span class="n">methylome_CpG_mlevel_bedGraph</span><span class="o">=</span><span class="s2">&quot;bismark/meth_cpg/</span><span class="si">{sample}</span><span class="s2">.bedGraph.gz&quot;</span><span class="p">,</span>
        <span class="n">read_base_meth_state_cpg</span><span class="o">=</span><span class="s2">&quot;bismark/meth/CpG_context_</span><span class="si">{sample}</span><span class="s2">.txt.gz&quot;</span><span class="p">,</span>
        <span class="n">read_base_meth_state_chg</span><span class="o">=</span><span class="s2">&quot;bismark/meth/CHG_context_</span><span class="si">{sample}</span><span class="s2">.txt.gz&quot;</span><span class="p">,</span>
        <span class="n">read_base_meth_state_chh</span><span class="o">=</span><span class="s2">&quot;bismark/meth/CHH_context_</span><span class="si">{sample}</span><span class="s2">.txt.gz&quot;</span>


<span class="n">rule</span> <span class="n">bismark_deduplicate_pe</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">.PE.bam&quot;</span><span class="p">,</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">bam</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/deduplicated/</span><span class="si">{sample}</span><span class="s2">.PE.bam&quot;</span><span class="p">),</span>
        <span class="n">report</span><span class="o">=</span><span class="s2">&quot;bismark/deduplicated/</span><span class="si">{sample}</span><span class="s2">.PE.deduplication_report.txt&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Deduplicating </span><span class="si">{wildcards.sample}</span><span class="s2"> pair-ended mapping&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">30</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bismark/deduplicated/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="s2">&quot;&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bismark/deduplicate_bismark&quot;</span>


<span class="n">use</span> <span class="n">rule</span> <span class="n">bismark_deduplicate_pe</span> <span class="k">as</span> <span class="n">bismark_deduplicate_se</span> <span class="k">with</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">.SE.bam&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">bam</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/deduplicated/</span><span class="si">{sample}</span><span class="s2">.SE.bam&quot;</span><span class="p">),</span>
        <span class="n">report</span><span class="o">=</span><span class="s2">&quot;bismark/deduplicated/</span><span class="si">{sample}</span><span class="s2">.SE.deduplication_report.txt&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Deduplicating </span><span class="si">{wildcards.sample}</span><span class="s2"> single-ended mapping&quot;</span>


<span class="n">rule</span> <span class="n">bismark_mapping_pair</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">genome</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">fq_1</span><span class="o">=</span><span class="s2">&quot;reads/</span><span class="si">{sample}</span><span class="s2">.1.fq.gz&quot;</span><span class="p">,</span>
        <span class="n">fq_2</span><span class="o">=</span><span class="s2">&quot;reads/</span><span class="si">{sample}</span><span class="s2">.2.fq.gz&quot;</span><span class="p">,</span>
        <span class="n">bismark_indexes_dir</span><span class="o">=</span><span class="s2">&quot;biskmark/index/Bisulfite_Genome&quot;</span><span class="p">,</span>
        <span class="n">genomic_freq</span><span class="o">=</span><span class="s2">&quot;biskmark/index/genomic_nucleotide_frequencies.txt&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">bam</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">.PE.bam&quot;</span><span class="p">),</span>
        <span class="n">report</span><span class="o">=</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">_PE_report.txt&quot;</span><span class="p">,</span>
        <span class="n">nucleotide_stats</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">.PE.nucleotide_stats.txt&quot;</span><span class="p">),</span>
        <span class="n">bam_unmapped_1</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">_unmapped_reads_1.fq.gz&quot;</span><span class="p">),</span>
        <span class="n">bam_unmapped_2</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">_unmapped_reads_2.fq.gz&quot;</span><span class="p">),</span>
        <span class="n">ambiguous_1</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">_ambiguous_reads_1.fq.gz&quot;</span><span class="p">),</span>
        <span class="n">ambiguous_2</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">_ambiguous_reads_2.fq.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Mapping pair-ended </span><span class="si">{wildcards.sample}</span><span class="s2"> with Bismark&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">8</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span> <span class="o">*</span> <span class="mi">120</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">60</span> <span class="o">*</span> <span class="mi">3</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bismark/mapping/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="k">lambda</span> <span class="n">w</span><span class="p">:</span> <span class="sa">f</span><span class="s2">&quot;--fastq --bam --gzip --phred33-quals --ambiguous --unmapped --nucleotide_coverage &quot;</span><span class="p">,</span> <span class="c1"># --rg_tag &#39;@RG\\tID:{w.sample}\\tSM:{w.sample}\\tPU:{w.sample}\\tPL:ILLUMINA\\tCN:IGR\\tDS:MiraSeq\\tPG:BOWTIE2&#39;&quot;,</span>
        <span class="c1">#basename=&quot;{sample}&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bismark/bismark&quot;</span>


<span class="n">use</span> <span class="n">rule</span> <span class="n">bismark_mapping_pair</span> <span class="k">as</span> <span class="n">bismark_mapping_single</span> <span class="k">with</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">genome</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">fq</span><span class="o">=</span><span class="s2">&quot;reads/</span><span class="si">{sample}</span><span class="s2">.fq.gz&quot;</span><span class="p">,</span>
        <span class="n">bismark_indexes_dir</span><span class="o">=</span><span class="s2">&quot;biskmark/index/Bisulfite_Genome&quot;</span><span class="p">,</span>
        <span class="n">genomic_freq</span><span class="o">=</span><span class="s2">&quot;biskmark/index/genomic_nucleotide_frequencies.txt&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">bam</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">.SE.bam&quot;</span><span class="p">),</span>
        <span class="n">report</span><span class="o">=</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">_SE_report.txt&quot;</span><span class="p">,</span>
        <span class="n">nucleotide_stats</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bismark/bams/</span><span class="si">{sample}</span><span class="s2">.SE.nucleotide_stats.txt&quot;</span><span class="p">),</span>
        <span class="c1">#bam_unmapped_1=temp(&quot;bismark/bams/{sample}_unmapped_reads_1.fq.gz&quot;),</span>
        <span class="c1">#ambiguous_1=temp(&quot;bismark/bams/{sample}_ambiguous_reads_1.fq.gz&quot;)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Mapping single-ended </span><span class="si">{wildcards.sample}</span><span class="s2"> with Bismark&quot;</span>



<span class="n">rule</span> <span class="n">bismark_genome_preparation</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;biskmark/index/sequence.fasta&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="n">directory</span><span class="p">(</span><span class="s2">&quot;biskmark/index/Bisulfite_Genome&quot;</span><span class="p">)),</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;biskmark/index/genomic_nucleotide_frequencies.txt&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Indexing genome with Bismark&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">7</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span> <span class="o">*</span> <span class="mi">40</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">60</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bismark/genome_preparation.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;--genomic_composition&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bismark/bismark_genome_preparation&quot;</span>


<span class="n">rule</span> <span class="n">link_genome</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;biskmark/index/sequence.fasta&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Linking fasta genome for bismark&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">512</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">5</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bismark/genome_linking.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="s2">&quot;-sfr&quot;</span>
    <span class="n">shell</span><span class="p">:</span>
        <span class="s2">&quot;ln </span><span class="si">{params}</span><span class="s2"> </span><span class="si">{input}</span><span class="s2"> </span><span class="si">{output}</span><span class="s2"> &gt; </span><span class="si">{log}</span><span class="s2"> 2&gt;&amp;1&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely, as long as the dependencies between the rules remain as listed here.
For additional parameters in each individual wrapper, please refer to their corresponding documentation (see links below).</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</section>
<section id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this headline"></a></h2>
<p>The following individual wrappers are used in this meta-wrapper:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/bismark/bismark2report.html#bio-bismark-bismark2report"><span class="std std-ref">BISMARK2REPORT</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/bismark/bismark_methylation_extractor.html#bio-bismark-bismark-methylation-extractor"><span class="std std-ref">BISMARK_METHYLATION_EXTRACTOR</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/bismark/deduplicate_bismark.html#bio-bismark-deduplicate-bismark"><span class="std std-ref">DEDUPLICATE_BISMARK</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/bismark/bismark_genome_preparation.html#bio-bismark-bismark-genome-preparation"><span class="std std-ref">BISMARK_GENOME_PREPARATION</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/bismark/bismark.html#bio-bismark-bismark"><span class="std std-ref">BISMARK</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="bioinfokit.html" class="btn btn-neutral float-left" title="DESEQ2_POST_PROCESS" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="bismark_multisample.html" class="btn btn-neutral float-right" title="BISKMARK_MULTISAMPLE" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>