

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>DADA2-SE &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
  <link rel="stylesheet" href="../_static/theme.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="DOWNLOAD REFERENCE" href="download_references.html" />
    <link rel="prev" title="DADA2-PE" href="dada2_pe.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Snakemake Wrappers
          

          
          </a>

          
            
            
              <div class="version">
                heads/Unofficial
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Wrappers and Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bwa_mapping.html">BWA_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_pe.html">DADA2-PE</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">DADA2-SE</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="download_references.html">DOWNLOAD REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastq_screen_indexer.html">FASTQ_SCREEN_INDEXER</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_arriba.html">STAR-ARRIBA</a></li>
</ul>
</li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../meta-wrappers.html">Meta-Wrappers</a> &raquo;</li>
        
      <li>DADA2-SE</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/meta-wrappers/dada2_se.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="dada2-se">
<span id="id1"></span><h1>DADA2-SE<a class="headerlink" href="#dada2-se" title="Permalink to this headline">¶</a></h1>
<p>A subworkflow for processing single-end sequences from metabarcoding projects in order to construct ASV tables using <code class="docutils literal notranslate"><span class="pre">DADA2</span></code>.  The example is based on the data provided in the <code class="docutils literal notranslate"><span class="pre">R</span></code> package. For more details, see the <a class="reference external" href="https://benjjneb.github.io/dada2/index.html">official website</a>. While the <a class="reference external" href="https://benjjneb.github.io/dada2/tutorial.html">tutorial</a> is tailored for paired-end sequences, useful information can be found regarding common functions to singled-end sequences processing.</p>
<div class="section" id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this headline">¶</a></h2>
<p>This meta-wrapper can be used by integrating the following into your workflow:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Make sure that you set the `truncLen=` option in the rule `dada2_filter_and_trim_se` according</span>
<span class="c1"># to the results of the quality profile checks (after rule `dada2_quality_profile_se` has finished on all samples).</span>
<span class="c1"># If in doubt, check https://benjjneb.github.io/dada2/tutorial.html#inspect-read-quality-profiles</span>

<span class="n">rule</span> <span class="nb">all</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="c1"># In a first run of this meta-wrapper, comment out all other inputs and only keep this one.</span>
        <span class="c1"># Looking at the resulting plot, adjust the `truncLen` in rule `dada2_filter_trim_se` and then</span>
        <span class="c1"># rerun with all inputs uncommented.</span>
        <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;reports/dada2/quality-profile/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{orientation}</span><span class="s2">-quality-profile.png&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;a&quot;</span><span class="p">,</span><span class="s2">&quot;b&quot;</span><span class="p">],</span> <span class="n">orientation</span><span class="o">=</span><span class="mi">1</span>
        <span class="p">),</span>
        <span class="s2">&quot;results/dada2/taxa.RDS&quot;</span>

<span class="n">rule</span> <span class="n">dada2_quality_profile_se</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="c1"># FASTQ file without primer sequences</span>
        <span class="s2">&quot;trimmed/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{orientation}</span><span class="s2">.fastq.gz&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;reports/dada2/quality-profile/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{orientation}</span><span class="s2">-quality-profile.png&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/dada2/quality-profile/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{orientation}</span><span class="s2">-quality-profile-se.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.68.0-353-g70fc74bae/bio/dada2/quality-profile&quot;</span>

<span class="n">rule</span> <span class="n">dada2_filter_trim_se</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="c1"># Single-end files without primer sequences</span>
        <span class="n">fwd</span><span class="o">=</span><span class="s2">&quot;trimmed/</span><span class="si">{sample}</span><span class="s2">.1.fastq.gz&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">filt</span><span class="o">=</span><span class="s2">&quot;filtered-se/</span><span class="si">{sample}</span><span class="s2">.1.fastq.gz&quot;</span><span class="p">,</span>
        <span class="n">stats</span><span class="o">=</span><span class="s2">&quot;reports/dada2/filter-trim-se/</span><span class="si">{sample}</span><span class="s2">.tsv&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="c1"># Set the maximum expected errors tolerated in filtered reads</span>
        <span class="n">maxEE</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
        <span class="c1"># Set the number of kept bases</span>
        <span class="n">truncLen</span><span class="o">=</span><span class="mi">240</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/dada2/filter-trim-se/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span> <span class="c1"># set desired number of threads here</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.68.0-353-g70fc74bae/bio/dada2/filter-trim&quot;</span>

<span class="n">rule</span> <span class="n">dada2_learn_errors</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
    <span class="c1"># Quality filtered and trimmed forward FASTQ files (potentially compressed)</span>
        <span class="n">expand</span><span class="p">(</span><span class="s2">&quot;filtered-se/</span><span class="si">{sample}</span><span class="s2">.{{orientation}}.fastq.gz&quot;</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;a&quot;</span><span class="p">,</span><span class="s2">&quot;b&quot;</span><span class="p">])</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">err</span><span class="o">=</span><span class="s2">&quot;results/dada2/model_</span><span class="si">{orientation}</span><span class="s2">.RDS&quot;</span><span class="p">,</span><span class="c1"># save the error model</span>
        <span class="n">plot</span><span class="o">=</span><span class="s2">&quot;reports/dada2/errors_</span><span class="si">{orientation}</span><span class="s2">.png&quot;</span><span class="p">,</span><span class="c1"># plot observed and estimated rates</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">randomize</span><span class="o">=</span><span class="kc">True</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/dada2/learn-errors/learn-errors_</span><span class="si">{orientation}</span><span class="s2">.log&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span> <span class="c1"># set desired number of threads here</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.68.0-353-g70fc74bae/bio/dada2/learn-errors&quot;</span>

<span class="n">rule</span> <span class="n">dada2_dereplicate_fastq</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
    <span class="c1"># Quality filtered FASTQ file</span>
        <span class="s2">&quot;filtered-se/</span><span class="si">{fastq}</span><span class="s2">.fastq.gz&quot;</span>
    <span class="n">output</span><span class="p">:</span>
    <span class="c1"># Dereplicated sequences stored as `derep-class` object in a RDS file</span>
        <span class="s2">&quot;uniques/</span><span class="si">{fastq}</span><span class="s2">.RDS&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/dada2/dereplicate-fastq/</span><span class="si">{fastq}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.68.0-353-g70fc74bae/bio/dada2/dereplicate-fastq&quot;</span>

<span class="n">rule</span> <span class="n">dada2_sample_inference</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
    <span class="c1"># Dereplicated (aka unique) sequences of the sample</span>
        <span class="n">derep</span><span class="o">=</span><span class="s2">&quot;uniques/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{orientation}</span><span class="s2">.RDS&quot;</span><span class="p">,</span>
        <span class="n">err</span><span class="o">=</span><span class="s2">&quot;results/dada2/model_</span><span class="si">{orientation}</span><span class="s2">.RDS&quot;</span> <span class="c1"># Error model</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;denoised/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{orientation}</span><span class="s2">.RDS&quot;</span> <span class="c1"># Inferred sample composition</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/dada2/sample-inference/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{orientation}</span><span class="s2">.log&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span> <span class="c1"># set desired number of threads here</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.68.0-353-g70fc74bae/bio/dada2/sample-inference&quot;</span>

<span class="n">rule</span> <span class="n">dada2_make_table_se</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
    <span class="c1"># Inferred composition</span>
        <span class="n">expand</span><span class="p">(</span><span class="s2">&quot;denoised/</span><span class="si">{sample}</span><span class="s2">.1.RDS&quot;</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;a&#39;</span><span class="p">,</span><span class="s1">&#39;b&#39;</span><span class="p">])</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;results/dada2/seqTab-se.RDS&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;a&#39;</span><span class="p">,</span><span class="s1">&#39;b&#39;</span><span class="p">]</span> <span class="c1"># Sample names instead of paths</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/dada2/make-table/make-table-se.log&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span> <span class="c1"># set desired number of threads here</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.68.0-353-g70fc74bae/bio/dada2/make-table&quot;</span>

<span class="n">rule</span> <span class="n">dada2_remove_chimeras</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;results/dada2/seqTab-se.RDS&quot;</span> <span class="c1"># Sequence table</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;results/dada2/seqTab.nochimeras.RDS&quot;</span> <span class="c1"># Chimera-free sequence table</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/dada2/remove-chimeras/remove-chimeras.log&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span> <span class="c1"># set desired number of threads here</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.68.0-353-g70fc74bae/bio/dada2/remove-chimeras&quot;</span>

<span class="n">rule</span> <span class="n">dada2_collapse_nomismatch</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;results/dada2/seqTab.nochimeras.RDS&quot;</span> <span class="c1"># Chimera-free sequence table</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;results/dada2/seqTab.collapsed.RDS&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/dada2/collapse-nomismatch/collapse-nomismatch.log&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span> <span class="c1"># set desired number of threads here</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.68.0-353-g70fc74bae/bio/dada2/collapse-nomismatch&quot;</span>

<span class="n">rule</span> <span class="n">dada2_assign_taxonomy</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">seqs</span><span class="o">=</span><span class="s2">&quot;results/dada2/seqTab.collapsed.RDS&quot;</span><span class="p">,</span> <span class="c1"># Chimera-free sequence table</span>
        <span class="n">refFasta</span><span class="o">=</span><span class="s2">&quot;resources/example_train_set.fa.gz&quot;</span> <span class="c1"># Reference FASTA for taxonomy</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;results/dada2/taxa.RDS&quot;</span> <span class="c1"># Taxonomic assignments</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/dada2/assign-taxonomy/assign-taxonomy.log&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span> <span class="c1"># set desired number of threads here</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.68.0-353-g70fc74bae/bio/dada2/assign-taxonomy&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely, as long as the dependencies between the rules remain as listed here.
For additional parameters in each individual wrapper, please refer to their corresponding documentation (see links below).</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</div>
<div class="section" id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this headline">¶</a></h2>
<p>The following individual wrappers are used in this meta-wrapper:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/dada2/quality-profile.html#bio-dada2-quality-profile"><span class="std std-ref">DADA2_QUALITY_PROFILES</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/dada2/filter-trim.html#bio-dada2-filter-trim"><span class="std std-ref">DADA2_FILTER_TRIM</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/dada2/learn-errors.html#bio-dada2-learn-errors"><span class="std std-ref">DADA2_LEARN_ERRORS</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/dada2/dereplicate-fastq.html#bio-dada2-dereplicate-fastq"><span class="std std-ref">DADA2_DEREPLICATE_FASTQ</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/dada2/sample-inference.html#bio-dada2-sample-inference"><span class="std std-ref">DADA2_SAMPLE_INFERENCE</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/dada2/make-table.html#bio-dada2-make-table"><span class="std std-ref">DADA2_MAKE_TABLE</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/dada2/remove-chimeras.html#bio-dada2-remove-chimeras"><span class="std std-ref">DADA2_REMOVE_CHIMERAS</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/dada2/collapse-nomismatch.html#bio-dada2-collapse-nomismatch"><span class="std std-ref">DADA2_COLLAPSE_NOMISMATCH</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/dada2/assign-taxonomy.html#bio-dada2-assign-taxonomy"><span class="std std-ref">DADA2_ASSIGN_TAXONOMY</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</div>
<div class="section" id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Charlie Pauvert</p></li>
</ul>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="download_references.html" class="btn btn-neutral float-right" title="DOWNLOAD REFERENCE" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="dada2_pe.html" class="btn btn-neutral float-left" title="DADA2-PE" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2016, The Snakemake team

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>