<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>DOWNLOAD REFERENCE &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="EACON CYTOSCAN" href="eacon_cytoscan.html" />
    <link rel="prev" title="DESEQ2_POST_PROCESS" href="deseq2_post_process.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bcftools_annotate.html">BCFTOOLS ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bioinfokit.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="bismark.html">BISKMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="bismark_multisample.html">BISKMARK_MULTISAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bowtie2.html">BOWTIE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_fixmate.html">BWA_FIXMATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_mapping.html">BWA_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="clusterprofiler.html">CLUSTERP PROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="compress_index_vcf.html">COMPRESS_INDEX_VCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="coverage_variant_region.html">COVERAGE_VARIANT_REGION</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_pe.html">DADA2-PE</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_se.html">DADA2-SE</a></li>
<li class="toctree-l2"><a class="reference internal" href="deseq2_post_process.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">DOWNLOAD REFERENCE</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="eacon_cytoscan.html">EACON CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_oncoscan.html">EACON ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_post_process.html">EACON POST PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastq_screen_indexer.html">FASTQ_SCREEN_INDEXER</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk_bqsr.html">GATK_BQSR</a></li>
<li class="toctree-l2"><a class="reference internal" href="index_datasets.html">INDEX_DATASETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="maftools.html">MAFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller.html">META_CALLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller_germline.html">META_CALLER_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="msi_sensor_pro.html">MSISENSOR_PRO</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_germline.html">MUTECT2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_somatic.html">MUTECT2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_annotate.html">SNPEFF_ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="somatic_tmb.html">SOMATIC_TMB</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_arriba.html">STAR-ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_mapping.html">STAR_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="tximport_deseq2.html">TXIMPORT_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_calling.html">VARSCAN2_CALLING</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_germline.html">VARSCAN2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_somatic.html">VARSCAN2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="vcf_post_process.html">VCF_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="wisecondorx.html">WISECONDORX</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../meta-wrappers.html">Meta-Wrappers</a> &raquo;</li>
      <li>DOWNLOAD REFERENCE</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/meta-wrappers/download_references.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="download-reference">
<span id="id1"></span><h1>DOWNLOAD REFERENCE<a class="headerlink" href="#download-reference" title="Permalink to this heading"></a></h1>
<p>Download sequence and annotations from Ensembl</p>
<section id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this heading"></a></h2>
<p>This meta-wrapper can be used by integrating the following into your workflow:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="k">try</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">config</span> <span class="o">==</span> <span class="nb">dict</span><span class="p">():</span>
        <span class="n">config</span> <span class="o">=</span> <span class="p">{</span>
            <span class="s2">&quot;download_references&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="s2">&quot;fasta_datatype&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;dna&quot;</span><span class="p">,</span> <span class="s2">&quot;cdna&quot;</span><span class="p">,</span> <span class="s2">&quot;ncrna&quot;</span><span class="p">],</span>
                <span class="s2">&quot;build_release_organism&quot;</span><span class="p">:</span> <span class="p">[</span>
                    <span class="s2">&quot;GRCh38.99.homo_sapiens&quot;</span><span class="p">,</span>
                    <span class="s2">&quot;GRCm38.99.mus_musculus&quot;</span>
                <span class="p">]</span>
            <span class="p">}</span>
        <span class="p">}</span>
<span class="k">except</span> <span class="ne">NameError</span><span class="p">:</span>
    <span class="n">config</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s2">&quot;download_references&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;fasta_datatype&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;dna&quot;</span><span class="p">,</span> <span class="s2">&quot;cdna&quot;</span><span class="p">,</span> <span class="s2">&quot;ncrna&quot;</span><span class="p">],</span>
            <span class="s2">&quot;build_release_organism&quot;</span><span class="p">:</span> <span class="p">[</span>
                <span class="s2">&quot;GRCh38.99.homo_sapiens&quot;</span><span class="p">,</span>
                <span class="s2">&quot;GRCm38.99.mus_musculus&quot;</span>
            <span class="p">]</span>
        <span class="p">}</span>
    <span class="p">}</span>

<span class="n">fasta_datatype</span> <span class="o">=</span> <span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;download_references&quot;</span><span class="p">,</span> <span class="p">{})</span><span class="o">.</span><span class="n">get</span><span class="p">(</span>
    <span class="s2">&quot;fasta_datatype&quot;</span><span class="p">,</span> <span class="p">[</span><span class="s2">&quot;dna&quot;</span><span class="p">,</span> <span class="s2">&quot;cdna&quot;</span><span class="p">,</span> <span class="s2">&quot;ncrna&quot;</span><span class="p">]</span>
<span class="p">)</span>
<span class="n">build_release_organism</span> <span class="o">=</span> <span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;download_references&quot;</span><span class="p">,</span> <span class="p">{})</span><span class="o">.</span><span class="n">get</span><span class="p">(</span>
    <span class="s2">&quot;build_release_organism&quot;</span><span class="p">,</span>
    <span class="p">[</span>
        <span class="s2">&quot;GRCh38.99.homo_sapiens&quot;</span><span class="p">,</span>
        <span class="s2">&quot;GRCm38.99.mus_musculus&quot;</span>
    <span class="p">]</span>
<span class="p">)</span>

<span class="n">rule</span> <span class="nb">all</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">fasta</span> <span class="o">=</span> <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;refs/ensembl/</span><span class="si">{build_release_organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.fasta&quot;</span><span class="p">,</span>
            <span class="n">build_release_organism</span><span class="o">=</span><span class="n">build_release_organism</span><span class="p">,</span>
            <span class="n">datatype</span><span class="o">=</span><span class="n">fasta_datatype</span>
        <span class="p">),</span>
        <span class="n">fasta_index</span> <span class="o">=</span> <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;refs/ensembl/</span><span class="si">{build_release_organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.fasta.fai&quot;</span><span class="p">,</span>
            <span class="n">build_release_organism</span><span class="o">=</span><span class="n">build_release_organism</span><span class="p">,</span>
            <span class="n">datatype</span><span class="o">=</span><span class="n">fasta_datatype</span>
        <span class="p">),</span>
        <span class="n">fasta_dict</span> <span class="o">=</span> <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;refs/ensembl/</span><span class="si">{build_release_organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.dict&quot;</span><span class="p">,</span>
            <span class="n">build_release_organism</span><span class="o">=</span><span class="n">build_release_organism</span><span class="p">,</span>
            <span class="n">datatype</span><span class="o">=</span><span class="n">fasta_datatype</span>
        <span class="p">),</span>
        <span class="n">gtf</span> <span class="o">=</span> <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;refs/ensembl/</span><span class="si">{build_release_organism}</span><span class="s2">.gtf&quot;</span><span class="p">,</span>
            <span class="n">build_release_organism</span><span class="o">=</span><span class="n">build_release_organism</span>
        <span class="p">),</span>
        <span class="n">vcf</span> <span class="o">=</span> <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;refs/ensembl/</span><span class="si">{build_release_organism}</span><span class="s2">.all.vcf.gz&quot;</span><span class="p">,</span>
            <span class="n">build_release_organism</span><span class="o">=</span><span class="n">build_release_organism</span>
        <span class="p">)</span>


<span class="n">rule</span> <span class="n">get_genome</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;refs/ensembl/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.fasta&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">512</span><span class="p">,</span> <span class="mi">2048</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">120</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">species</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{organism}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">datatype</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{datatype}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">build</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{build}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">release</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{release}</span><span class="s2">&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/get_genome/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.log&quot;</span>
    <span class="n">cache</span><span class="p">:</span> <span class="kc">True</span>  <span class="c1"># save space and time with between workflow caching (see docs)</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/reference/ensembl-sequence&quot;</span>


<span class="n">rule</span> <span class="n">get_annotation</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;refs/ensembl/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.gtf&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">512</span><span class="p">,</span> <span class="mi">2048</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">120</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">species</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{organism}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">release</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{release}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">build</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{build}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">fmt</span><span class="o">=</span><span class="s2">&quot;gtf&quot;</span><span class="p">,</span>
        <span class="n">flavor</span><span class="o">=</span><span class="s2">&quot;&quot;</span> <span class="c1"># optional, e.g. chr_patch_hapl_scaff, see Ensembl FTP.</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/get_annotation/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.log&quot;</span>
    <span class="n">cache</span><span class="p">:</span> <span class="kc">True</span>  <span class="c1"># save space and time with between workflow caching (see docs)</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/reference/ensembl-annotation&quot;</span>


<span class="n">rule</span> <span class="n">samtools_faidx_reference</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;refs/ensembl/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.fasta&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">512</span><span class="p">,</span> <span class="mi">2048</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">120</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;refs/ensembl/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.fasta.fai&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="s2">&quot;&quot;</span> <span class="c1"># optional params string</span>
    <span class="n">cache</span><span class="p">:</span> <span class="kc">True</span>
    <span class="n">group</span><span class="p">:</span> <span class="s2">&quot;index_fasta&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/samtools/faidx&quot;</span>


<span class="n">rule</span> <span class="n">create_dict</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;refs/ensembl/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.fasta&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;refs/ensembl/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.dict&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/picard/create_dict/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.</span><span class="si">{datatype}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;&quot;</span>  <span class="c1"># optional: extra arguments for picard.</span>
    <span class="n">cache</span><span class="p">:</span> <span class="kc">True</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span> <span class="mi">2048</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">120</span>
    <span class="n">group</span><span class="p">:</span> <span class="s2">&quot;index_fasta&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/picard/createsequencedictionary&quot;</span>


<span class="n">rule</span> <span class="n">get_variation_with_contig_lengths</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">fai</span><span class="o">=</span><span class="s2">&quot;refs/ensembl/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.dna.fasta.fai&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">vcf</span><span class="o">=</span><span class="s2">&quot;refs/ensembl/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.all.vcf.gz&quot;</span>
    <span class="n">cache</span><span class="p">:</span> <span class="kc">True</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">512</span><span class="p">,</span> <span class="mi">2048</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">120</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">species</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{organism}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">release</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{release}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">build</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{build}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="nb">type</span><span class="o">=</span><span class="s2">&quot;all&quot;</span> <span class="c1"># one of &quot;all&quot;, &quot;somatic&quot;, &quot;structural_variation&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/get_variation/</span><span class="si">{build}</span><span class="s2">.</span><span class="si">{release}</span><span class="s2">.</span><span class="si">{organism}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/reference/ensembl-variation&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely, as long as the dependencies between the rules remain as listed here.
For additional parameters in each individual wrapper, please refer to their corresponding documentation (see links below).</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</section>
<section id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this heading"></a></h2>
<p>The following individual wrappers are used in this meta-wrapper:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/reference/ensembl-sequence.html#bio-reference-ensembl-sequence"><span class="std std-ref">ENSEMBL-SEQUENCE</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/reference/ensembl-annotation.html#bio-reference-ensembl-annotation"><span class="std std-ref">ENSEMBL-ANNOTATION</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/reference/ensembl-variation.html#bio-reference-ensembl-variation"><span class="std std-ref">ENSEMBL-VARIATION</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/samtools/faidx.html#bio-samtools-faidx"><span class="std std-ref">SAMTOOLS FAIDX</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/picard/createsequencedictionary.html#bio-picard-createsequencedictionary"><span class="std std-ref">PICARD CREATESEQUENCEDICTIONARY</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this heading"></a></h2>
<p>Do not forget to cache these downloads!</p>
<p>The samtools index step is here to include genome intervals in the VCF index. By doing so, the VCF is compatible with GATK for variant calling.</p>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="deseq2_post_process.html" class="btn btn-neutral float-left" title="DESEQ2_POST_PROCESS" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="eacon_cytoscan.html" class="btn btn-neutral float-right" title="EACON CYTOSCAN" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>