<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>MUTECT2_SOMATIC &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="SALMON" href="salmon.html" />
    <link rel="prev" title="MUTECT2_GERMLINE" href="mutect2_germline.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bcftools_annotate.html">BCFTOOLS ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bioinfokit.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="bismark.html">BISKMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="bismark_multisample.html">BISKMARK_MULTISAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bowtie2.html">BOWTIE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_fixmate.html">BWA_FIXMATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_mapping.html">BWA_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="clusterprofiler.html">CLUSTERP PROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="compress_index_vcf.html">COMPRESS_INDEX_VCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="coverage_variant_region.html">COVERAGE_VARIANT_REGION</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_pe.html">DADA2-PE</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_se.html">DADA2-SE</a></li>
<li class="toctree-l2"><a class="reference internal" href="deseq2_post_process.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="download_references.html">DOWNLOAD REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_cytoscan.html">EACON CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_oncoscan.html">EACON ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_post_process.html">EACON POST PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastq_screen_indexer.html">FASTQ_SCREEN_INDEXER</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk_bqsr.html">GATK_BQSR</a></li>
<li class="toctree-l2"><a class="reference internal" href="index_datasets.html">INDEX_DATASETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="maftools.html">MAFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller.html">META_CALLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller_germline.html">META_CALLER_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="msi_sensor_pro.html">MSISENSOR_PRO</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_germline.html">MUTECT2_GERMLINE</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">MUTECT2_SOMATIC</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_annotate.html">SNPEFF_ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="somatic_tmb.html">SOMATIC_TMB</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_arriba.html">STAR-ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_mapping.html">STAR_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="tximport_deseq2.html">TXIMPORT_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_calling.html">VARSCAN2_CALLING</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_germline.html">VARSCAN2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_somatic.html">VARSCAN2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="vcf_post_process.html">VCF_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="wisecondorx.html">WISECONDORX</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../meta-wrappers.html">Meta-Wrappers</a> &raquo;</li>
      <li>MUTECT2_SOMATIC</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/meta-wrappers/mutect2_somatic.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="mutect2-somatic">
<span id="id1"></span><h1>MUTECT2_SOMATIC<a class="headerlink" href="#mutect2-somatic" title="Permalink to this heading"></a></h1>
<p>Call variants on BAM files with Mutect2, considering a pair of bam file.</p>
<section id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this heading"></a></h2>
<p>This meta-wrapper can be used by integrating the following into your workflow:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">typing</span> <span class="kn">import</span> <span class="n">Union</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>

<span class="n">worflow_source_dir</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="nb">next</span><span class="p">(</span><span class="nb">iter</span><span class="p">(</span><span class="n">workflow</span><span class="o">.</span><span class="n">get_sources</span><span class="p">())))</span><span class="o">.</span><span class="n">absolute</span><span class="p">()</span><span class="o">.</span><span class="n">parent</span>
<span class="n">common</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">worflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;../../../../bigr_pipelines/common/python&quot;</span><span class="p">)</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">common</span><span class="p">)</span>

<span class="kn">from</span> <span class="nn">file_manager</span> <span class="kn">import</span> <span class="o">*</span>

<span class="n">default_config_mutect2_somatic</span> <span class="o">=</span> <span class="p">{</span>
    <span class="c1"># Your genome sequence</span>
    <span class="s2">&quot;genome&quot;</span><span class="p">:</span> <span class="s2">&quot;reference/genome.fasta&quot;</span><span class="p">,</span>
    <span class="c1"># Path to a VCF containing AF fields</span>
    <span class="s2">&quot;known&quot;</span><span class="p">:</span> <span class="s2">&quot;reference/dbsnp.vcf.gz&quot;</span><span class="p">,</span>
    <span class="c1"># Path to a BED containing the kit&#39;s catpured regions</span>
    <span class="s2">&quot;bed&quot;</span><span class="p">:</span> <span class="s2">&quot;reference/regions.bed&quot;</span><span class="p">,</span>
    <span class="c1"># Path to dbSNP vcf, its tbi should be aside.</span>
    <span class="s2">&quot;dbsnp&quot;</span><span class="p">:</span> <span class="s2">&quot;reference/dbsnp.vcf.gz&quot;</span><span class="p">,</span>
    <span class="c1"># List of samples</span>
    <span class="s2">&quot;sample_list&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;S1&quot;</span><span class="p">,</span> <span class="s2">&quot;S2&quot;</span><span class="p">],</span>
    <span class="c1"># Extra parameters</span>
    <span class="s2">&quot;gatk_crate_stomatic_pon_extra&quot;</span><span class="p">:</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
    <span class="s2">&quot;gatk_get_pileup_summaries_extra&quot;</span><span class="p">:</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
    <span class="s2">&quot;gatk_calculate_tumor_contamination_extra&quot;</span><span class="p">:</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
    <span class="s2">&quot;gatk_filter_mutect_calls_extra&quot;</span><span class="p">:</span> <span class="s2">&quot;&quot;</span>
<span class="p">}</span>

<span class="k">try</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">config</span> <span class="o">==</span> <span class="nb">dict</span><span class="p">():</span>
        <span class="n">config</span> <span class="o">=</span> <span class="n">default_config_mutect2_somatic</span>
<span class="k">except</span> <span class="ne">NameError</span><span class="p">:</span>
    <span class="n">config</span> <span class="o">=</span> <span class="n">default_config_mutect2_somatic</span>

<span class="k">def</span> <span class="nf">mutect2_input</span><span class="p">(</span><span class="n">config</span><span class="p">:</span> <span class="nb">dict</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">Union</span><span class="p">[</span><span class="nb">list</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span> <span class="nb">str</span><span class="p">]])</span> <span class="o">-&gt;</span> <span class="nb">dict</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="nb">str</span><span class="p">]:</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    PoN should be used if and only if more than 40 normal samples are</span>
<span class="sd">    provided.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">input_dict</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span>
        <span class="n">fasta</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">fasta_index</span><span class="o">=</span><span class="n">get_fai</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">fasta_dict</span><span class="o">=</span><span class="n">get_dict</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="nb">map</span><span class="o">=</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span><span class="p">,</span>
        <span class="n">map_index</span><span class="o">=</span><span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span><span class="p">),</span>
        <span class="n">tumor</span><span class="o">=</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_tumor.bam&quot;</span><span class="p">,</span>
        <span class="n">tumor_index</span><span class="o">=</span><span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_tumor.bam&quot;</span><span class="p">),</span>
        <span class="n">germline</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;known&quot;</span><span class="p">],</span>
        <span class="n">germline_tbi</span><span class="o">=</span><span class="n">get_tbi</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;known&quot;</span><span class="p">]),</span>
        <span class="n">intervals</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bed&quot;</span><span class="p">]</span>
    <span class="p">)</span>
    <span class="c1"># if len(config[&quot;sample_list&quot;]) &gt;= 40:</span>
    <span class="c1">#     input_dict[&quot;pon&quot;] = &quot;gatk/pon/PoN.g.vcf.gz&quot;</span>
    <span class="c1">#     input_dict[&quot;pon_index&quot;] = get_tbi(&quot;gatk/pon/PoN.g.vcf.gz&quot;)</span>

    <span class="k">return</span> <span class="n">input_dict</span>

<span class="c1">##################################</span>
<span class="c1">### Correct multiallelic sites ###</span>
<span class="c1">##################################</span>

<span class="n">rule</span> <span class="n">split_multiallelic_mutect2</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">call</span><span class="o">=</span><span class="s2">&quot;mutect2/filter/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">,</span>
        <span class="n">idx</span><span class="o">=</span><span class="n">get_tbi</span><span class="p">(</span><span class="s2">&quot;mutect2/filter/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">),</span>
        <span class="n">fasta</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;bcftools/mutect2/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Splitting Mutect2 multiallelic sites </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">5120</span><span class="p">,</span> <span class="mi">15360</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">35</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span> <span class="o">=</span> <span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span>
            <span class="s2">&quot;gatk_bcftools_split_multiallelic&quot;</span><span class="p">,</span>
            <span class="s2">&quot;-m -both --check-ref w&quot;</span>
        <span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bcftools/norm/mutect2/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bcftools/norm&quot;</span>


<span class="c1">###########################</span>
<span class="c1">### Filter Mutect calls ###</span>
<span class="c1">###########################</span>

<span class="n">rule</span> <span class="n">gatk_filter_mutect_calls</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">vcf</span><span class="o">=</span><span class="s2">&quot;mutect2/call/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">,</span>
        <span class="n">vcf_tbi</span><span class="o">=</span><span class="n">get_tbi</span><span class="p">(</span><span class="s2">&quot;mutect2/call/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">),</span>
        <span class="n">ref</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">ref_index</span><span class="o">=</span><span class="n">get_fai</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">ref_dict</span><span class="o">=</span><span class="n">get_dict</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">bam</span><span class="o">=</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_tumor.bam&quot;</span><span class="p">,</span>
        <span class="n">bam_index</span><span class="o">=</span><span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_tumor.bam&quot;</span><span class="p">),</span>
        <span class="c1">#f1r2=&quot;mutect2/f1r2/{sample}.tar.gz&quot;,</span>
        <span class="n">f1r2</span><span class="o">=</span><span class="s2">&quot;gatk/orientation_model/</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">.artifacts-prior.tar.gz&quot;</span><span class="p">,</span>
        <span class="n">contamination</span><span class="o">=</span><span class="s2">&quot;summary/</span><span class="si">{sample}</span><span class="s2">_calculate_contamination.table&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">vcf</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;mutect2/filter/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">),</span>
        <span class="n">vcf_index</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;mutect2/filter/</span><span class="si">{sample}</span><span class="s2">.vcf.gz.tbi&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Filtering GATK calls on </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">5120</span><span class="p">,</span> <span class="mi">15360</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">35</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span>
            <span class="s2">&quot;gatk_filter_mutect_calls_extra&quot;</span><span class="p">,</span>
            <span class="s2">&quot;--create-output-variant-index --min-median-mapping-quality 35&quot;</span>
        <span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/mutect2/filter/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gatk/filtermutectcalls&quot;</span>


<span class="c1">###########################################</span>
<span class="c1">### Estimate cross-sample contamination ###</span>
<span class="c1">###########################################</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">Estimate possible contaminations</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">calculate_tumor_contamination</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">summary</span><span class="o">=</span><span class="s2">&quot;gatk/getpileupsummaries/</span><span class="si">{sample}</span><span class="s2">_tumor_getpileupsummaries.table&quot;</span><span class="p">,</span>
        <span class="n">normal</span><span class="o">=</span><span class="s2">&quot;gatk/getpileupsummaries/</span><span class="si">{sample}</span><span class="s2">_normal_getpileupsummaries.table&quot;</span><span class="p">,</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">table</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;summary/</span><span class="si">{sample}</span><span class="s2">_calculate_contamination.table&quot;</span><span class="p">),</span>
        <span class="n">segmentation</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;summary/</span><span class="si">{sample}</span><span class="s2">_segments.table&quot;</span><span class="p">)</span>
    <span class="n">group</span><span class="p">:</span>
        <span class="s2">&quot;Contamination_Estimate&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Summarizing read support for known variant sites to further &quot;</span>
        <span class="s2">&quot;estimate contamination on </span><span class="si">{wildcards.sample}</span><span class="s2"> (on tumor only)&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">5120</span><span class="p">,</span> <span class="mi">15360</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">35</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;gatk_calculate_tumor_contamination_extra&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/gatk/CalculateContamination/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gatk/calculatecontamination&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">Summarize the read support over known variants</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">get_pileup_summaries</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">bam</span><span class="o">=</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_</span><span class="si">{status}</span><span class="s2">.bam&quot;</span><span class="p">,</span>
        <span class="n">bam_index</span><span class="o">=</span><span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_</span><span class="si">{status}</span><span class="s2">.bam&quot;</span><span class="p">),</span>
        <span class="n">intervals</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bed&quot;</span><span class="p">],</span>
        <span class="n">variants</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;known&quot;</span><span class="p">],</span>
        <span class="n">variants_index</span><span class="o">=</span><span class="n">get_tbi</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;known&quot;</span><span class="p">])</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">table</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span>
            <span class="s2">&quot;gatk/getpileupsummaries/</span><span class="si">{sample}</span><span class="s2">_</span><span class="si">{status}</span><span class="s2">_getpileupsummaries.table&quot;</span>
        <span class="p">)</span>
    <span class="n">group</span><span class="p">:</span>
        <span class="s2">&quot;Contamination_Estimate&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Summarizing read support for known variant sites to further &quot;</span>
        <span class="s2">&quot;estimate contamination on </span><span class="si">{wildcards.sample}</span><span class="s2"> (</span><span class="si">{wildcards.status}</span><span class="s2">)&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">5120</span><span class="p">,</span> <span class="mi">15360</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">35</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;gatk_get_pileup_summaries_extra&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/gatk/GetPileupSummaries/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{status}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gatk/getpileupsummaries&quot;</span>



<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">Build orientation bias model to filter false positive calls</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">gatk_learn_read_orientation_model</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">f1r2</span><span class="o">=</span><span class="s2">&quot;mutect2/f1r2/</span><span class="si">{sample}</span><span class="s2">.tar.gz&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;gatk/orientation_model/</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">.artifacts-prior.tar.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Learning orientation bias in </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">60</span><span class="p">,</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">8192</span><span class="p">,</span> <span class="mi">20480</span><span class="p">),</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;gatk/orientation_model/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gatk/learnreadorientationmodel&quot;</span>



<span class="c1">######################</span>
<span class="c1">### Actual Calling ###</span>
<span class="c1">######################</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule calls somatic variants with GATK Mutect2</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">mutect2_somatic</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="o">**</span><span class="n">mutect2_input</span><span class="p">(</span><span class="n">config</span><span class="p">)</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">vcf</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;mutect2/call/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">),</span>
        <span class="n">vcf_index</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;mutect2/call/</span><span class="si">{sample}</span><span class="s2">.vcf.gz.tbi&quot;</span><span class="p">),</span>
        <span class="n">f1r2</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;mutect2/f1r2/</span><span class="si">{sample}</span><span class="s2">.tar.gz&quot;</span><span class="p">),</span>
        <span class="c1">#bam=temp(&quot;mutect2/bam/{sample}.bam&quot;)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Calling variants on </span><span class="si">{wildcards.sample}</span><span class="s2"> with GATK Mutect2&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">20</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">60</span> <span class="o">*</span> <span class="mi">15</span><span class="p">,</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">8192</span><span class="p">,</span> <span class="mi">20480</span><span class="p">),</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">output</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;--create-output-variant-index &quot;</span>
            <span class="s2">&quot;--max-reads-per-alignment-start 0 &quot;</span>
            <span class="s2">&quot;--annotation AlleleFraction &quot;</span>
            <span class="s2">&quot;--annotation AS_QualByDepth &quot;</span>
            <span class="s2">&quot;--annotation BaseQuality &quot;</span>
            <span class="s2">&quot;--annotation Coverage &quot;</span>
            <span class="s2">&quot;--annotation MappingQuality &quot;</span>
            <span class="s2">&quot;--annotation PossibleDeNovo &quot;</span>
            <span class="s2">&quot;--annotation QualByDepth &quot;</span>
            <span class="s2">&quot;--annotation ReferenceBases &quot;</span>
            <span class="s2">&quot;--annotation StrandOddsRatio &quot;</span>
            <span class="s2">&quot;--disable-read-filter MateOnSameContigOrNoMappedMateReadFilter &quot;</span>
            <span class="s2">&quot;--tumor-sample </span><span class="si">{}</span><span class="s2">_tumor &quot;</span>
            <span class="s2">&quot;--normal-sample </span><span class="si">{}</span><span class="s2">_normal &quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span>
                <span class="n">wildcards</span><span class="o">.</span><span class="n">sample</span><span class="p">,</span>
                <span class="n">wildcards</span><span class="o">.</span><span class="n">sample</span>
            <span class="p">)</span>
        <span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/gatk/mutect2/call/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gatk/mutect&quot;</span>


<span class="c1">################################</span>
<span class="c1">### Building Panel of Normal ###</span>
<span class="c1">################################</span>

<span class="n">rule</span> <span class="n">gatk_crate_stomatic_pon</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">ref</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">ref_index</span><span class="o">=</span><span class="n">get_fai</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">ref_dict</span><span class="o">=</span><span class="n">get_dict</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">bams</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;sample_list&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">bams_index</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span><span class="p">),</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;sample_list&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">gvcfs</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;mutect2/pon_call/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;sample_list&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">gvcfs_tbi</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="n">get_tbi</span><span class="p">(</span><span class="s2">&quot;mutect2/pon_call/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">),</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;sample_list&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">intervals</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bed&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">gvcf</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;gatk/pon/PoN.g.vcf.gz&quot;</span><span class="p">),</span>
        <span class="n">gatk_tmp</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="n">directory</span><span class="p">(</span><span class="s2">&quot;gatk/temp/PoN/&quot;</span><span class="p">))</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Building PoN over Mutect2 normal germline callings&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="p">(</span><span class="mi">1024</span> <span class="o">*</span> <span class="mi">5</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/gatk/pon/pon.gvcf.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;gatk_crate_stomatic_pon_extra&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gatk/createsomaticpanelofnormals&quot;</span>


<span class="n">rule</span> <span class="n">gatk_genomics_db_import</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">ref</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">ref_index</span><span class="o">=</span><span class="n">get_fai</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">ref_dict</span><span class="o">=</span><span class="n">get_dict</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">bams</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;sample_list&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">bams_index</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span><span class="p">),</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;sample_list&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">gvcfs</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;mutect2/pon_call/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;sample_list&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">gvcfs_tbi</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="n">get_tbi</span><span class="p">(</span><span class="s2">&quot;mutect2/pon_call/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">),</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;sample_list&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">intervals</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bed&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">genomicsdb</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;gatk/genomicsdb/pon_db&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Building PoN database&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">5120</span><span class="p">,</span> <span class="mi">15360</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">35</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span>
        <span class="n">db_action</span><span class="o">=</span><span class="s2">&quot;create&quot;</span><span class="p">,</span>
        <span class="n">intervals</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">intput</span><span class="p">:</span> <span class="nb">input</span><span class="o">.</span><span class="n">intervals</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/gatk/genomicsdbimport/pon.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gatk/genomicsdbimport&quot;</span>


<span class="n">rule</span> <span class="n">gatk_mutect2_germline_normal</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">fasta</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">fasta_index</span><span class="o">=</span><span class="n">get_fai</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">fasta_dict</span><span class="o">=</span><span class="n">get_dict</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="nb">map</span><span class="o">=</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span><span class="p">,</span>
        <span class="n">map_index</span><span class="o">=</span><span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;sambamba/markdup</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span><span class="p">),</span>
        <span class="n">germline</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;known&quot;</span><span class="p">],</span>
        <span class="n">germline_tbi</span><span class="o">=</span><span class="n">get_tbi</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;known&quot;</span><span class="p">]),</span>
        <span class="n">intervals</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bed&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">vcf</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;mutect2/pon_call/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Calling germline variants on </span><span class="si">{wildcards.sample}</span><span class="s2"> with GATK Mutect2,&quot;</span>
        <span class="s2">&quot;on normal sample only: it will be used for PoN&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">4</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">300</span><span class="p">,</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">8192</span><span class="p">,</span> <span class="mi">20480</span><span class="p">),</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="p">(</span>
            <span class="s2">&quot;--max-reads-per-alignment-start 0 &quot;</span>
            <span class="s2">&quot;--normal Mutect2_</span><span class="si">{sample}</span><span class="s2">_normal &quot;</span>
            <span class="s2">&quot;--disable-read-filter MateOnSameContigOrNoMappedMateReadFilter &quot;</span>
        <span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/gatk/mutect2/pon_call/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gatk/mutect&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely, as long as the dependencies between the rules remain as listed here.
For additional parameters in each individual wrapper, please refer to their corresponding documentation (see links below).</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</section>
<section id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this heading"></a></h2>
<p>The following individual wrappers are used in this meta-wrapper:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/gatk/filtermutectcalls.html#bio-gatk-filtermutectcalls"><span class="std std-ref">GATK FILTERMUTECTCALLS</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/gatk/learnreadorientationmodel.html#bio-gatk-learnreadorientationmodel"><span class="std std-ref">GATK LEARNREADORIENTATIONMODEL</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/gatk/calculatecontamination.html#bio-gatk-calculatecontamination"><span class="std std-ref">GATK CALCULATECONTAMINATION</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/gatk/getpileupsummaries.html#bio-gatk-getpileupsummaries"><span class="std std-ref">GATK GETPILEUPSUMMARIES</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/gatk/mutect.html#bio-gatk-mutect"><span class="std std-ref">GATK MUTECT2</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/bcftools/norm.html#bio-bcftools-norm"><span class="std std-ref">BCFTOOLS NORM</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/gatk/createsomaticpanelofnormals.html#bio-gatk-createsomaticpanelofnormals"><span class="std std-ref">GATK CREATESOMATICPANELOFNORMALS</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/gatk/genomicsdbimport.html#bio-gatk-genomicsdbimport"><span class="std std-ref">GATK GENOMICSDBIMPORT</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this heading"></a></h2>
<p>Bam are expected to be mate-fixed (see bwa_fixmate meta-wrapper).</p>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="mutect2_germline.html" class="btn btn-neutral float-left" title="MUTECT2_GERMLINE" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="salmon.html" class="btn btn-neutral float-right" title="SALMON" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>