<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>TXIMPORT_DESEQ2 &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="VARSCAN2_CALLING" href="varscan2_calling.html" />
    <link rel="prev" title="STAR_MAPPING" href="star_mapping.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bcftools_annotate.html">BCFTOOLS ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bioinfokit.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="bismark.html">BISKMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="bismark_multisample.html">BISKMARK_MULTISAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bowtie2.html">BOWTIE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_fixmate.html">BWA_FIXMATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_mapping.html">BWA_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="clusterprofiler.html">CLUSTERP PROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="compress_index_vcf.html">COMPRESS_INDEX_VCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="coverage_variant_region.html">COVERAGE_VARIANT_REGION</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_pe.html">DADA2-PE</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_se.html">DADA2-SE</a></li>
<li class="toctree-l2"><a class="reference internal" href="deseq2_post_process.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="download_references.html">DOWNLOAD REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_cytoscan.html">EACON CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_oncoscan.html">EACON ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_post_process.html">EACON POST PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastq_screen_indexer.html">FASTQ_SCREEN_INDEXER</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk_bqsr.html">GATK_BQSR</a></li>
<li class="toctree-l2"><a class="reference internal" href="index_datasets.html">INDEX_DATASETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="maftools.html">MAFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller.html">META_CALLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller_germline.html">META_CALLER_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="msi_sensor_pro.html">MSISENSOR_PRO</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_germline.html">MUTECT2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_somatic.html">MUTECT2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_annotate.html">SNPEFF_ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="somatic_tmb.html">SOMATIC_TMB</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_arriba.html">STAR-ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_mapping.html">STAR_MAPPING</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">TXIMPORT_DESEQ2</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_calling.html">VARSCAN2_CALLING</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_germline.html">VARSCAN2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_somatic.html">VARSCAN2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="vcf_post_process.html">VCF_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="wisecondorx.html">WISECONDORX</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../meta-wrappers.html">Meta-Wrappers</a> &raquo;</li>
      <li>TXIMPORT_DESEQ2</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/meta-wrappers/tximport_deseq2.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="tximport-deseq2">
<span id="id1"></span><h1>TXIMPORT_DESEQ2<a class="headerlink" href="#tximport-deseq2" title="Permalink to this heading"></a></h1>
<p>Import counts with tximport and run DESeq2</p>
<section id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this heading"></a></h2>
<p>This meta-wrapper can be used by integrating the following into your workflow:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">default_tximport_deseq2_config</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;gtf&quot;</span><span class="p">:</span> <span class="s2">&quot;annotation.gtf&quot;</span><span class="p">,</span>
    <span class="s2">&quot;design&quot;</span><span class="p">:</span> <span class="s2">&quot;design.tsv&quot;</span><span class="p">,</span>
    <span class="s2">&quot;output_prefixes&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;DGE&quot;</span><span class="p">],</span>
    <span class="s2">&quot;comparison_levels&quot;</span><span class="p">:</span> <span class="p">[[</span><span class="s2">&quot;Cond&quot;</span><span class="p">,</span> <span class="s2">&quot;A&quot;</span><span class="p">,</span> <span class="s2">&quot;B&quot;</span><span class="p">]],</span>
    <span class="s2">&quot;samples_per_prefixes&quot;</span><span class="p">:</span> <span class="p">{</span><span class="s2">&quot;DGE&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;S1&quot;</span><span class="p">,</span> <span class="s2">&quot;S2&quot;</span><span class="p">,</span> <span class="s2">&quot;S3&quot;</span><span class="p">,</span> <span class="s2">&quot;S4&quot;</span><span class="p">]}</span>
<span class="p">}</span>

<span class="k">try</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">config</span> <span class="o">==</span> <span class="nb">dict</span><span class="p">():</span>
        <span class="n">config</span> <span class="o">=</span> <span class="n">default_tximport_deseq2_config</span>
<span class="k">except</span> <span class="ne">NameError</span><span class="p">:</span>
    <span class="n">config</span> <span class="o">=</span> <span class="n">default_tximport_deseq2_config</span>

<span class="c1"># Requires a list of constrasts as defined in DESeq2:</span>
<span class="c1"># [factor, ref, test]</span>
<span class="c1"># This list should be named: comparison_levels</span>

<span class="c1"># Requires a list of comparison names, in the same order as the previous levels</span>
<span class="c1"># It should be named: output_prefixes</span>

<span class="n">contrasts</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span>
    <span class="n">config</span><span class="p">[</span><span class="s2">&quot;output_prefixes&quot;</span><span class="p">],</span>
    <span class="n">config</span><span class="p">[</span><span class="s2">&quot;comparison_levels&quot;</span><span class="p">]</span>
<span class="p">))</span>

<span class="n">batch_effect</span> <span class="o">=</span> <span class="nb">any</span><span class="p">(</span>
    <span class="n">level</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;BatchEffect&quot;</span> <span class="k">for</span> <span class="n">level</span> <span class="ow">in</span> <span class="n">config</span><span class="p">[</span><span class="s2">&quot;comparison_levels&quot;</span><span class="p">]</span>
<span class="p">)</span>


<span class="c1">###############</span>
<span class="c1">### Reports ###</span>
<span class="c1">###############</span>

<span class="n">normalized_counts_rst</span> <span class="o">=</span> <span class="p">(</span>
    <span class="s2">&quot;reports/normalized_counts.rst&quot;</span>
    <span class="k">if</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="s2">&quot;reports/normalized_counts.rst&quot;</span><span class="p">)</span>
    <span class="k">else</span> <span class="kc">None</span>
<span class="p">)</span>

<span class="n">deseq2_rst</span> <span class="o">=</span> <span class="p">(</span>
    <span class="s2">&quot;reports/deseq2_tsv.rst&quot;</span>
    <span class="k">if</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="s2">&quot;reports/deseq2_tsv.rst&quot;</span><span class="p">)</span>
    <span class="k">else</span> <span class="kc">None</span>
<span class="p">)</span>


<span class="c1">###################</span>
<span class="c1">### Readability ###</span>
<span class="c1">###################</span>


<span class="n">rule</span> <span class="n">rbt_csv_report</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/</span><span class="si">{comparison}</span><span class="s2">.</span><span class="si">{content}</span><span class="s2">.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">directory</span><span class="p">(</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/</span><span class="si">{comparison}</span><span class="s2">.</span><span class="si">{content}</span><span class="s2">/&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Making DESeq2 results readable and searchable&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">5</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/rbt/csv-report/</span><span class="si">{comparison}</span><span class="s2">.</span><span class="si">{content}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="s1">&#39;--separator &quot;</span><span class="se">\\</span><span class="s1">t&quot; &#39;</span>
        <span class="s1">&#39;--sort-column stat_change &#39;</span>
        <span class="s1">&#39;--sort-order ascending &#39;</span>
        <span class="s1">&#39;--rows-per-page 50&#39;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/rbt/csvreport&quot;</span>


<span class="n">rule</span> <span class="n">deseq2_readable</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">tsv</span> <span class="o">=</span> <span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/wald.</span><span class="si">{comparison}</span><span class="s2">.tsv&quot;</span><span class="p">,</span>
        <span class="n">gene2gene</span> <span class="o">=</span> <span class="s2">&quot;tximport/gene2gene.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">fc_sig</span> <span class="o">=</span> <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/</span><span class="si">{comparison}</span><span class="s2">.DE_genes_with_FC.tsv&quot;</span><span class="p">),</span>
        <span class="n">padj_sig</span> <span class="o">=</span> <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/</span><span class="si">{comparison}</span><span class="s2">.DE_genes_with_padj.tsv&quot;</span><span class="p">),</span>
        <span class="n">complete</span> <span class="o">=</span> <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/</span><span class="si">{comparison}</span><span class="s2">.complete_results.tsv&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Annotating output results for </span><span class="si">{wildcards.comparison}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span> <span class="o">=</span> <span class="p">(</span>
            <span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span> <span class="mi">10240</span><span class="p">)</span>
        <span class="p">),</span>
        <span class="n">time_min</span> <span class="o">=</span> <span class="p">(</span>
            <span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">20</span><span class="p">,</span> <span class="mi">200</span><span class="p">)</span>
        <span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/deseq2/readable/</span><span class="si">{comparison}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/pandas/deseq2_to_gseaapp&quot;</span>


<span class="c1">##############</span>
<span class="c1">### DESeq2 ###</span>
<span class="c1">##############</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule performs the size factor and dispersions estimations as well as the</span>
<span class="sd">wald test.</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">deseq2</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">dds</span><span class="o">=</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/dds.</span><span class="si">{comparison}</span><span class="s2">.RDS&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">rds</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/wald.</span><span class="si">{comparison}</span><span class="s2">.RDS&quot;</span><span class="p">),</span>
        <span class="n">deseq2_tsv</span><span class="o">=</span><span class="n">report</span><span class="p">(</span>
            <span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/wald.</span><span class="si">{comparison}</span><span class="s2">.tsv&quot;</span><span class="p">,</span>
            <span class="n">caption</span><span class="o">=</span><span class="n">deseq2_rst</span><span class="p">,</span>
            <span class="n">category</span><span class="o">=</span><span class="s2">&quot;DESeq2 results&quot;</span>
        <span class="p">),</span>
        <span class="n">normalized_counts</span><span class="o">=</span><span class="n">report</span><span class="p">(</span>
            <span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/dst.</span><span class="si">{comparison}</span><span class="s2">.tsv&quot;</span><span class="p">,</span>
            <span class="n">caption</span><span class="o">=</span><span class="n">normalized_counts_rst</span> <span class="p">,</span>
            <span class="n">category</span><span class="o">=</span><span class="s2">&quot;Normalized counts&quot;</span>
        <span class="p">),</span>
        <span class="n">dst</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/dst.</span><span class="si">{comparison}</span><span class="s2">.RDS&quot;</span><span class="p">),</span>
        <span class="n">intermediar_values</span><span class="o">=</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/mcols.</span><span class="si">{comparison}</span><span class="s2">.tsv&quot;</span><span class="p">,</span>
        <span class="n">assays_mu</span><span class="o">=</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/assays.mu.</span><span class="si">{comparison}</span><span class="s2">.tsv&quot;</span><span class="p">,</span>
        <span class="n">filter_theta</span><span class="o">=</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/filter.theta.</span><span class="si">{comparison}</span><span class="s2">.tsv&quot;</span><span class="p">,</span>
        <span class="n">metadata</span><span class="o">=</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/metadata.</span><span class="si">{comparison}</span><span class="s2">.tsv&quot;</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Running DESeq2 analysis for </span><span class="si">{wildcards.comparison}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">2048</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">60</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">contrast</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">:</span> <span class="n">contrasts</span><span class="p">[</span><span class="n">wildcards</span><span class="o">.</span><span class="n">comparison</span><span class="p">],</span>
        <span class="c1">#name=lambda wildcards: &quot;_&quot;.join([</span>
        <span class="c1">#   contrasts[wildcards.comparison][0],</span>
        <span class="c1">#   contrasts[wildcards.comparison][1],</span>
        <span class="c1">#   &quot;vs&quot;,</span>
        <span class="c1">#   contrasts[wildcards.comparison][2]</span>
        <span class="c1">#])</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/deseq2/deseq/</span><span class="si">{comparison}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/deseq2/DESeq&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule formats counts for DESeq2. The design matrix and its corresponding</span>
<span class="sd">formula are included.</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">deseq2_dataset_from_tximport</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">tximport</span><span class="o">=</span><span class="s2">&quot;tximport/txi.</span><span class="si">{comparison}</span><span class="s2">.RDS&quot;</span><span class="p">,</span>
        <span class="n">coldata</span><span class="o">=</span><span class="s2">&quot;deseq2/designs/</span><span class="si">{comparison}</span><span class="s2">.tsv&quot;</span><span class="p">,</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">dds</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;deseq2/</span><span class="si">{comparison}</span><span class="s2">/dds.</span><span class="si">{comparison}</span><span class="s2">.RDS&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Formatting </span><span class="si">{wildcards.comparison}</span><span class="s2"> counts for DESeq2&quot;</span><span class="p">,</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">3072</span><span class="p">,</span> <span class="mi">20480</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">design</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">:</span> <span class="p">(</span>
            <span class="sa">f</span><span class="s2">&quot;~</span><span class="si">{</span><span class="n">contrasts</span><span class="p">[</span><span class="n">wildcards</span><span class="o">.</span><span class="n">comparison</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2">&quot;</span>
            <span class="k">if</span> <span class="p">(</span><span class="n">batch_effect</span> <span class="ow">is</span> <span class="kc">False</span><span class="p">)</span> <span class="ow">or</span> <span class="n">wildcards</span><span class="o">.</span><span class="n">comparison</span> <span class="o">==</span> <span class="s2">&quot;BatchEffect&quot;</span>
            <span class="k">else</span> <span class="sa">f</span><span class="s2">&quot;~BatchEffect+</span><span class="si">{</span><span class="n">contrasts</span><span class="p">[</span><span class="n">wildcards</span><span class="o">.</span><span class="n">comparison</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2">&quot;</span>
        <span class="p">),</span>
        <span class="n">levels</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">:</span> <span class="n">contrasts</span><span class="p">[</span><span class="n">wildcards</span><span class="o">.</span><span class="n">comparison</span><span class="p">][</span><span class="mi">1</span><span class="p">:],</span>
        <span class="n">factor</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">:</span> <span class="n">contrasts</span><span class="p">[</span><span class="n">wildcards</span><span class="o">.</span><span class="n">comparison</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span>
        <span class="n">ref_level</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">:</span> <span class="n">contrasts</span><span class="p">[</span><span class="n">wildcards</span><span class="o">.</span><span class="n">comparison</span><span class="p">][</span><span class="o">-</span><span class="mi">1</span><span class="p">],</span>
        <span class="n">remove_zeros</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
        <span class="n">count_filter</span><span class="o">=</span><span class="mi">5</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/deseq2/deseq2_dataset_from_tximport/</span><span class="si">{comparison}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/deseq2/DESeqDataSetFromTximport&quot;</span>


<span class="c1">##########################</span>
<span class="c1">### Counts aggregation ###</span>
<span class="c1">##########################</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule imports counts from tables to R data object. Its memory requirements</span>
<span class="sd">are linked to the number of samples</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">tximport</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">quant</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">:</span> <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;salmon/pseudo_mapping/</span><span class="si">{sample}</span><span class="s2">/quant.sf&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;samples_per_prefixes&quot;</span><span class="p">][</span><span class="n">wildcards</span><span class="o">.</span><span class="n">comparison</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">tx_to_gene</span><span class="o">=</span><span class="s2">&quot;tximport/tx2gene.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">txi</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;tximport/txi.</span><span class="si">{comparison}</span><span class="s2">.RDS&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Importing counts in DESeq2 for </span><span class="si">{wildcards.comparison}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="nb">input</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="nb">input</span><span class="o">.</span><span class="n">quant</span><span class="p">)</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span>
            <span class="s2">&quot;tximport_extra&quot;</span><span class="p">,</span>
            <span class="s2">&quot;type=&#39;salmon&#39;, ignoreTxVersion=TRUE, ignoreAfterBar=TRUE&quot;</span>
        <span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/tximport/</span><span class="si">{comparison}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/tximport&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule build the conversion table from transcript to genes and their names.</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">tx_to_gene_deseq2</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">gtf</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;gtf&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">tx2gene_small</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;tximport/tx2gene.tsv&quot;</span><span class="p">),</span>
        <span class="n">tx2gene_large</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;tximport/transcripts2genes.tsv&quot;</span><span class="p">),</span>
        <span class="n">gene2gene_large</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;tximport/gene2gene.tsv&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Building transcripts/genes conversion table&quot;</span>
    <span class="n">cache</span><span class="p">:</span> <span class="kc">True</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">2048</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">gencode</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
        <span class="n">header</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
        <span class="n">positions</span> <span class="o">=</span> <span class="kc">True</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/tximport/tx2gene.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/gtf/tx2gene&quot;</span>

<span class="c1">######################</span>
<span class="c1">### Design parsing ###</span>
<span class="c1">######################</span>


<span class="n">rule</span> <span class="n">split_design</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">design</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;design&quot;</span><span class="p">][</span><span class="s2">&quot;path&quot;</span><span class="p">],</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;deseq2/designs/</span><span class="si">{comparison}</span><span class="s2">.tsv&quot;</span><span class="p">,</span>
            <span class="n">comparison</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;output_prefixes&quot;</span><span class="p">]</span>
        <span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Expanding design in order to make results more readeble&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">10</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/deseq2/split_design.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">columns_to_aggregate</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;design&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;aggregate_col&quot;</span><span class="p">),</span>
        <span class="n">columns_to_remove</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;design&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;remove_col&quot;</span><span class="p">)</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/BiGR/split_design&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely, as long as the dependencies between the rules remain as listed here.
For additional parameters in each individual wrapper, please refer to their corresponding documentation (see links below).</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</section>
<section id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this heading"></a></h2>
<p>The following individual wrappers are used in this meta-wrapper:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/rbt/csvreport.html#bio-rbt-csvreport"><span class="std std-ref">RBT CSV-REPORT</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/gtf/tx2gene.html#bio-gtf-tx2gene"><span class="std std-ref">TX2GENE</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/BiGR/split_design.html#bio-bigr-split-design"><span class="std std-ref">SPLIT_DESIGN</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/tximport.html#bio-tximport"><span class="std std-ref">TXIMPORT</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/deseq2/DESeqDataSetFromTximport.html#bio-deseq2-deseqdatasetfromtximport"><span class="std std-ref">DESEQ DATASET FROM TXIMPORT</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/deseq2/DESeq.html#bio-deseq2-deseq"><span class="std std-ref">DESEQ</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/pandas/deseq2_to_gseaapp.html#bio-pandas-deseq2-to-gseaapp"><span class="std std-ref">DESEQ2 TO GSEAAPP</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this heading"></a></h2>
<p>The R statistical formula must refer to columns in design file.</p>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="star_mapping.html" class="btn btn-neutral float-left" title="STAR_MAPPING" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="varscan2_calling.html" class="btn btn-neutral float-right" title="VARSCAN2_CALLING" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>