

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>VARSCAN2_GERMLINE &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
  <link rel="stylesheet" href="../_static/theme.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="VARSCAN2_SOMATIC" href="varscan2_somatic.html" />
    <link rel="prev" title="VARSCAN2_CALLING" href="varscan2_calling.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          

          
          </a>

          
            
            
              <div class="version">
                heads/Unofficial
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Wrappers, Meta-Wrappers and Pipelines</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bioinfokit.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_fixmate.html">BWA_FIXMATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_mapping.html">BWA_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="clean_ngs.html">CLEAN_NGS</a></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_profiler.html">CLUSTER_PROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="compress_index_vcf.html">COMPRESS_INDEX_VCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_pe.html">DADA2-PE</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_se.html">DADA2-SE</a></li>
<li class="toctree-l2"><a class="reference internal" href="deseq2_post_process.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="download_references.html">DOWNLOAD REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_cytoscan.html">EACON CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_oncoscan.html">EACON CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_post_process.html">EACON POST PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastq_screen_indexer.html">FASTQ_SCREEN_INDEXER</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk_bqsr.html">GATK_BQSR</a></li>
<li class="toctree-l2"><a class="reference internal" href="index_datasets.html">INDEX_DATASETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller.html">META_CALLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller_germline.html">META_CALLER_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_germline.html">MUTECT2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_annotate.html">SNPEFF_ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_arriba.html">STAR-ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="tximport_deseq2.html">TXIMPORT_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_calling.html">VARSCAN2_CALLING</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">VARSCAN2_GERMLINE</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_somatic.html">VARSCAN2_SOMATIC</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../meta-wrappers.html">Meta-Wrappers</a> &raquo;</li>
        
      <li>VARSCAN2_GERMLINE</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/meta-wrappers/varscan2_germline.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="varscan2-germline">
<span id="id1"></span><h1>VARSCAN2_GERMLINE<a class="headerlink" href="#varscan2-germline" title="Permalink to this headline">¶</a></h1>
<p>Call variants on BAM files with Varscan2, considering a single bam file.</p>
<div class="section" id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this headline">¶</a></h2>
<p>This meta-wrapper can be used by integrating the following into your workflow:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">default_config_varscan2_germline</span> <span class="o">=</span> <span class="p">{</span>
    <span class="c1"># Your genome sequence</span>
    <span class="s2">&quot;genome&quot;</span><span class="p">:</span> <span class="s2">&quot;reference/genome.fasta&quot;</span><span class="p">,</span>
    <span class="c1"># Path to a BED containing the kit&#39;s catpured regions</span>
    <span class="s2">&quot;bed&quot;</span><span class="p">:</span> <span class="s2">&quot;reference/regions.bed&quot;</span>
<span class="p">}</span>

<span class="k">try</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">config</span> <span class="o">==</span> <span class="nb">dict</span><span class="p">():</span>
        <span class="n">config</span> <span class="o">=</span> <span class="n">default_config_varscan2_germline</span>
<span class="k">except</span> <span class="ne">NameError</span><span class="p">:</span>
    <span class="n">config</span> <span class="o">=</span> <span class="n">default_config_varscan2_germline</span>

<span class="k">def</span> <span class="nf">get_fai</span><span class="p">(</span><span class="n">genome_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
    <span class="k">return</span> <span class="n">genome_path</span> <span class="o">+</span> <span class="s2">&quot;.fai&quot;</span>

<span class="k">def</span> <span class="nf">get_bai</span><span class="p">(</span><span class="n">bam_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
    <span class="k">return</span> <span class="n">bam_path</span> <span class="o">+</span> <span class="s2">&quot;.bai&quot;</span>

<span class="c1"># # This module includes VCF compression and indexation</span>
<span class="c1"># module compress_index_vcf:</span>
<span class="c1">#     snakefile: &quot;../../compress_index_vcf/test/Snakefile&quot;</span>
<span class="c1">#     config: config</span>
<span class="c1">#</span>
<span class="c1"># use rule * from compress_index_vcf as compress_index_vcf_*</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule concats snp and indel callings from Varscan2 in order to produce a</span>
<span class="sd">full VCF file with both kind of variations.</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">bcftools_concat</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">calls</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;varscan2/calling/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{kind}</span><span class="s2">.vcf.gz&quot;</span><span class="p">,</span>
            <span class="n">kind</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;snp&quot;</span><span class="p">,</span> <span class="s2">&quot;indel&quot;</span><span class="p">],</span>
            <span class="n">allow_missing</span><span class="o">=</span><span class="kc">True</span>
        <span class="p">),</span>
        <span class="n">calls_index</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;varscan2/calling/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{kind}</span><span class="s2">.vcf.gz.tbi&quot;</span><span class="p">,</span>
            <span class="n">kind</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;snp&quot;</span><span class="p">,</span> <span class="s2">&quot;indel&quot;</span><span class="p">],</span>
            <span class="n">allow_missing</span><span class="o">=</span><span class="kc">True</span>
        <span class="p">),</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;vascan2/concat/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">)</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">2</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">1025</span><span class="p">,</span> <span class="mi">4096</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="s2">&quot;--output-type z --remove-duplicates --allow-overlaps&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/varscan/pileup2indel/concat/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;/bio/bcftools/concat&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule calls small indel variants with Varscan2. Results are set to unzipped</span>
<span class="sd">VCF because BCFTools merging will be faster that way. Thus, unzipped VCF are</span>
<span class="sd">temporary files.</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">varscan2_indel_calling</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">pileup</span><span class="o">=</span><span class="s2">&quot;samtools/mpileup/</span><span class="si">{sample}</span><span class="s2">.mpileup.gz&quot;</span><span class="p">,</span>
        <span class="n">sample_list</span><span class="o">=</span><span class="s2">&quot;varscan2/mpileup2cns/</span><span class="si">{sample}</span><span class="s2">.sample.list&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;varscan2/calling/</span><span class="si">{sample}</span><span class="s2">.indel.vcf&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Calling Indels on </span><span class="si">{wildcards.sample}</span><span class="s2"> with Varscan2&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">2</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">4096</span><span class="p">,</span> <span class="mi">15360</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;--p-value 0.05 --variants&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/varscan/pileup2indel/call/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;/bio/varscan/mpileup2indel&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule calls snp variants with Varscan2. Results are set to unzipped</span>
<span class="sd">VCF because BCFTools merging will be faster that way. Thus, unzipped VCF are</span>
<span class="sd">temporary files.</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">varscan2_snp_calling</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">pileup</span><span class="o">=</span><span class="s2">&quot;samtools/mpileup/</span><span class="si">{sample}</span><span class="s2">.mpileup.gz&quot;</span><span class="p">,</span>
        <span class="n">sample_list</span><span class="o">=</span><span class="s2">&quot;varscan2/mpileup2cns/</span><span class="si">{sample}</span><span class="s2">.sample.list&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;varscan2/calling/</span><span class="si">{sample}</span><span class="s2">.snp.vcf&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Calling SNP on </span><span class="si">{wildcards.sample}</span><span class="s2"> with Varscan2&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">2</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">4096</span><span class="p">,</span> <span class="mi">15360</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;--p-value 0.05 --variants&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/varscan/pileup2snp/call/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;/bio/varscan/mpileup2snp&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule provides a list of sample names (here, only one) for Varscan2</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">varscan2_sample_list</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;varscan2/mpileup2cns/</span><span class="si">{sample}</span><span class="s2">.sample.list&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Building sample list for Varscan2 mpileup2cns&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="mi">128</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="mi">2</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="s1">&#39;&quot;</span><span class="si">{sample}</span><span class="s1">&quot;&#39;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/varscan2/mpileup2cns/</span><span class="si">{sample}</span><span class="s2">.list.log&quot;</span>
    <span class="n">shell</span><span class="p">:</span>
        <span class="s2">&quot;echo </span><span class="si">{params}</span><span class="s2"> &gt; </span><span class="si">{output}</span><span class="s2"> 2&gt; </span><span class="si">{log}</span><span class="s2">&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule runs samtools mpileup to list each single difference between mapped</span>
<span class="sd">reads and reference genome</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">samtools_mpilup</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">bam</span><span class="o">=</span><span class="s2">&quot;gatk/recal_bam/</span><span class="si">{sample}</span><span class="s2">.bam&quot;</span><span class="p">,</span>
        <span class="n">bam_index</span><span class="o">=</span><span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;gatk/recal_bam/</span><span class="si">{sample}</span><span class="s2">.bam&quot;</span><span class="p">),</span>
        <span class="n">reference_genome</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">reference_genome_idx</span><span class="o">=</span><span class="n">get_fai</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">bed</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bed&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;samtools/mpileup/</span><span class="si">{sample}</span><span class="s2">.mpileup.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Building mpilup on </span><span class="si">{wildcards.sample}</span><span class="s2"> with samtools&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">2</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">4096</span><span class="p">,</span> <span class="mi">20480</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">120</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/samtools/mpileup/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;--count-orphans --no-BAQ&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;/bio/samtools/mpileup&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely, as long as the dependencies between the rules remain as listed here.
For additional parameters in each individual wrapper, please refer to their corresponding documentation (see links below).</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</div>
<div class="section" id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this headline">¶</a></h2>
<p>The following individual wrappers are used in this meta-wrapper:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/samtools/mpileup.html#bio-samtools-mpileup"><span class="std std-ref">SAMTOOLS MPILEUP</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/varscan/mpileup2snp.html#bio-varscan-mpileup2snp"><span class="std std-ref">VARSCAN MPILEUP2SNP</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/varscan/mpileup2indel.html#bio-varscan-mpileup2indel"><span class="std std-ref">VARSCAN MPILEUP2INDEL</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/bcftools/concat.html#bio-bcftools-concat"><span class="std std-ref">BCFTOOLS CONCAT</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</div>
<div class="section" id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this headline">¶</a></h2>
<p>Bam are expected to be mate-fixed (see bwa_fixmate meta-wrapper), and recalibrated (see gatk_bqsr meta-wrapper).</p>
</div>
<div class="section" id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="varscan2_somatic.html" class="btn btn-neutral float-right" title="VARSCAN2_SOMATIC" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="varscan2_calling.html" class="btn btn-neutral float-left" title="VARSCAN2_CALLING" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2016, The Snakemake team.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>