<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>VARSCAN2_SOMATIC &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="VCF_POST_PROCESS" href="vcf_post_process.html" />
    <link rel="prev" title="VARSCAN2_GERMLINE" href="varscan2_germline.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bcftools_annotate.html">BCFTOOLS ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bioinfokit.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="bismark.html">BISKMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="bismark_multisample.html">BISKMARK_MULTISAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bowtie2.html">BOWTIE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_fixmate.html">BWA_FIXMATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa_mapping.html">BWA_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="clusterprofiler.html">CLUSTERP PROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="compress_index_vcf.html">COMPRESS_INDEX_VCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="coverage_variant_region.html">COVERAGE_VARIANT_REGION</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_pe.html">DADA2-PE</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2_se.html">DADA2-SE</a></li>
<li class="toctree-l2"><a class="reference internal" href="deseq2_post_process.html">DESEQ2_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="download_references.html">DOWNLOAD REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_cytoscan.html">EACON CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_oncoscan.html">EACON ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon_post_process.html">EACON POST PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastq_screen_indexer.html">FASTQ_SCREEN_INDEXER</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk_bqsr.html">GATK_BQSR</a></li>
<li class="toctree-l2"><a class="reference internal" href="index_datasets.html">INDEX_DATASETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="maftools.html">MAFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller.html">META_CALLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="meta_caller_germline.html">META_CALLER_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="msi_sensor_pro.html">MSISENSOR_PRO</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_germline.html">MUTECT2_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2_somatic.html">MUTECT2_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_annotate.html">SNPEFF_ANNOTATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="somatic_tmb.html">SOMATIC_TMB</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_arriba.html">STAR-ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="star_mapping.html">STAR_MAPPING</a></li>
<li class="toctree-l2"><a class="reference internal" href="tximport_deseq2.html">TXIMPORT_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_calling.html">VARSCAN2_CALLING</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan2_germline.html">VARSCAN2_GERMLINE</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">VARSCAN2_SOMATIC</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="vcf_post_process.html">VCF_POST_PROCESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="wisecondorx.html">WISECONDORX</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../meta-wrappers.html">Meta-Wrappers</a> &raquo;</li>
      <li>VARSCAN2_SOMATIC</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/meta-wrappers/varscan2_somatic.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="varscan2-somatic">
<span id="id1"></span><h1>VARSCAN2_SOMATIC<a class="headerlink" href="#varscan2-somatic" title="Permalink to this heading"></a></h1>
<p>Call variants on BAM files with Varscan2, considering a normal background and a tumor state.</p>
<section id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this heading"></a></h2>
<p>This meta-wrapper can be used by integrating the following into your workflow:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>

<span class="n">worflow_source_dir</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="nb">next</span><span class="p">(</span><span class="nb">iter</span><span class="p">(</span><span class="n">workflow</span><span class="o">.</span><span class="n">get_sources</span><span class="p">())))</span><span class="o">.</span><span class="n">absolute</span><span class="p">()</span><span class="o">.</span><span class="n">parent</span>
<span class="n">common</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">worflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;../../../../bigr_pipelines/common/python&quot;</span><span class="p">)</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">common</span><span class="p">)</span>

<span class="kn">from</span> <span class="nn">file_manager</span> <span class="kn">import</span> <span class="o">*</span>

<span class="n">default_config_varscan2_somatic</span> <span class="o">=</span> <span class="p">{</span>
    <span class="c1"># Your genome sequence</span>
    <span class="s2">&quot;genome&quot;</span><span class="p">:</span> <span class="s2">&quot;reference/genome.fasta&quot;</span><span class="p">,</span>
    <span class="c1"># Path to a BED containing the kit&#39;s catpured regions</span>
    <span class="s2">&quot;bed&quot;</span><span class="p">:</span> <span class="s2">&quot;reference/regions.bed&quot;</span><span class="p">,</span>
    <span class="c1"># Extra parameters</span>
    <span class="s2">&quot;mpileup_extra&quot;</span><span class="p">:</span> <span class="s2">&quot;--count-orphans --no-BAQ&quot;</span><span class="p">,</span>
    <span class="s2">&quot;varscan_somatic_extra&quot;</span><span class="p">:</span> <span class="s2">&quot;--somatic-p-value 0.1 --variants --output-type g&quot;</span><span class="p">,</span>
    <span class="s2">&quot;varscan_bcftools_concat_extra&quot;</span><span class="p">:</span> <span class="s2">&quot;--remove-duplicates --allow-overlaps&quot;</span>
<span class="p">}</span>

<span class="k">try</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">config</span> <span class="o">==</span> <span class="nb">dict</span><span class="p">():</span>
        <span class="n">config</span> <span class="o">=</span> <span class="n">default_config_varscan2_somatic</span>
<span class="k">except</span> <span class="ne">NameError</span><span class="p">:</span>
    <span class="n">config</span> <span class="o">=</span> <span class="n">default_config_varscan2_somatic</span>


<span class="c1">## Required modules :</span>
<span class="c1">## This module includes fasta indexes and fasta dictionnaries</span>
<span class="c1"># module index_datasets:</span>
<span class="c1">#     snakefile: &quot;../../index_datasets/test/Snakefile&quot;</span>
<span class="c1">#     config: config</span>
<span class="c1">#</span>
<span class="c1"># use rule * from index_datasets as index_datasets_*</span>

<span class="c1">## This module includes GATK recalibration</span>
<span class="c1"># module gatk_bqsr:</span>
<span class="c1">#     snakefile: &quot;../../gatk_bqsr/test/Snakefile&quot;</span>
<span class="c1">#     config: config</span>
<span class="c1">#</span>
<span class="c1"># use rule * from gatk_bqsr as gatk_bqsr_*</span>

<span class="c1">## This module includes BWA mapping and Samtools corrections</span>
<span class="c1"># module bwa_fixmate:</span>
<span class="c1">#     snakefile: &quot;../../bwa_fixmate/test/Snakefile&quot;</span>
<span class="c1">#     config: config</span>
<span class="c1">#</span>
<span class="c1"># use rule * from bwa_fixmate as bwa_fixmate_*</span>

<span class="c1">##################################</span>
<span class="c1">### Correct multiallelic sites ###</span>
<span class="c1">##################################</span>

<span class="n">rule</span> <span class="n">split_multiallelic_varscan2</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">call</span><span class="o">=</span><span class="s2">&quot;varscan2/concat/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">,</span>
        <span class="n">idx</span><span class="o">=</span><span class="n">get_tbi</span><span class="p">(</span><span class="s2">&quot;varscan2/concat/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">),</span>
        <span class="n">fasta</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;bcftools/varscan2/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Splitting Mutect2 multiallelic sites </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">5120</span><span class="p">,</span> <span class="mi">15360</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">35</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span> <span class="o">=</span> <span class="s2">&quot;-m -both --check-ref w&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bcftools/norm/mutect2/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bcftools/norm&quot;</span>


<span class="c1">####################################</span>
<span class="c1">### Complete and correct headers ###</span>
<span class="c1">####################################</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule renames samples for further merges. It also adds chromosome</span>
<span class="sd">information and filters on provided regions</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">bcftools_reheader</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">vcf</span><span class="o">=</span><span class="p">(</span>
            <span class="s2">&quot;varscan2/concat/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span>
            <span class="k">if</span> <span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;indel&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span> <span class="ow">is</span> <span class="kc">False</span>
            <span class="k">else</span> <span class="s2">&quot;varscan2/somatic/</span><span class="si">{sample}</span><span class="s2">.snp.vcf&quot;</span>
        <span class="p">),</span>
        <span class="n">vcf_tbi</span><span class="o">=</span><span class="n">get_tbi</span><span class="p">(</span>
            <span class="s2">&quot;varscan2/concat/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span>
            <span class="k">if</span> <span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;indel&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span> <span class="ow">is</span> <span class="kc">False</span>
            <span class="k">else</span> <span class="s2">&quot;varscan2/somatic/</span><span class="si">{sample}</span><span class="s2">.snp.vcf&quot;</span>
        <span class="p">),</span>
        <span class="c1">#samples=&quot;varscan2/mpileup2cns/{sample}.sample.list&quot;,</span>
        <span class="n">fasta</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">fai</span><span class="o">=</span><span class="n">get_fai</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">regions</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bed&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;varscan2/reheaded/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Reheading </span><span class="si">{wildcards.sample}</span><span class="s2"> with fasta index and new sample names&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">2</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">25</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bcftools/reheader/varscan2/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span>
        <span class="n">view_extra</span><span class="o">=</span><span class="s2">&quot;&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bcftools/reheader&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule provides a list of sample names (here, only one) for Varscan2</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">varscan2_sample_list</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;varscan2/mpileup2cns/</span><span class="si">{sample}</span><span class="s2">.sample.list&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Building sample list for Varscan2 mpileup2cns&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="mi">1024</span> <span class="o">*</span> <span class="mi">10</span> <span class="o">*</span> <span class="n">attempt</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="mi">2</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="s1">&#39;&quot;varscan2_</span><span class="si">{sample}</span><span class="s1">_tumor</span><span class="se">\n</span><span class="s1">varscan2_</span><span class="si">{sample}</span><span class="s1">_normal&quot;&#39;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/varscan2/samples/</span><span class="si">{sample}</span><span class="s2">.list.log&quot;</span>
    <span class="n">shell</span><span class="p">:</span>
        <span class="s2">&quot;echo -e </span><span class="si">{params}</span><span class="s2"> &gt; </span><span class="si">{output}</span><span class="s2"> 2&gt; </span><span class="si">{log}</span><span class="s2">&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule concats snp and indel callings from Varscan2 in order to produce a</span>
<span class="sd">full VCF file with both kind of variations.</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">bcftools_concat</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">calls</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;varscan2/somatic/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{content}</span><span class="s2">.vcf.gz&quot;</span><span class="p">,</span>
            <span class="n">content</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;snp&quot;</span><span class="p">,</span> <span class="s2">&quot;indel&quot;</span><span class="p">],</span>
            <span class="n">allow_missing</span><span class="o">=</span><span class="kc">True</span>
        <span class="p">),</span>
        <span class="n">calls_index</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;varscan2/somatic/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{content}</span><span class="s2">.vcf.gz.tbi&quot;</span><span class="p">,</span>
            <span class="n">content</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;snp&quot;</span><span class="p">,</span> <span class="s2">&quot;indel&quot;</span><span class="p">],</span>
            <span class="n">allow_missing</span><span class="o">=</span><span class="kc">True</span>
        <span class="p">)</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;varscan2/concat/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Concatenating varscan calling for </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">2</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">1025</span><span class="p">,</span> <span class="mi">4096</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span>
            <span class="s2">&quot;bcftools_concat_extra&quot;</span><span class="p">,</span>
            <span class="s2">&quot;--remove-duplicates --allow-overlaps&quot;</span>
        <span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bcftools/concat/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/bcftools/concat&quot;</span>

<span class="c1">#######################</span>
<span class="c1">### Variant Calling ###</span>
<span class="c1">#######################</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule performs germline calling with Varscan2</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">varscan2_somatic</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">mpileup</span><span class="o">=</span><span class="s2">&quot;samtools/mpileup/</span><span class="si">{sample}</span><span class="s2">.mpileup.gz&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">snp</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;varscan2/somatic/</span><span class="si">{sample}</span><span class="s2">.snp.vcf&quot;</span><span class="p">),</span>
        <span class="n">indel</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;varscan2/somatic/</span><span class="si">{sample}</span><span class="s2">.indel.vcf&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Calling variants on </span><span class="si">{wildcards.sample}</span><span class="s2"> with Varscan2 mpileup2cns&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">2</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">8192</span><span class="p">,</span> <span class="mi">20480</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span>
            <span class="s2">&quot;varscan_somatic_extra&quot;</span><span class="p">,</span>
            <span class="s2">&quot;--somatic-p-value 0.2 --variants --output-type g&quot;</span>
        <span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/varscan2/somatic/</span><span class="si">{sample}</span><span class="s2">.call.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/varscan/somatic&quot;</span>


<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This rule runs samtools mpileup to list each single difference between mapped</span>
<span class="sd">reads and reference genome</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="n">rule</span> <span class="n">samtools_mpilup</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">bam</span><span class="o">=</span><span class="p">[</span>
            <span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_tumor.bam&quot;</span><span class="p">,</span>
            <span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span>
        <span class="p">],</span>
        <span class="n">bam_index</span><span class="o">=</span><span class="p">[</span>
            <span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_tumor.bam&quot;</span><span class="p">),</span>
            <span class="n">get_bai</span><span class="p">(</span><span class="s2">&quot;sambamba/markdup/</span><span class="si">{sample}</span><span class="s2">_normal.bam&quot;</span><span class="p">)</span>
        <span class="p">],</span>
        <span class="n">reference_genome</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
        <span class="n">reference_genome_idx</span><span class="o">=</span><span class="n">get_fai</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;genome&quot;</span><span class="p">]),</span>
        <span class="n">bed</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;bed&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;samtools/mpileup/</span><span class="si">{sample}</span><span class="s2">.mpileup.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Building mpilup on </span><span class="si">{wildcards.sample}</span><span class="s2"> with samtools (tumor/normal)&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">2</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">4096</span><span class="p">,</span> <span class="mi">20480</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">120</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/samtools/mpileup/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;mpileup_extra&quot;</span><span class="p">,</span> <span class="s2">&quot;--count-orphans --no-BAQ&quot;</span><span class="p">)</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/samtools/mpileup&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely, as long as the dependencies between the rules remain as listed here.
For additional parameters in each individual wrapper, please refer to their corresponding documentation (see links below).</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</section>
<section id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this heading"></a></h2>
<p>The following individual wrappers are used in this meta-wrapper:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/bcftools/concat.html#bio-bcftools-concat"><span class="std std-ref">BCFTOOLS CONCAT</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/varscan/somatic.html#bio-varscan-somatic"><span class="std std-ref">VARSCAN SOMATIC</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/samtools/mpileup.html#bio-samtools-mpileup"><span class="std std-ref">SAMTOOLS MPILEUP</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this heading"></a></h2>
<p>Bam are expected to be mate-fixed (see bwa_fixmate meta-wrapper), and recalibrated (see gatk_bqsr meta-wrapper).</p>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="varscan2_germline.html" class="btn btn-neutral float-left" title="VARSCAN2_GERMLINE" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="vcf_post_process.html" class="btn btn-neutral float-right" title="VCF_POST_PROCESS" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>