<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>BISEQ &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="BISMARK" href="bismark.html" />
    <link rel="prev" title="BIOMART" href="biomaRt/mouse_to_human.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../wrappers.html">Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="BiGR.html">BIGR</a></li>
<li class="toctree-l2"><a class="reference internal" href="Rsamtools.html">RSAMTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="SomaticSignatures.html">SOMATICSIGNATURES</a></li>
<li class="toctree-l2"><a class="reference internal" href="VariantAnnotation.html">VARIANTANNOTATION</a></li>
<li class="toctree-l2"><a class="reference internal" href="adapterremoval.html">ADAPTERREMOVAL</a></li>
<li class="toctree-l2"><a class="reference internal" href="arriba.html">ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="art.html">ART</a></li>
<li class="toctree-l2"><a class="reference internal" href="assembly-stats.html">ASSEMBLY-STATS</a></li>
<li class="toctree-l2"><a class="reference internal" href="awk.html">AWK</a></li>
<li class="toctree-l2"><a class="reference internal" href="bamtools.html">BAMTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="bbtools.html">BBTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="bcftools.html">BCFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="bcl2fastq.html">BCF2FASTQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="bedtools.html">BEDTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="benchmark.html">BENCHMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="biobambam2.html">BIOBAMBAM2</a></li>
<li class="toctree-l2"><a class="reference internal" href="bioinfokit.html">BIOINFOKIT</a></li>
<li class="toctree-l2"><a class="reference internal" href="biomaRt.html">BIOMART</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">BISEQ</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#software-dependencies">Software dependencies</a></li>
<li class="toctree-l3"><a class="reference internal" href="#input-output">Input/Output</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
<li class="toctree-l3"><a class="reference internal" href="#code">Code</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="bismark.html">BISMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="blast.html">BLAST</a></li>
<li class="toctree-l2"><a class="reference internal" href="bowtie2.html">BOWTIE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="busco.html">BUSCO</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa.html">BWA</a></li>
<li class="toctree-l2"><a class="reference internal" href="bwa-mem2.html">BWA-MEM2</a></li>
<li class="toctree-l2"><a class="reference internal" href="cairosvg.html">CAIROSVG</a></li>
<li class="toctree-l2"><a class="reference internal" href="check_md5.html">CHECK MD5</a></li>
<li class="toctree-l2"><a class="reference internal" href="chipseeker.html">CHIPSEEKER</a></li>
<li class="toctree-l2"><a class="reference internal" href="clustalo.html">CLUSTALO</a></li>
<li class="toctree-l2"><a class="reference internal" href="clusterProfiler.html">CLUSTERPROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="compress.html">COMPRESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="concatenate_fastq.html">CONCATENATE FASTQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="consensusclusterplus.html">CONSENSUSCLUSTERPLUS</a></li>
<li class="toctree-l2"><a class="reference internal" href="cp.html">CP (BASH)</a></li>
<li class="toctree-l2"><a class="reference internal" href="cpat.html">CPAT</a></li>
<li class="toctree-l2"><a class="reference internal" href="cutadapt.html">CUTADAPT</a></li>
<li class="toctree-l2"><a class="reference internal" href="dada2.html">DADA2</a></li>
<li class="toctree-l2"><a class="reference internal" href="deeptools.html">DEEPTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="deepvariant.html">DEEPVARIANT</a></li>
<li class="toctree-l2"><a class="reference internal" href="delly.html">DELLY</a></li>
<li class="toctree-l2"><a class="reference internal" href="deseq2.html">DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="diamond.html">DIAMOND</a></li>
<li class="toctree-l2"><a class="reference internal" href="eacon.html">EACON</a></li>
<li class="toctree-l2"><a class="reference internal" href="enhancedVolcano.html">ENHANCEDVOLCANO</a></li>
<li class="toctree-l2"><a class="reference internal" href="ensembl.html">ENSEMBL</a></li>
<li class="toctree-l2"><a class="reference internal" href="epic.html">EPIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="facets.html">FACETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastp.html">FASTP</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastq_screen.html">FASTQ_SCREEN</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastqc.html">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="fgbio.html">FGBIO</a></li>
<li class="toctree-l2"><a class="reference internal" href="fgsea.html">FGSEA</a></li>
<li class="toctree-l2"><a class="reference internal" href="filtlong.html">FILTLONG</a></li>
<li class="toctree-l2"><a class="reference internal" href="freebayes.html">FREEBAYES</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk.html">GATK</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk3.html">GATK3</a></li>
<li class="toctree-l2"><a class="reference internal" href="gdc-api.html">GDC-API</a></li>
<li class="toctree-l2"><a class="reference internal" href="gdc-client.html">GDC-CLIENT</a></li>
<li class="toctree-l2"><a class="reference internal" href="gencode.html">GENCODE</a></li>
<li class="toctree-l2"><a class="reference internal" href="genomepy.html">GENOMEPY</a></li>
<li class="toctree-l2"><a class="reference internal" href="gridss.html">GRIDSS</a></li>
<li class="toctree-l2"><a class="reference internal" href="gsva.html">GSVA</a></li>
<li class="toctree-l2"><a class="reference internal" href="gtf.html">GTF</a></li>
<li class="toctree-l2"><a class="reference internal" href="hap.py.html">HAP.PY</a></li>
<li class="toctree-l2"><a class="reference internal" href="hisat2.html">HISAT2</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmmer.html">HMMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="homer.html">HOMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="iRODS.html">IRODS</a></li>
<li class="toctree-l2"><a class="reference internal" href="igv-reports.html">IGV-REPORTS</a></li>
<li class="toctree-l2"><a class="reference internal" href="immunedeconv.html">IMMUNEDECONV</a></li>
<li class="toctree-l2"><a class="reference internal" href="infernal.html">INFERNAL</a></li>
<li class="toctree-l2"><a class="reference internal" href="isoformSwitchAnalyseR.html">ISOFORMSWITCHANALYSER</a></li>
<li class="toctree-l2"><a class="reference internal" href="jannovar.html">JANNOVAR</a></li>
<li class="toctree-l2"><a class="reference internal" href="kallisto.html">KALLISTO</a></li>
<li class="toctree-l2"><a class="reference internal" href="last.html">LAST</a></li>
<li class="toctree-l2"><a class="reference internal" href="liftoff.html">LIFTOFF</a></li>
<li class="toctree-l2"><a class="reference internal" href="lofreq.html">LOFREQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="macs2.html">MACS2</a></li>
<li class="toctree-l2"><a class="reference internal" href="maftools.html">MAFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="mapdamage2.html">MAPDAMAGE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="mashmap.html">MASHMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="md5sum.html">MD5SUM</a></li>
<li class="toctree-l2"><a class="reference internal" href="microphaser.html">MICROPHASER</a></li>
<li class="toctree-l2"><a class="reference internal" href="minimap2.html">MINIMAP2</a></li>
<li class="toctree-l2"><a class="reference internal" href="mixcr.html">MIXCR</a></li>
<li class="toctree-l2"><a class="reference internal" href="mlst.html">MLST</a></li>
<li class="toctree-l2"><a class="reference internal" href="mosdepth.html">MOSDEPTH</a></li>
<li class="toctree-l2"><a class="reference internal" href="msisensor.html">MSISENSOR</a></li>
<li class="toctree-l2"><a class="reference internal" href="msisensor_pro.html">MSISENSOR_PRO</a></li>
<li class="toctree-l2"><a class="reference internal" href="multiqc.html">MULTIQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="nanosim-h.html">NANOSIM-H</a></li>
<li class="toctree-l2"><a class="reference internal" href="nextflow.html">NEXTFLOW</a></li>
<li class="toctree-l2"><a class="reference internal" href="ngs-disambiguate.html">NGS-DISAMBIGUATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="open-cravat.html">OPEN-CRAVAT</a></li>
<li class="toctree-l2"><a class="reference internal" href="optitype.html">OPTITYPE</a></li>
<li class="toctree-l2"><a class="reference internal" href="paladin.html">PALADIN</a></li>
<li class="toctree-l2"><a class="reference internal" href="pandas.html">PANDAS</a></li>
<li class="toctree-l2"><a class="reference internal" href="pandora.html">PANDORA</a></li>
<li class="toctree-l2"><a class="reference internal" href="pbmm2.html">PBMM2</a></li>
<li class="toctree-l2"><a class="reference internal" href="pcaExplorer.html">PCAEXPLORER</a></li>
<li class="toctree-l2"><a class="reference internal" href="pear.html">PEAR</a></li>
<li class="toctree-l2"><a class="reference internal" href="pfam.html">PFAM</a></li>
<li class="toctree-l2"><a class="reference internal" href="pheatmap.html">PHEATMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="picard.html">PICARD</a></li>
<li class="toctree-l2"><a class="reference internal" href="pindel.html">PINDEL</a></li>
<li class="toctree-l2"><a class="reference internal" href="plass.html">PLASS</a></li>
<li class="toctree-l2"><a class="reference internal" href="preseq.html">PRESEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="primerclip.html">PRIMERCLIP</a></li>
<li class="toctree-l2"><a class="reference internal" href="prosolo.html">PROSOLO</a></li>
<li class="toctree-l2"><a class="reference internal" href="ptrimmer.html">PTRIMMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="pyfastaq.html">PYFASTAQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="qualimap.html">QUALIMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="rasusa.html">RASUSA</a></li>
<li class="toctree-l2"><a class="reference internal" href="razers3.html">RAZERS3</a></li>
<li class="toctree-l2"><a class="reference internal" href="rbt.html">RBT</a></li>
<li class="toctree-l2"><a class="reference internal" href="rebaler.html">REBALER</a></li>
<li class="toctree-l2"><a class="reference internal" href="reference.html">REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="refgenie.html">REFGENIE</a></li>
<li class="toctree-l2"><a class="reference internal" href="rest.html">REST</a></li>
<li class="toctree-l2"><a class="reference internal" href="rsem.html">RSEM</a></li>
<li class="toctree-l2"><a class="reference internal" href="rubic.html">RUBIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="sambamba.html">SAMBAMBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="samtools.html">SAMTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="sansa.html">SANSA</a></li>
<li class="toctree-l2"><a class="reference internal" href="seaborn.html">SEABORN</a></li>
<li class="toctree-l2"><a class="reference internal" href="sed.html">SED</a></li>
<li class="toctree-l2"><a class="reference internal" href="seqtk.html">SEQTK</a></li>
<li class="toctree-l2"><a class="reference internal" href="shovill.html">SHOVILL</a></li>
<li class="toctree-l2"><a class="reference internal" href="sickle.html">SICKLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="sigprofiler.html">SIGPROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="simplifyenrichment.html">SIMPLIFYENRICHMENT</a></li>
<li class="toctree-l2"><a class="reference internal" href="sleuth.html">SLEUTH</a></li>
<li class="toctree-l2"><a class="reference internal" href="snp-mutator.html">SNP-MUTATOR</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff.html">SNPEFF</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="sort.html">SORT</a></li>
<li class="toctree-l2"><a class="reference internal" href="sourmash.html">SOURMASH</a></li>
<li class="toctree-l2"><a class="reference internal" href="spades.html">SPADES</a></li>
<li class="toctree-l2"><a class="reference internal" href="spliceai.html">SPLICE AI</a></li>
<li class="toctree-l2"><a class="reference internal" href="sra-tools.html">SRA-TOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="star.html">STAR</a></li>
<li class="toctree-l2"><a class="reference internal" href="strelka.html">STRELKA</a></li>
<li class="toctree-l2"><a class="reference internal" href="strling.html">STRLING</a></li>
<li class="toctree-l2"><a class="reference internal" href="subread.html">SUBREAD</a></li>
<li class="toctree-l2"><a class="reference internal" href="tabix.html">TABIX</a></li>
<li class="toctree-l2"><a class="reference internal" href="tmb.html">TMB</a></li>
<li class="toctree-l2"><a class="reference internal" href="transdecoder.html">TRANSDECODER</a></li>
<li class="toctree-l2"><a class="reference internal" href="trim_galore.html">TRIM_GALORE</a></li>
<li class="toctree-l2"><a class="reference internal" href="trimmomatic.html">TRIMMOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="trinity.html">TRINITY</a></li>
<li class="toctree-l2"><a class="reference internal" href="tximeta.html">TXIMETA</a></li>
<li class="toctree-l2"><a class="reference internal" href="tximport.html">TXIMPORT</a></li>
<li class="toctree-l2"><a class="reference internal" href="ucsc.html">UCSC</a></li>
<li class="toctree-l2"><a class="reference internal" href="umis.html">UMIS</a></li>
<li class="toctree-l2"><a class="reference internal" href="unicycler.html">UNICYCLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="vardict.html">VARDICT</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_distribution.html">VARIANT_DISTRIBUTION</a></li>
<li class="toctree-l2"><a class="reference internal" href="variantoccurence.html">VARIANTOCCURENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="varscan.html">VARSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="vcftools.html">VCFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="vembrane.html">VEMBRANE</a></li>
<li class="toctree-l2"><a class="reference internal" href="vep.html">VEP</a></li>
<li class="toctree-l2"><a class="reference internal" href="vg.html">VG</a></li>
<li class="toctree-l2"><a class="reference internal" href="wgsim.html">WGSIM</a></li>
<li class="toctree-l2"><a class="reference internal" href="wisecondorx.html">WISECONDORX</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../wrappers.html">Wrappers</a> &raquo;</li>
      <li>BISEQ</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/wrappers/biseq.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="biseq">
<span id="bio-biseq"></span><h1>BISEQ<a class="headerlink" href="#biseq" title="Permalink to this heading"></a></h1>
<p>Analyse differentially methylated regions with BiSeq</p>
<p><strong>URL</strong>:</p>
<section id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this heading"></a></h2>
<p>This wrapper can be used in the following way:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">rule</span> <span class="n">test_biseq</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">bismark</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;path/to/bismark&quot;</span><span class="p">],</span>
        <span class="n">coldata</span> <span class="o">=</span> <span class="s2">&quot;path/to/coldata.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">covboxplot</span> <span class="o">=</span> <span class="s2">&quot;path/to/covboxplot.png&quot;</span><span class="p">,</span>
        <span class="n">raw</span> <span class="o">=</span> <span class="s2">&quot;path/to/biseq.RDS&quot;</span><span class="p">,</span>
        <span class="n">clustersites</span> <span class="o">=</span> <span class="s2">&quot;path/to/clustered_sites.tsv&quot;</span><span class="p">,</span>
        <span class="n">smooth_covboxplot</span> <span class="o">=</span> <span class="s2">&quot;path/to/smoothed_covboxplot.png&quot;</span><span class="p">,</span>
        <span class="n">groupwise_scatterplot</span> <span class="o">=</span> <span class="s2">&quot;path/to/scatterplot.png&quot;</span><span class="p">,</span>
        <span class="n">beta</span> <span class="o">=</span> <span class="s2">&quot;path/to/beta.regression.tsv&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">perc_samples</span> <span class="o">=</span> <span class="mi">4</span><span class="o">/</span><span class="mi">5</span><span class="p">,</span>
         <span class="n">min_sites</span> <span class="o">=</span> <span class="mi">20</span><span class="p">,</span>
         <span class="n">max_dist</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span>
         <span class="n">bandwidth</span> <span class="o">=</span> <span class="mi">80</span><span class="p">,</span>
         <span class="n">grid_dist</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
         <span class="n">formula</span> <span class="o">=</span> <span class="s2">&quot;~group&quot;</span><span class="p">,</span>
         <span class="n">link</span> <span class="o">=</span> <span class="s2">&quot;probit&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely.</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</section>
<section id="software-dependencies">
<h2>Software dependencies<a class="headerlink" href="#software-dependencies" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">bioconductor-biseq==1.32.0</span></code></p></li>
</ul>
</section>
<section id="input-output">
<h2>Input/Output<a class="headerlink" href="#input-output" title="Permalink to this heading"></a></h2>
<p><strong>Input:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">bismark</span></code>: CpG contextx from methylation extractor of bismark</p></li>
</ul>
<p><strong>Output:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">beta</span></code>: TSV formatted table containing differentially methylated clusters</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">raw</span></code>: Optional RDS-formatted BSraw object</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">covboxplot</span></code>: Optional boxplot of the coverage of CpG-sites</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">clustersites</span></code>: Optional table containing high density CpG cluster of regions</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">smooth_covboxplot</span></code>: Optional boxplot of the coverage of smoothed CpG-sites</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">groupwise_scatterplot</span></code>: Optional scatter plot giving the methilation level across two groups</p></li>
</ul>
</section>
<section id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this heading"></a></h2>
<p>notes</p>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
<section id="code">
<h2>Code<a class="headerlink" href="#code" title="Permalink to this heading"></a></h2>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="c1">#!/usr/bin/R</span>

<span class="n">base</span><span class="o">::</span><span class="nf">library</span><span class="p">(</span><span class="n">package</span><span class="o">=</span><span class="s">&quot;BiSeq&quot;</span><span class="p">,</span> <span class="n">quietly</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">);</span>



<span class="c1"># Graphical parameters</span>
<span class="n">w</span> <span class="o">&lt;-</span> <span class="m">1024</span><span class="p">;</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;w&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">w</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;w&quot;</span><span class="p">]]);</span>
<span class="p">}</span>
<span class="n">h</span> <span class="o">&lt;-</span> <span class="m">768</span><span class="p">;</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;h&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">h</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;h&quot;</span><span class="p">]]);</span>
<span class="p">}</span>

<span class="c1"># Convert input file path to string</span>
<span class="n">files</span> <span class="o">&lt;-</span> <span class="nf">sapply</span><span class="p">(</span>
  <span class="n">snakemake</span><span class="o">@</span><span class="n">input</span><span class="p">[[</span><span class="s">&quot;bismark&quot;</span><span class="p">]],</span>
  <span class="nf">function</span><span class="p">(</span><span class="n">p</span><span class="p">)</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">p</span><span class="p">)</span>
<span class="p">);</span>
<span class="n">base</span><span class="o">::</span><span class="nf">print</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="n">files</span><span class="p">));</span>

<span class="c1"># Import bismark into R</span>
<span class="n">coldata</span> <span class="o">&lt;-</span> <span class="n">utils</span><span class="o">::</span><span class="nf">read.table</span><span class="p">(</span>
  <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">input</span><span class="p">[[</span><span class="s">&quot;coldata&quot;</span><span class="p">]]),</span>
<span class="p">);</span>
<span class="nf">row.names</span><span class="p">(</span><span class="n">coldata</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">coldata</span><span class="o">$</span><span class="n">Sample_id</span>
<span class="n">base</span><span class="o">::</span><span class="nf">print</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="n">coldata</span><span class="p">));</span>

<span class="n">rrbs</span> <span class="o">&lt;-</span> <span class="n">BiSeq</span><span class="o">::</span><span class="nf">readBismark</span><span class="p">(</span>
  <span class="n">files</span><span class="o">=</span><span class="n">base</span><span class="o">::</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">files</span><span class="p">),</span>
  <span class="n">colData</span><span class="o">=</span><span class="n">coldata</span>
<span class="p">);</span>
<span class="n">base</span><span class="o">::</span><span class="nf">print</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="nf">rowRanges</span><span class="p">(</span><span class="n">rrbs</span><span class="p">)));</span>
<span class="n">base</span><span class="o">::</span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;CpG results from bismark loaded&quot;</span><span class="p">);</span>

<span class="c1"># Optional coverage and quality controls</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;covboxplot&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">covboxplot_path</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">[[</span><span class="s">&quot;covboxplot&quot;</span><span class="p">]]);</span>

  <span class="nf">png</span><span class="p">(</span>
    <span class="n">filename</span> <span class="o">=</span> <span class="n">covboxplot_path</span><span class="p">,</span>
    <span class="n">width</span> <span class="o">=</span> <span class="n">w</span><span class="p">,</span>
    <span class="n">height</span> <span class="o">=</span> <span class="n">h</span><span class="p">,</span>
    <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;px&quot;</span><span class="p">,</span>
    <span class="n">type</span> <span class="o">=</span> <span class="s">&quot;cairo&quot;</span>
  <span class="p">);</span>

  <span class="n">BiSeq</span><span class="o">::</span><span class="nf">covBoxplots</span><span class="p">(</span><span class="n">object</span><span class="o">=</span><span class="n">rrbs</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="s">&quot;cornflowerblue&quot;</span><span class="p">,</span> <span class="n">las</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span>

  <span class="nf">dev.off</span><span class="p">();</span>
  <span class="n">base</span><span class="o">::</span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;covBoxplot saved&quot;</span><span class="p">);</span>
<span class="p">}</span>

<span class="c1"># Optional RDS base object</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;raw&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">utils</span><span class="o">::</span><span class="nf">saveRDS</span><span class="p">(</span><span class="n">rrbs</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">[[</span><span class="s">&quot;raw&quot;</span><span class="p">]]));</span>
<span class="p">}</span>


<span class="c1"># Find BS clusters</span>
<span class="n">group_name</span> <span class="o">&lt;-</span> <span class="s">&quot;groups&quot;</span><span class="p">;</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;groups&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">group_name</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;groups&quot;</span><span class="p">]]);</span>
<span class="p">}</span>
<span class="n">groups</span> <span class="o">&lt;-</span> <span class="nf">colData</span><span class="p">(</span><span class="n">rrbs</span><span class="p">)[[</span><span class="n">group_name</span><span class="p">]];</span>

<span class="n">perc_samples</span> <span class="o">&lt;-</span> <span class="m">4</span><span class="o">/</span><span class="m">5</span><span class="p">;</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;perc_samples&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">perc_samples</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;perc_samples&quot;</span><span class="p">]]);</span>
<span class="p">}</span>

<span class="n">min_sites</span> <span class="o">&lt;-</span> <span class="m">20</span><span class="p">;</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;min_sites&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">min_sites</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;min_sites&quot;</span><span class="p">]]);</span>
<span class="p">}</span>

<span class="n">max_dist</span> <span class="o">&lt;-</span> <span class="m">100</span><span class="p">;</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;max_dist&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">max_dist</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;max_dist&quot;</span><span class="p">]]);</span>
<span class="p">}</span>

<span class="n">rrbs_clust_unlim</span> <span class="o">&lt;-</span> <span class="n">BiSeq</span><span class="o">::</span><span class="nf">clusterSites</span><span class="p">(</span>
  <span class="n">object</span><span class="o">=</span><span class="n">rrbs</span><span class="p">,</span>
  <span class="n">groups</span><span class="o">=</span><span class="n">groups</span><span class="p">,</span>
  <span class="n">perc.samples</span><span class="o">=</span><span class="n">perc_samples</span><span class="p">,</span>
  <span class="n">min.sites</span><span class="o">=</span><span class="n">min_sites</span><span class="p">,</span>
  <span class="n">max.dist</span><span class="o">=</span><span class="n">max_dist</span>
<span class="p">);</span>

<span class="c1"># Optional cluster sites table</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;clustersites&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">clusters_sites</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">rowRanges</span><span class="p">(</span><span class="n">rrbs_clust_unlim</span><span class="p">));</span>

  <span class="n">utils</span><span class="o">::</span><span class="nf">write.table</span><span class="p">(</span>
    <span class="n">x</span><span class="o">=</span><span class="n">clusters_sites</span><span class="p">,</span>
    <span class="n">file</span><span class="o">=</span><span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">[[</span><span class="s">&quot;clustersites&quot;</span><span class="p">]]),</span>
    <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span>
    <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
    <span class="n">col.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span>
    <span class="n">row.names</span><span class="o">=</span><span class="kc">FALSE</span>
  <span class="p">);</span>
<span class="p">}</span>

<span class="c1"># Filter</span>
<span class="n">ind_cov</span> <span class="o">&lt;-</span> <span class="nf">totalReads</span><span class="p">(</span><span class="n">rrbs_clust_unlim</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">;</span>
<span class="n">quant</span> <span class="o">&lt;-</span> <span class="nf">quantile</span><span class="p">(</span><span class="nf">totalReads</span><span class="p">(</span><span class="n">rrbs_clust_unlim</span><span class="p">)[</span><span class="n">ind_cov</span><span class="p">],</span> <span class="m">0.9</span><span class="p">);</span>
<span class="n">rrbs_clust_lim</span> <span class="o">&lt;-</span> <span class="nf">limitCov</span><span class="p">(</span><span class="n">object</span><span class="o">=</span><span class="n">rrbs_clust_unlim</span><span class="p">,</span> <span class="n">maxCov</span> <span class="o">=</span> <span class="n">quant</span><span class="p">);</span>

<span class="nf">if </span><span class="p">(</span><span class="s">&quot;smooth_covboxplot&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">covboxplot_path</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span>
    <span class="n">x</span> <span class="o">=</span> <span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">[[</span><span class="s">&quot;smooth_covboxplot&quot;</span><span class="p">]]</span>
  <span class="p">);</span>

  <span class="nf">png</span><span class="p">(</span>
    <span class="n">filename</span> <span class="o">=</span> <span class="n">covboxplot_path</span><span class="p">,</span>
    <span class="n">width</span> <span class="o">=</span> <span class="n">w</span><span class="p">,</span>
    <span class="n">height</span> <span class="o">=</span> <span class="n">h</span><span class="p">,</span>
    <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;px&quot;</span><span class="p">,</span>
    <span class="n">type</span> <span class="o">=</span> <span class="s">&quot;cairo&quot;</span>
  <span class="p">);</span>

  <span class="n">BiSeq</span><span class="o">::</span><span class="nf">covBoxplots</span><span class="p">(</span><span class="n">object</span><span class="o">=</span><span class="n">rrbs_clust_lim</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="s">&quot;cornflowerblue&quot;</span><span class="p">,</span> <span class="n">las</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span>

  <span class="nf">dev.off</span><span class="p">();</span>
<span class="p">}</span>

<span class="c1"># Predict methylation levels</span>
<span class="n">bandwidth</span> <span class="o">&lt;-</span> <span class="m">80</span><span class="p">;</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&#39;bandwidth&#39;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">bandwidth</span> <span class="o">&lt;-</span> <span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;bandwidth&quot;</span><span class="p">]];</span>
<span class="p">}</span>

<span class="n">grid_dist</span> <span class="o">&lt;-</span> <span class="kc">NULL</span><span class="p">;</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;grid_dist&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">grid_dist</span> <span class="o">&lt;-</span> <span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;grid_dist&quot;</span><span class="p">]];</span>
<span class="p">}</span>

<span class="n">predictedMeth</span> <span class="o">&lt;-</span> <span class="n">BiSeq</span><span class="o">::</span><span class="nf">predictMeth</span><span class="p">(</span>
  <span class="n">object</span><span class="o">=</span><span class="n">rrbs_clust_lim</span><span class="p">,</span>
  <span class="n">h</span><span class="o">=</span><span class="n">bandwidth</span><span class="p">,</span>
  <span class="n">grid_dist</span><span class="o">=</span><span class="n">grid_dist</span><span class="p">,</span>
  <span class="n">mc.cores</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">threads</span>
<span class="p">);</span>

<span class="c1"># Optionally plot groupwise scatterplot</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;groupwise_scatterplot&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">levels</span> <span class="o">&lt;-</span> <span class="nf">unique</span><span class="p">(</span><span class="n">coldata</span><span class="p">[[</span><span class="n">groups</span><span class="p">]]);</span>
  <span class="n">left</span> <span class="o">&lt;-</span> <span class="n">predictedMeth</span><span class="p">[,</span> <span class="nf">colData</span><span class="p">(</span><span class="n">predictedMeth</span><span class="p">)[[</span><span class="n">groups</span><span class="p">]]</span> <span class="o">==</span> <span class="n">levels</span><span class="p">[</span><span class="m">1</span><span class="p">]];</span>
  <span class="n">right</span> <span class="o">&lt;-</span> <span class="n">predictedMeth</span><span class="p">[,</span> <span class="nf">colData</span><span class="p">(</span><span class="n">predictedMeth</span><span class="p">)[[</span><span class="n">groups</span><span class="p">]]</span> <span class="o">==</span> <span class="n">levels</span><span class="p">[</span><span class="m">2</span><span class="p">]];</span>
  <span class="n">mean_left</span> <span class="o">&lt;-</span> <span class="nf">rowMeans</span><span class="p">(</span><span class="nf">methLevel</span><span class="p">(</span><span class="n">left</span><span class="p">));</span>
  <span class="n">mean_right</span> <span class="o">&lt;-</span> <span class="nf">rowMeans</span><span class="p">(</span><span class="nf">methLevel</span><span class="p">(</span><span class="n">right</span><span class="p">));</span>

  <span class="n">groupwise_scatterplot_path</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span>
    <span class="n">x</span> <span class="o">=</span> <span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">[[</span><span class="s">&quot;groupwise_scatterplot&quot;</span><span class="p">]]</span>
  <span class="p">);</span>

  <span class="nf">png</span><span class="p">(</span>
    <span class="n">filename</span> <span class="o">=</span> <span class="n">groupwise_scatterplot_path</span><span class="p">,</span>
    <span class="n">width</span> <span class="o">=</span> <span class="n">w</span><span class="p">,</span>
    <span class="n">height</span> <span class="o">=</span> <span class="n">h</span><span class="p">,</span>
    <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;px&quot;</span><span class="p">,</span>
    <span class="n">type</span> <span class="o">=</span> <span class="s">&quot;cairo&quot;</span>
  <span class="p">);</span>

  <span class="nf">plot</span><span class="p">(</span>
    <span class="n">mean_left</span><span class="p">,</span>
    <span class="n">mean_right</span><span class="p">,</span>
    <span class="n">col</span><span class="o">=</span><span class="s">&quot;blue&quot;</span><span class="p">,</span>
    <span class="n">xlab</span><span class="o">=</span><span class="n">base</span><span class="o">::</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Methylation in&quot;</span><span class="p">,</span> <span class="n">levels</span><span class="p">[</span><span class="m">1</span><span class="p">],</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot; &quot;</span><span class="p">),</span>
    <span class="n">ylab</span><span class="o">=</span><span class="n">base</span><span class="o">::</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Methylation in&quot;</span><span class="p">,</span> <span class="n">levels</span><span class="p">[</span><span class="m">2</span><span class="p">],</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot; &quot;</span><span class="p">)</span>
  <span class="p">)</span>

  <span class="nf">dev.off</span><span class="p">();</span>
<span class="p">}</span>

<span class="c1"># Fit beta model</span>
<span class="n">formula</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.formula</span><span class="p">(</span><span class="s">&quot;~groups&quot;</span><span class="p">);</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;formula&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">formula</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.formula</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;formula&quot;</span><span class="p">]]);</span>
<span class="p">}</span>

<span class="n">link</span> <span class="o">&lt;-</span> <span class="s">&quot;probit&quot;</span><span class="p">;</span>
<span class="nf">if </span><span class="p">(</span><span class="s">&quot;link&quot;</span> <span class="o">%in%</span> <span class="n">base</span><span class="o">::</span><span class="nf">names</span><span class="p">(</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">link</span> <span class="o">&lt;-</span> <span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">params</span><span class="p">[[</span><span class="s">&quot;link&quot;</span><span class="p">]]);</span>
<span class="p">}</span>

<span class="n">beta_results</span> <span class="o">&lt;-</span> <span class="n">BiSeq</span><span class="o">::</span><span class="nf">betaRegression</span><span class="p">(</span>
  <span class="n">formula</span><span class="o">=</span><span class="n">formula</span><span class="p">,</span>
  <span class="n">link</span><span class="o">=</span><span class="n">link</span><span class="p">,</span>
  <span class="n">object</span><span class="o">=</span><span class="n">predictedMeth</span><span class="p">,</span>
  <span class="n">mc.cores</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">threads</span><span class="p">,</span>
  <span class="n">type</span><span class="o">=</span><span class="s">&quot;BR&quot;</span>
<span class="p">);</span>

<span class="n">utils</span><span class="o">::</span><span class="nf">write.table</span><span class="p">(</span>
  <span class="n">x</span><span class="o">=</span><span class="n">beta_results</span><span class="p">,</span>
  <span class="n">file</span><span class="o">=</span><span class="n">base</span><span class="o">::</span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">snakemake</span><span class="o">@</span><span class="n">output</span><span class="p">[[</span><span class="s">&quot;beta&quot;</span><span class="p">]]),</span>
  <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span>
  <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
  <span class="n">col.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span>
  <span class="n">row.names</span><span class="o">=</span><span class="kc">FALSE</span>
<span class="p">);</span>
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="biomaRt/mouse_to_human.html" class="btn btn-neutral float-left" title="BIOMART" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="bismark.html" class="btn btn-neutral float-right" title="BISMARK" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>