<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>MULTIQC_RNASEQ_REPORT &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
        <script src="../../_static/clipboard.min.js"></script>
        <script src="../../_static/copybutton.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="ONCOKB_ANNOTATE" href="oncokb_annotate.html" />
    <link rel="prev" title="MSISENSOR_TABLE" href="msisensor_table.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../../wrappers.html">Wrappers</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../BiGR.html">BIGR</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="cancer_gene_census_annotate.html">CANCER_GENE_CENSUS_ANNOTATE</a></li>
<li class="toctree-l3"><a class="reference internal" href="copy.html">IGR_COPY</a></li>
<li class="toctree-l3"><a class="reference internal" href="drop_duplicates_md5.html">DROP_MD5_DUPLICATES</a></li>
<li class="toctree-l3"><a class="reference internal" href="fix_vcf.html">FIX VCF</a></li>
<li class="toctree-l3"><a class="reference internal" href="gleaves_compatibility.html">GLEAVES_COMPATIBILITY</a></li>
<li class="toctree-l3"><a class="reference internal" href="mane_to_vcftools_annotate.html">MANE_TO_VCFTOOLS_ANNOTATE</a></li>
<li class="toctree-l3"><a class="reference internal" href="msisensor_table.html">MSISENSOR_TABLE</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">MULTIQC_RNASEQ_REPORT</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l4"><a class="reference internal" href="#software-dependencies">Software dependencies</a></li>
<li class="toctree-l4"><a class="reference internal" href="#input-output">Input/Output</a></li>
<li class="toctree-l4"><a class="reference internal" href="#params">Params</a></li>
<li class="toctree-l4"><a class="reference internal" href="#authors">Authors</a></li>
<li class="toctree-l4"><a class="reference internal" href="#code">Code</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="oncokb_annotate.html">ONCOKB_ANNOTATE</a></li>
<li class="toctree-l3"><a class="reference internal" href="pharmkdb_annotate.html">CANCER_GENE_CENSUS_ANNOTATE</a></li>
<li class="toctree-l3"><a class="reference internal" href="rename_snpsift_maf_cols.html">RENAME_SNPSIFT_MAF_COLS</a></li>
<li class="toctree-l3"><a class="reference internal" href="split_design.html">SPLIT_DESIGN</a></li>
<li class="toctree-l3"><a class="reference internal" href="split_vcf_features.html">SPLIT VCF FEATURES</a></li>
<li class="toctree-l3"><a class="reference internal" href="split_vcf_multiallelic.html">SPLIT VCF MULTIALLELIC</a></li>
<li class="toctree-l3"><a class="reference internal" href="translate_vcf_seqoia.html">TRANSLATE_VCF_SEQOIA</a></li>
<li class="toctree-l3"><a class="reference internal" href="vcf_format_to_info.html">VCF_FORMAT_TO_INFO</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../Rsamtools.html">RSAMTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../SomaticSignatures.html">SOMATICSIGNATURES</a></li>
<li class="toctree-l2"><a class="reference internal" href="../VariantAnnotation.html">VARIANTANNOTATION</a></li>
<li class="toctree-l2"><a class="reference internal" href="../adapterremoval.html">ADAPTERREMOVAL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../arriba.html">ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../art.html">ART</a></li>
<li class="toctree-l2"><a class="reference internal" href="../assembly-stats.html">ASSEMBLY-STATS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../awk.html">AWK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bamtools.html">BAMTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bbtools.html">BBTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bcftools.html">BCFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bcl2fastq.html">BCF2FASTQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bedtools.html">BEDTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../benchmark.html">BENCHMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bgzip.html">BGZIP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../biobambam2.html">BIOBAMBAM2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bioinfokit.html">BIOINFOKIT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../biomaRt.html">BIOMART</a></li>
<li class="toctree-l2"><a class="reference internal" href="../biseq.html">BISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bismark.html">BISMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../blast.html">BLAST</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bowtie2.html">BOWTIE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../busco.html">BUSCO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bwa.html">BWA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bwa-mem2.html">BWA-MEM2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cairosvg.html">CAIROSVG</a></li>
<li class="toctree-l2"><a class="reference internal" href="../check_md5.html">CHECK MD5</a></li>
<li class="toctree-l2"><a class="reference internal" href="../chipseeker.html">CHIPSEEKER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../clustalo.html">CLUSTALO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../clusterProfiler.html">CLUSTERPROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../compress.html">COMPRESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../concatenate_fastq.html">CONCATENATE FASTQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../consensusclusterplus.html">CONSENSUSCLUSTERPLUS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cp.html">CP (BASH)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cpat.html">CPAT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cutadapt.html">CUTADAPT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../dada2.html">DADA2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../deeptools.html">DEEPTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../deepvariant.html">DEEPVARIANT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../delly.html">DELLY</a></li>
<li class="toctree-l2"><a class="reference internal" href="../deseq2.html">DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../diamond.html">DIAMOND</a></li>
<li class="toctree-l2"><a class="reference internal" href="../dragmap.html">DRAGMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../eacon.html">EACON</a></li>
<li class="toctree-l2"><a class="reference internal" href="../enhancedVolcano.html">ENHANCEDVOLCANO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ensembl.html">ENSEMBL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../epic.html">EPIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../facets.html">FACETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fastp.html">FASTP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fastq_screen.html">FASTQ_SCREEN</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fastqc.html">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fgbio.html">FGBIO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fgsea.html">FGSEA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtlong.html">FILTLONG</a></li>
<li class="toctree-l2"><a class="reference internal" href="../freebayes.html">FREEBAYES</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gatk.html">GATK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gatk3.html">GATK3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gdc-api.html">GDC-API</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gdc-client.html">GDC-CLIENT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gencode.html">GENCODE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../genomepy.html">GENOMEPY</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gridss.html">GRIDSS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gsva.html">GSVA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gtf.html">GTF</a></li>
<li class="toctree-l2"><a class="reference internal" href="../hap.py.html">HAP.PY</a></li>
<li class="toctree-l2"><a class="reference internal" href="../hisat2.html">HISAT2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../hmmer.html">HMMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../homer.html">HOMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../iRODS.html">IRODS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../igv-reports.html">IGV-REPORTS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../immunedeconv.html">IMMUNEDECONV</a></li>
<li class="toctree-l2"><a class="reference internal" href="../infernal.html">INFERNAL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../isoformSwitchAnalyseR.html">ISOFORMSWITCHANALYSER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../jannovar.html">JANNOVAR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../kallisto.html">KALLISTO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../last.html">LAST</a></li>
<li class="toctree-l2"><a class="reference internal" href="../liftoff.html">LIFTOFF</a></li>
<li class="toctree-l2"><a class="reference internal" href="../lofreq.html">LOFREQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../macs2.html">MACS2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../maftools.html">MAFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../manta.html">MANTA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mapdamage2.html">MAPDAMAGE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mashmap.html">MASHMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../md5sum.html">MD5SUM</a></li>
<li class="toctree-l2"><a class="reference internal" href="../microphaser.html">MICROPHASER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../minimap2.html">MINIMAP2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mixcr.html">MIXCR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mlst.html">MLST</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mosdepth.html">MOSDEPTH</a></li>
<li class="toctree-l2"><a class="reference internal" href="../msisensor.html">MSISENSOR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../msisensor_pro.html">MSISENSOR_PRO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../multiqc.html">MULTIQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../muscle.html">MUSCLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../nanosim-h.html">NANOSIM-H</a></li>
<li class="toctree-l2"><a class="reference internal" href="../nextflow.html">NEXTFLOW</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ngs-disambiguate.html">NGS-DISAMBIGUATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../open-cravat.html">OPEN-CRAVAT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../optitype.html">OPTITYPE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../paladin.html">PALADIN</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pandas.html">PANDAS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pandora.html">PANDORA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pbmm2.html">PBMM2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pcaExplorer.html">PCAEXPLORER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pear.html">PEAR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pfam.html">PFAM</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pheatmap.html">PHEATMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../picard.html">PICARD</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pindel.html">PINDEL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../plass.html">PLASS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../preseq.html">PRESEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../primerclip.html">PRIMERCLIP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../prosolo.html">PROSOLO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ptrimmer.html">PTRIMMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pyfastaq.html">PYFASTAQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../qualimap.html">QUALIMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rasusa.html">RASUSA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../razers3.html">RAZERS3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rbt.html">RBT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rebaler.html">REBALER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../reference.html">REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../refgenie.html">REFGENIE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rest.html">REST</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rsem.html">RSEM</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rubic.html">RUBIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sambamba.html">SAMBAMBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../samtools.html">SAMTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sansa.html">SANSA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../seaborn.html">SEABORN</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sed.html">SED</a></li>
<li class="toctree-l2"><a class="reference internal" href="../seqtk.html">SEQTK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../shovill.html">SHOVILL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sickle.html">SICKLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sigprofiler.html">SIGPROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../simplifyenrichment.html">SIMPLIFYENRICHMENT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sleuth.html">SLEUTH</a></li>
<li class="toctree-l2"><a class="reference internal" href="../snp-mutator.html">SNP-MUTATOR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../snpeff.html">SNPEFF</a></li>
<li class="toctree-l2"><a class="reference internal" href="../snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sort.html">SORT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sourmash.html">SOURMASH</a></li>
<li class="toctree-l2"><a class="reference internal" href="../spades.html">SPADES</a></li>
<li class="toctree-l2"><a class="reference internal" href="../spliceai.html">SPLICE AI</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sra-tools.html">SRA-TOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../star.html">STAR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../strelka.html">STRELKA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../strling.html">STRLING</a></li>
<li class="toctree-l2"><a class="reference internal" href="../subread.html">SUBREAD</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tabix.html">TABIX</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tmb.html">TMB</a></li>
<li class="toctree-l2"><a class="reference internal" href="../transdecoder.html">TRANSDECODER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../trim_galore.html">TRIM_GALORE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../trimmomatic.html">TRIMMOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../trinity.html">TRINITY</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tximeta.html">TXIMETA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tximport.html">TXIMPORT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ucsc.html">UCSC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../umis.html">UMIS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../unicycler.html">UNICYCLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vardict.html">VARDICT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../variant_distribution.html">VARIANT_DISTRIBUTION</a></li>
<li class="toctree-l2"><a class="reference internal" href="../variantoccurence.html">VARIANTOCCURENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../varscan.html">VARSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vcftools.html">VCFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vembrane.html">VEMBRANE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vep.html">VEP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../verifybamid.html">VERIFYBAMID</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vg.html">VG</a></li>
<li class="toctree-l2"><a class="reference internal" href="../wgsim.html">WGSIM</a></li>
<li class="toctree-l2"><a class="reference internal" href="../wisecondorx.html">WISECONDORX</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../meta-wrappers.html">Meta-Wrappers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../../wrappers.html">Wrappers</a> &raquo;</li>
          <li><a href="../BiGR.html">BIGR</a> &raquo;</li>
      <li>MULTIQC_RNASEQ_REPORT</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/wrappers/BiGR/multiqc_rnaseq_report.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="multiqc-rnaseq-report">
<span id="bio-bigr-multiqc-rnaseq-report"></span><h1>MULTIQC_RNASEQ_REPORT<a class="headerlink" href="#multiqc-rnaseq-report" title="Permalink to this headline"></a></h1>
<p>Build a result report for RNASeq analysis</p>
<p><strong>URL</strong>:</p>
<section id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this headline"></a></h2>
<p>This wrapper can be used in the following way:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">rule</span> <span class="n">multiqc_config</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
         <span class="n">pairwise_scatterplot</span> <span class="o">=</span> <span class="s2">&quot;image.png&quot;</span><span class="p">,</span>
         <span class="n">clustermap_sample</span> <span class="o">=</span> <span class="s2">&quot;image.png&quot;</span><span class="p">,</span>
         <span class="n">pca_plot</span> <span class="o">=</span> <span class="s2">&quot;image.png&quot;</span><span class="p">,</span>
         <span class="n">pca_axes_correlation</span> <span class="o">=</span> <span class="s2">&quot;image.png&quot;</span><span class="p">,</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">multiqc_config</span> <span class="o">=</span> <span class="s2">&quot;multiqc_config.yaml&quot;</span><span class="p">,</span>
        <span class="n">plots</span> <span class="o">=</span> <span class="p">[</span>
            <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;pairwise_scatterplot_mqc.png&quot;</span><span class="p">),</span>
            <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;clustermap_sample_mqc.png&quot;</span><span class="p">),</span>
            <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;pca_plot_mqc.png&quot;</span><span class="p">),</span>
            <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;pca_axes_correlation_mqc.png&quot;</span><span class="p">)</span>
        <span class="p">]</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/multiqc/config.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">title</span> <span class="o">=</span> <span class="s2">&quot;My title&quot;</span><span class="p">,</span>
        <span class="n">subtitle</span> <span class="o">=</span> <span class="s2">&quot;My subtitle that catches readers&quot;</span><span class="p">,</span>
        <span class="n">intro_text</span> <span class="o">=</span> <span class="s2">&quot;Intro text that summs up may things&quot;</span><span class="p">,</span>
        <span class="n">report_comment</span> <span class="o">=</span> <span class="s2">&quot;A comment about this report&quot;</span><span class="p">,</span>
        <span class="n">show_analysis_paths</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
        <span class="n">show_analysis_time</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
        <span class="n">custom_logo</span> <span class="o">=</span> <span class="s1">&#39;../IGR_Logo.jpeg&#39;</span><span class="p">,</span>
        <span class="n">custom_logo_url</span> <span class="o">=</span> <span class="s1">&#39;https://gitlab.com/bioinfo_gustaveroussy/bigr&#39;</span><span class="p">,</span>
        <span class="n">custom_logo_title</span> <span class="o">=</span> <span class="s1">&#39;BiGR, Gustave Roussy Intitute&#39;</span><span class="p">,</span>
        <span class="n">report_header_info</span> <span class="o">=</span> <span class="p">[</span>
            <span class="p">{</span><span class="s2">&quot;Contact E-mail&quot;</span><span class="p">:</span> <span class="s2">&quot;my_address@gustaveroussy.fr&quot;</span><span class="p">},</span>
            <span class="p">{</span><span class="s2">&quot;Application Type&quot;</span><span class="p">:</span> <span class="s2">&quot;RNA-seq&quot;</span><span class="p">},</span>
            <span class="p">{</span><span class="s2">&quot;Project Type&quot;</span><span class="p">:</span> <span class="s2">&quot;Application&quot;</span><span class="p">},</span>
            <span class="p">{</span><span class="s2">&quot;Sequencing Platform&quot;</span><span class="p">:</span> <span class="s2">&quot;HiSeq 2500 High Output V4&quot;</span><span class="p">},</span>
            <span class="p">{</span><span class="s2">&quot;Sequencing Setup&quot;</span><span class="p">:</span> <span class="s2">&quot;2x125&quot;</span><span class="p">}</span>
        <span class="p">]</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;0.79.0-2121-g9eee8d1be/bio/BiGR/multiqc_rnaseq_report&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely.</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</section>
<section id="software-dependencies">
<h2>Software dependencies<a class="headerlink" href="#software-dependencies" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">bioconda::multiqc==1.9</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge::python==3.9.5</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge::pandas==1.2.5</span></code></p></li>
</ul>
</section>
<section id="input-output">
<h2>Input/Output<a class="headerlink" href="#input-output" title="Permalink to this headline"></a></h2>
<p><strong>Input:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">clustermap_sample</span></code>: Optional clustered heatmap of all samples</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">clustermap_genes</span></code>: Optional clustered heatmap of DEG</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">pca_axes_correlation</span></code>: Optional barplot of the correlation between samples and experimental design factors.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">pairwise_scatterplot</span></code>: Optional pairwise scatterplot</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">pca_plot</span></code>: Optional PCA</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">volcanoplot</span></code>: Optional volcano plot</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">distro_expr</span></code>: Optional distribution of genes expression per sample</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">ma_plot</span></code>: Optional MA-Plot</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">consensus_cluster_plus</span></code>: Optional consensus_cluster_plus result</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">titv_plot</span></code>: Optional Ti/Tv plot from maftools</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">maf_summary</span></code>: Optional maf summary</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">oncoplot</span></code>: Optional oncoplot</p></li>
</ul>
<p><strong>Output:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">multiqc_config</span></code>: Path to a yaml file that holds MultiQC config</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">plots</span></code>: A list of graphs depending on the input files</p></li>
</ul>
</section>
<section id="params">
<h2>Params<a class="headerlink" href="#params" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">title</span></code>: Optional report title</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">subtitle</span></code>: Optional report subtitle</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">intro_text</span></code>: Optional intro text,</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">report_comment</span></code>: Optional comment,</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">show_analysis_paths</span></code>: Boolean. Display (default) or hide results paths on the computing machine.,</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">show_analysis_time</span></code>: Boolean. Display (default) or hide computation times.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">custom_logo</span></code>: Optional path to a PNG logo,</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">custom_logo_url</span></code>: Optional path to a web page,</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">custom_logo_title</span></code>: Optional text as a replacement of the logo.,</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">report_header_info</span></code>: [{‘Contact E-mail’: ‘An optional email address’}, {‘Application Type’: ‘Optional type of analysis’}, {‘Project Type’: ‘Optional kind of analysed project’}, {‘Sequencing Platform’: ‘Optional info towards the sequencing platform’}, {‘Sequencing Setup’: ‘Optional info about the sequencing kit/method’}]</p></li>
</ul>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
<section id="code">
<h2>Code<a class="headerlink" href="#code" title="Permalink to this headline"></a></h2>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/usr/bin/python3.8</span>
<span class="c1"># -*- coding: utf-8 -*-</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">From the given files, produce specific MultiQC yaml files that witll ship</span>
<span class="sd">content info MultiQC automatically</span>
<span class="sd">&quot;&quot;&quot;</span>

<span class="kn">import</span> <span class="nn">yaml</span>

<span class="kn">from</span> <span class="nn">shutil</span> <span class="kn">import</span> <span class="n">copyfile</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>
<span class="kn">from</span> <span class="nn">snakemake.shell</span> <span class="kn">import</span> <span class="n">shell</span>
<span class="kn">from</span> <span class="nn">typing</span> <span class="kn">import</span> <span class="n">Any</span><span class="p">,</span> <span class="n">Dict</span>
<span class="kn">import</span> <span class="nn">logging</span>

<span class="c1"># Snakemake loggings</span>
<span class="n">log</span> <span class="o">=</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">log_fmt_shell</span><span class="p">(</span><span class="n">stdout</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">stderr</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">append</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="c1"># Python loggings</span>
<span class="n">logging</span><span class="o">.</span><span class="n">basicConfig</span><span class="p">(</span>
    <span class="n">filename</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">log</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
    <span class="n">filemode</span><span class="o">=</span><span class="s2">&quot;w&quot;</span><span class="p">,</span>
    <span class="n">level</span><span class="o">=</span><span class="n">logging</span><span class="o">.</span><span class="n">DEBUG</span>
<span class="p">)</span>


<span class="n">section_dict</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;clustermap_sample&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;Clustermap_Sample&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Sample clustered heatmap&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;Per sample clustered heatmap. This plot has been build based on &quot;</span>
            <span class="s2">&quot;normalized counts from DESeq2.&lt;br&gt;&lt;br&gt;Colors run from red &quot;</span>
            <span class="s2">&quot;(identical) to blue (different). Samples are named using the &quot;</span>
            <span class="s2">&quot;following scheme: condition - sample_id (powered by in-house &quot;</span>
            <span class="s2">&quot;python scripts: pandas, seaborn and scipy)&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;clustermap_sample_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;clustermap_genes&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;Clustermap_Genes&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Heatmap of samples clustered over genes&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;Per genes clustered heatmap. This plot has been build based on &quot;</span>
            <span class="s2">&quot;filtered normalized counts from DESeq2.&lt;br&gt;&lt;br&gt;Colors run from red&quot;</span>
            <span class="s2">&quot; (identical) to blue (different). Samples are named using the &quot;</span>
            <span class="s2">&quot;following scheme: condition - sample_id. (powered by in-house &quot;</span>
            <span class="s2">&quot;python scripts: pandas, seaborn and scipy)&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;clustermap_gene_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;PCA_axes_correlation&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;PCA_axes_correlation&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;PCA Axes correlations&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;This histogram shows which factor has a higher  correlation with &quot;</span>
            <span class="s2">&quot;the first axe of the PCA. This may help to understand why a DGE &quot;</span>
            <span class="s2">&quot;returns odd results, or which factor masks another.&lt;br&gt;&lt;br&gt;&quot;</span>
            <span class="s2">&quot;This does not foresee anything for the upcomming differential &quot;</span>
            <span class="s2">&quot;analysis. (powered by pcaExplorer)&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;pca_axes_correlation_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;Pairwise_Scatterplot&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;Pairwise_Scatterplot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Pairwise Scatterplot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;This is a pairwise scatterplot. It contains: (1) in diagonal: the &quot;</span>
            <span class="s2">&quot;sample names, (2) in upper half the pearson correlation between &quot;</span>
            <span class="s2">&quot;samples, (3) in lower half, the scatterplot of each genic feature &quot;</span>
            <span class="s2">&quot;against each other. &lt;br&gt;&lt;br&gt;A value of 0.95 and above shows high &quot;</span>
            <span class="s2">&quot;similarities like ones observed in cell lines. A value between &quot;</span>
            <span class="s2">&quot;0.95 and 0.80 shows average similarities. A lesser value shows &quot;</span>
            <span class="s2">&quot;noticable dissimilarities.&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;pairwise_scatterplot_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;pca_plot&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;PCA_Plot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Principal Component Analysis&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;Principal Component Analysis (PCA) is a common method to display &quot;</span>
            <span class="s2">&quot;samples similarities and divergences.&lt;br&gt;&lt;br&gt; Each point represents &quot;</span>
            <span class="s2">&quot;a sample. Two points that are close from each others, are similar to &quot;</span>
            <span class="s2">&quot;each others. Two points, that are apart from each others, are &quot;</span>
            <span class="s2">&quot;different. (powered by pcaExplorer)&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;pca_plot_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;volcanoplot&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;Volcano_Plot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Volcano Plot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;The volcanoplot shows the repartition of genes across two &quot;</span>
            <span class="s2">&quot;dimensions: their log2(Fold Change) and the -log10(Adjusted PValue)&quot;</span>
            <span class="s2">&quot;&lt;br&gt;&lt;br&gt;Each point is a gene. The higher the point, the greater &quot;</span>
            <span class="s2">&quot;is the confidence we have in the fact that it is differentially &quot;</span>
            <span class="s2">&quot;expressed according to our factor of interest. The more apart a &quot;</span>
            <span class="s2">&quot;point is from the center of the graph, the more differentially &quot;</span>
            <span class="s2">&quot;expressed it is. (powered by EnhancedVolcano)&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;volcanoplot_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;distro_expr&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;Expression_Distribution&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Expression Distribution&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;The distribution of normalised genes expression is a common quality &quot;</span>
            <span class="s2">&quot;control performed over a differential gene analysis. &lt;br&gt;&lt;br&gt; We &quot;</span>
            <span class="s2">&quot;expect all samples to have similar distribution, otherwise, it would&quot;</span>
            <span class="s2">&quot; sign a possible error before the end of the normalization process.&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;distro_expr_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;distro_mu&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;Mu_Distribution&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;µ(ij) Distribution&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;The distribution of gene&#39;s µ(ij) in DESeq2 is a common quality &quot;</span>
            <span class="s2">&quot;control performed over a differential gene analysis. &lt;br&gt;&lt;br&gt; We &quot;</span>
            <span class="s2">&quot;expect all samples to have similar distribution, otherwise, it would&quot;</span>
            <span class="s2">&quot; sign a possible error before the end of the normalization process.&quot;</span>
            <span class="s2">&quot;&lt;br&gt;&lt;br&gt;µ(ij) are numerical values used by DESeq2 within its NB-GML.&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;distro_mu_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;pvalue_qc&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;PValue_QC&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Ajdusted P-Value distribution&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;Ajusted P-Values should have less &#39;non significative&#39; observations &quot;</span>
            <span class="s2">&quot;than significative ones. Thus, these variations of expression  &quot;</span>
            <span class="s2">&quot;should come from all studied chromosomes.&quot;</span>

        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;pvalue_qc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;inde_theta_filter&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;Independent_Filtering_Theta_Threshold&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Independent Filtering (Theta threshold)&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;Theta threshold (in red) if used in independent filtering to define &quot;</span>
            <span class="s2">&quot;weather an observation should be filtered out or not. This graph &quot;</span>
            <span class="s2">&quot;should have two knees, and the Theta value should come befor the &quot;</span>
            <span class="s2">&quot;first one.&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;theta_knee_plot_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;ma_plot&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;MA_Plot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;MA-plot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;The MAplot compares M (log(ratio)) to A (mean average).&lt;br&gt;&lt;br&gt; &quot;</span>
            <span class="s2">&quot;This helps to highlight possible normalisation issues. We expect no&quot;</span>
            <span class="s2">&quot; direct link between these two values; the graph should remain &quot;</span>
            <span class="s2">&quot;centered around zero. (powered by DESeq2)&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;ma_plot_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;independent_filter&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;Independent_Filtering&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Independent Filtering&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;Independent filtering allows to identify outlayers and possible &quot;</span>
            <span class="s2">&quot;false positive low p-values. These graphs are here to illustrate &quot;</span>
            <span class="s2">&quot;the amount of calls filtered out by this method.&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;independent_filtering_mqc.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;consensus_cluster_plus&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;consensus_cluster_plus&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Consensus Clustering&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;This graph has been done on DESeq2 normalized counts..&lt;br&gt;&lt;br&gt; &quot;</span>
            <span class="s2">&quot;This helps to highlight possible confounding effects. We expect &quot;</span>
            <span class="s2">&quot;our sample to be well separated over the factor of interest. (powered by &quot;</span>
            <span class="s2">&quot;&#39;consensus cluster plus&#39;)&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;consensus_cluster_plus.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;titv_plot&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;titv_plot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Transition over Transversions&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;This graph shows both the Transition/Transversion ratios and &quot;</span>
            <span class="s2">&quot;the contributions of 6 mutational conversion classes in each &quot;</span>
            <span class="s2">&quot;sample. (powered by MafTools)&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;titv_plot.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;maf_summary&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;maf_summary&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Summary of the MAF content&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;Six graphs are available:&lt;br&gt;&lt;br&gt; (1) Variant classification &quot;</span>
            <span class="s2">&quot;shows the absolute number of variants in each classes. &lt;br&gt;(2) &quot;</span>
            <span class="s2">&quot;Variant type shows the main types of variants and their abolute &quot;</span>
            <span class="s2">&quot;number in the maf file. &lt;br&gt;(3) SNV class shows the Ti/Tv mutation &quot;</span>
            <span class="s2">&quot;classes among the whole maf file. &lt;br&gt;(4) Variant per sample shows the&quot;</span>
            <span class="s2">&quot; number of variants in each sample (the colors are the same as &quot;</span>
            <span class="s2">&quot;within the (1)&#39;Variant classification&#39; graph. &lt;br&gt;(5) this graph &quot;</span>
            <span class="s2">&quot;highlights the dispersion of the absolute counts observed in (1)&quot;</span>
            <span class="s2">&quot;&#39;Variant_Classification&#39; graph, across the whole cohort. Finally&quot;</span>
            <span class="s2">&quot;&lt;br&gt;(6) Top X mutated genes, shows the top X genes with the highest &quot;</span>
            <span class="s2">&quot;number of mutations in the whole cohort.&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;mafsummary.png&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;oncoplot&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;section&quot;</span><span class="p">:</span> <span class="s2">&quot;oncoplot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;title&quot;</span><span class="p">:</span> <span class="s2">&quot;Oncoplot&quot;</span><span class="p">,</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s2">&quot;This graph is an oncoplot of the genes given in Y. The samples &quot;</span>
            <span class="s2">&quot;are on the X axis.&quot;</span>
        <span class="p">),</span>
        <span class="s2">&quot;plot_name&quot;</span><span class="p">:</span> <span class="s2">&quot;oncoplot.png&quot;</span>
    <span class="p">}</span>
<span class="p">}</span>


<span class="k">def</span> <span class="nf">write_yaml</span><span class="p">(</span><span class="n">output_yaml</span><span class="p">:</span> <span class="n">Path</span><span class="p">,</span> <span class="n">data</span><span class="p">:</span> <span class="n">Dict</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">Any</span><span class="p">])</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Save given dictionnary as Yaml-formatted text file</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">with</span> <span class="n">output_yaml</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="s2">&quot;w&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">outyaml</span><span class="p">:</span>
        <span class="n">yaml</span><span class="o">.</span><span class="n">dump</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">outyaml</span><span class="p">,</span> <span class="n">default_flow_style</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>


<span class="n">mqc_config</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;custom_data&quot;</span><span class="p">:</span> <span class="p">{},</span> <span class="s2">&quot;sp&quot;</span><span class="p">:</span> <span class="p">{},</span> <span class="s2">&quot;ignore_images&quot;</span><span class="p">:</span> <span class="kc">False</span><span class="p">}</span>
<span class="n">mqc_config</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">)</span>
<span class="n">config_outpath</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;multiqc_config&quot;</span><span class="p">])</span>

<span class="k">for</span> <span class="n">section</span><span class="p">,</span> <span class="n">plot</span> <span class="ow">in</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">output_dir</span> <span class="o">=</span> <span class="n">config_outpath</span><span class="o">.</span><span class="n">absolute</span><span class="p">()</span><span class="o">.</span><span class="n">parent</span>
    <span class="n">output_plot</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">config_outpath</span><span class="o">.</span><span class="n">absolute</span><span class="p">()</span><span class="o">.</span><span class="n">parent</span><span class="si">}</span><span class="s2">/</span><span class="si">{</span><span class="n">section</span><span class="si">}</span><span class="s2">_mqc.png&quot;</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span>
        <span class="s2">&quot;Working on </span><span class="si">%s</span><span class="s2"> (</span><span class="si">%s</span><span class="s2">). Saving it as </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">section</span><span class="p">,</span> <span class="n">plot</span><span class="p">,</span> <span class="n">output_plot</span>
    <span class="p">)</span>
    <span class="n">shell</span><span class="p">(</span><span class="s2">&quot;ln --relative --symbolic --force </span><span class="si">{plot}</span><span class="s2"> </span><span class="si">{output_plot}</span><span class="s2"> </span><span class="si">{log}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">mqc_config</span><span class="p">[</span><span class="s2">&quot;custom_data&quot;</span><span class="p">][</span><span class="n">section</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s2">&quot;section_name&quot;</span><span class="p">:</span> <span class="n">section_dict</span><span class="p">[</span><span class="n">section</span><span class="p">][</span><span class="s2">&quot;title&quot;</span><span class="p">],</span>
        <span class="s2">&quot;description&quot;</span><span class="p">:</span> <span class="n">section_dict</span><span class="p">[</span><span class="n">section</span><span class="p">][</span><span class="s1">&#39;description&#39;</span><span class="p">]</span>
    <span class="p">}</span>
    <span class="n">mqc_config</span><span class="p">[</span><span class="s2">&quot;sp&quot;</span><span class="p">][</span><span class="n">section</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;fn&quot;</span><span class="p">:</span> <span class="n">output_plot</span><span class="p">}</span>

<span class="n">write_yaml</span><span class="p">(</span>
    <span class="n">output_yaml</span><span class="o">=</span><span class="n">Path</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;multiqc_config&quot;</span><span class="p">]),</span>
    <span class="n">data</span><span class="o">=</span><span class="n">mqc_config</span>
<span class="p">)</span>
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="msisensor_table.html" class="btn btn-neutral float-left" title="MSISENSOR_TABLE" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="oncokb_annotate.html" class="btn btn-neutral float-right" title="ONCOKB_ANNOTATE" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>