<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>PLOT_GENES &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../../_static/doctools.js"></script>
        <script src="../../_static/sphinx_highlight.js"></script>
        <script src="../../_static/clipboard.min.js"></script>
        <script src="../../_static/copybutton.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="PLOT_SIGPROFILER_SINGLE_SAMPLE" href="plot_sigprofiler_single_sample.html" />
    <link rel="prev" title="PCA" href="pca.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../../wrappers.html">Wrappers</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../BiGR.html">BIGR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Rsamtools.html">RSAMTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../SomaticSignatures.html">SOMATICSIGNATURES</a></li>
<li class="toctree-l2"><a class="reference internal" href="../VariantAnnotation.html">VARIANTANNOTATION</a></li>
<li class="toctree-l2"><a class="reference internal" href="../adapterremoval.html">ADAPTERREMOVAL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../arriba.html">ARRIBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../art.html">ART</a></li>
<li class="toctree-l2"><a class="reference internal" href="../assembly-stats.html">ASSEMBLY-STATS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../awk.html">AWK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bamtools.html">BAMTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bbtools.html">BBTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bcftools.html">BCFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bcl2fastq.html">BCF2FASTQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bedtools.html">BEDTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../benchmark.html">BENCHMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../biobambam2.html">BIOBAMBAM2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bioinfokit.html">BIOINFOKIT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../biomaRt.html">BIOMART</a></li>
<li class="toctree-l2"><a class="reference internal" href="../biseq.html">BISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bismark.html">BISMARK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../blast.html">BLAST</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bowtie2.html">BOWTIE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../busco.html">BUSCO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bwa.html">BWA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../bwa-mem2.html">BWA-MEM2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cairosvg.html">CAIROSVG</a></li>
<li class="toctree-l2"><a class="reference internal" href="../check_md5.html">CHECK MD5</a></li>
<li class="toctree-l2"><a class="reference internal" href="../chipseeker.html">CHIPSEEKER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../clustalo.html">CLUSTALO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../clusterProfiler.html">CLUSTERPROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../compress.html">COMPRESS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../concatenate_fastq.html">CONCATENATE FASTQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../consensusclusterplus.html">CONSENSUSCLUSTERPLUS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cp.html">CP (BASH)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cpat.html">CPAT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cutadapt.html">CUTADAPT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../dada2.html">DADA2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../deeptools.html">DEEPTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../deepvariant.html">DEEPVARIANT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../delly.html">DELLY</a></li>
<li class="toctree-l2"><a class="reference internal" href="../deseq2.html">DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../diamond.html">DIAMOND</a></li>
<li class="toctree-l2"><a class="reference internal" href="../eacon.html">EACON</a></li>
<li class="toctree-l2"><a class="reference internal" href="../enhancedVolcano.html">ENHANCEDVOLCANO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ensembl.html">ENSEMBL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../epic.html">EPIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../facets.html">FACETS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fastp.html">FASTP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fastq_screen.html">FASTQ_SCREEN</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fastqc.html">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fgbio.html">FGBIO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fgsea.html">FGSEA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtlong.html">FILTLONG</a></li>
<li class="toctree-l2"><a class="reference internal" href="../freebayes.html">FREEBAYES</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gatk.html">GATK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gatk3.html">GATK3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gdc-api.html">GDC-API</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gdc-client.html">GDC-CLIENT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gencode.html">GENCODE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../genomepy.html">GENOMEPY</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gridss.html">GRIDSS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gsva.html">GSVA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gtf.html">GTF</a></li>
<li class="toctree-l2"><a class="reference internal" href="../hap.py.html">HAP.PY</a></li>
<li class="toctree-l2"><a class="reference internal" href="../hisat2.html">HISAT2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../hmmer.html">HMMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../homer.html">HOMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../iRODS.html">IRODS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../igv-reports.html">IGV-REPORTS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../immunedeconv.html">IMMUNEDECONV</a></li>
<li class="toctree-l2"><a class="reference internal" href="../infernal.html">INFERNAL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../isoformSwitchAnalyseR.html">ISOFORMSWITCHANALYSER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../jannovar.html">JANNOVAR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../kallisto.html">KALLISTO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../last.html">LAST</a></li>
<li class="toctree-l2"><a class="reference internal" href="../liftoff.html">LIFTOFF</a></li>
<li class="toctree-l2"><a class="reference internal" href="../lofreq.html">LOFREQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../macs2.html">MACS2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../maftools.html">MAFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mapdamage2.html">MAPDAMAGE2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mashmap.html">MASHMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../md5sum.html">MD5SUM</a></li>
<li class="toctree-l2"><a class="reference internal" href="../microphaser.html">MICROPHASER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../minimap2.html">MINIMAP2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mixcr.html">MIXCR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mlst.html">MLST</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mosdepth.html">MOSDEPTH</a></li>
<li class="toctree-l2"><a class="reference internal" href="../msisensor.html">MSISENSOR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../msisensor_pro.html">MSISENSOR_PRO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../multiqc.html">MULTIQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../nanosim-h.html">NANOSIM-H</a></li>
<li class="toctree-l2"><a class="reference internal" href="../nextflow.html">NEXTFLOW</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ngs-disambiguate.html">NGS-DISAMBIGUATE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../open-cravat.html">OPEN-CRAVAT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../optitype.html">OPTITYPE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../paladin.html">PALADIN</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pandas.html">PANDAS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pandora.html">PANDORA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pbmm2.html">PBMM2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pcaExplorer.html">PCAEXPLORER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pear.html">PEAR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pfam.html">PFAM</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pheatmap.html">PHEATMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../picard.html">PICARD</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pindel.html">PINDEL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../plass.html">PLASS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../preseq.html">PRESEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../primerclip.html">PRIMERCLIP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../prosolo.html">PROSOLO</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ptrimmer.html">PTRIMMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pyfastaq.html">PYFASTAQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="../qualimap.html">QUALIMAP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rasusa.html">RASUSA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../razers3.html">RAZERS3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rbt.html">RBT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rebaler.html">REBALER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../reference.html">REFERENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../refgenie.html">REFGENIE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rest.html">REST</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rsem.html">RSEM</a></li>
<li class="toctree-l2"><a class="reference internal" href="../rubic.html">RUBIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../salmon.html">SALMON</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sambamba.html">SAMBAMBA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../samtools.html">SAMTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sansa.html">SANSA</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../seaborn.html">SEABORN</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="box-counts.html">SEABORN BOX PLOTS</a></li>
<li class="toctree-l3"><a class="reference internal" href="categorical_swarmplot.html">SEABORN CATEGORICAL SWARMPLOT</a></li>
<li class="toctree-l3"><a class="reference internal" href="catplot.html">CATPLOT</a></li>
<li class="toctree-l3"><a class="reference internal" href="clustermap.html">SEABORN CLUSTERMAP</a></li>
<li class="toctree-l3"><a class="reference internal" href="clustermap_genes.html">SEABORN CLUSTERMAP GENES</a></li>
<li class="toctree-l3"><a class="reference internal" href="displot.html">DISPLOT</a></li>
<li class="toctree-l3"><a class="reference internal" href="gene-count.html">SEABORN GENE BOX PLOTS</a></li>
<li class="toctree-l3"><a class="reference internal" href="kmeans.html">KMEANS</a></li>
<li class="toctree-l3"><a class="reference internal" href="paired_dotplots.html">PAIRGRID</a></li>
<li class="toctree-l3"><a class="reference internal" href="pairwise-scatterplot.html">PARIWISE SCATTERPLOT</a></li>
<li class="toctree-l3"><a class="reference internal" href="pca.html">PCA</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">PLOT_GENES</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l4"><a class="reference internal" href="#software-dependencies">Software dependencies</a></li>
<li class="toctree-l4"><a class="reference internal" href="#input-output">Input/Output</a></li>
<li class="toctree-l4"><a class="reference internal" href="#params">Params</a></li>
<li class="toctree-l4"><a class="reference internal" href="#authors">Authors</a></li>
<li class="toctree-l4"><a class="reference internal" href="#code">Code</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="plot_sigprofiler_single_sample.html">PLOT_SIGPROFILER_SINGLE_SAMPLE</a></li>
<li class="toctree-l3"><a class="reference internal" href="pval-histogram.html">P-VALUE HISTOGRAM</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../sed.html">SED</a></li>
<li class="toctree-l2"><a class="reference internal" href="../seqtk.html">SEQTK</a></li>
<li class="toctree-l2"><a class="reference internal" href="../shovill.html">SHOVILL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sickle.html">SICKLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sigprofiler.html">SIGPROFILER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../simplifyenrichment.html">SIMPLIFYENRICHMENT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sleuth.html">SLEUTH</a></li>
<li class="toctree-l2"><a class="reference internal" href="../snp-mutator.html">SNP-MUTATOR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../snpeff.html">SNPEFF</a></li>
<li class="toctree-l2"><a class="reference internal" href="../snpsift.html">SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sort.html">SORT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sourmash.html">SOURMASH</a></li>
<li class="toctree-l2"><a class="reference internal" href="../spades.html">SPADES</a></li>
<li class="toctree-l2"><a class="reference internal" href="../spliceai.html">SPLICE AI</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sra-tools.html">SRA-TOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../star.html">STAR</a></li>
<li class="toctree-l2"><a class="reference internal" href="../strelka.html">STRELKA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../strling.html">STRLING</a></li>
<li class="toctree-l2"><a class="reference internal" href="../subread.html">SUBREAD</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tabix.html">TABIX</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tmb.html">TMB</a></li>
<li class="toctree-l2"><a class="reference internal" href="../transdecoder.html">TRANSDECODER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../trim_galore.html">TRIM_GALORE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../trimmomatic.html">TRIMMOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../trinity.html">TRINITY</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tximeta.html">TXIMETA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tximport.html">TXIMPORT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ucsc.html">UCSC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../umis.html">UMIS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../unicycler.html">UNICYCLER</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vardict.html">VARDICT</a></li>
<li class="toctree-l2"><a class="reference internal" href="../variant_distribution.html">VARIANT_DISTRIBUTION</a></li>
<li class="toctree-l2"><a class="reference internal" href="../variantoccurence.html">VARIANTOCCURENCE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../varscan.html">VARSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vcftools.html">VCFTOOLS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vembrane.html">VEMBRANE</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vep.html">VEP</a></li>
<li class="toctree-l2"><a class="reference internal" href="../vg.html">VG</a></li>
<li class="toctree-l2"><a class="reference internal" href="../wgsim.html">WGSIM</a></li>
<li class="toctree-l2"><a class="reference internal" href="../wisecondorx.html">WISECONDORX</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../meta-wrappers.html">Meta-Wrappers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../../wrappers.html">Wrappers</a> &raquo;</li>
          <li><a href="../seaborn.html">SEABORN</a> &raquo;</li>
      <li>PLOT_GENES</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/wrappers/seaborn/plot_deseq_genes.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="plot-genes">
<span id="bio-seaborn-plot-deseq-genes"></span><h1>PLOT_GENES<a class="headerlink" href="#plot-genes" title="Permalink to this heading"></a></h1>
<p>Plot gene information (dispersions, expression, …) to help DESeq2 understanding</p>
<p><strong>URL</strong>:</p>
<section id="example">
<h2>Example<a class="headerlink" href="#example" title="Permalink to this heading"></a></h2>
<p>This wrapper can be used in the following way:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">gene_list</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;ENSMUSG00000000028&quot;</span><span class="p">,</span> <span class="s2">&quot;ENSMUSG00000000056&quot;</span><span class="p">,</span> <span class="s2">&quot;ENSMUSG00000118603&quot;</span><span class="p">,</span> <span class="s2">&quot;ENSMUSG00000002147&quot;</span><span class="p">]</span>

<span class="n">rule</span> <span class="n">test_plot_deseq_genes</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">deseq</span><span class="o">=</span><span class="s2">&quot;wald.tsv&quot;</span><span class="p">,</span>
        <span class="n">intermediar</span><span class="o">=</span><span class="s2">&quot;inter.tsv&quot;</span><span class="p">,</span>
        <span class="n">dst</span><span class="o">=</span><span class="s2">&quot;dst.tsv&quot;</span><span class="p">,</span>
        <span class="n">assays</span><span class="o">=</span><span class="s2">&quot;assays.tsv&quot;</span><span class="p">,</span>
        <span class="n">gene2gene</span><span class="o">=</span><span class="s2">&quot;/home/tdayris/Documents/Clusters/flamingo/scratch/bioinfo_core/B21074_FRPO_01/data_output/DGE_No_Batch_Without955/tximport/gene2gene.tsv&quot;</span><span class="p">,</span>
        <span class="n">metadata</span><span class="o">=</span><span class="s2">&quot;metadata.tsv&quot;</span><span class="p">,</span>
        <span class="n">filter_theta</span><span class="o">=</span><span class="s2">&quot;theta.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">log_counts</span><span class="o">=</span><span class="s2">&quot;log_counts.png&quot;</span><span class="p">,</span>
        <span class="n">log_mu</span><span class="o">=</span><span class="s2">&quot;log_mu.png&quot;</span><span class="p">,</span>
        <span class="c1"># maplot=&quot;MAplot.png&quot;,</span>
        <span class="n">gene_plots</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;geneplots/</span><span class="si">{gene}</span><span class="s2">.png&quot;</span><span class="p">,</span>
            <span class="n">gene</span><span class="o">=</span><span class="n">gene_list</span>
        <span class="p">),</span>
        <span class="n">independent_filtering</span><span class="o">=</span><span class="s2">&quot;independent_filter.png&quot;</span><span class="p">,</span>
        <span class="n">pval</span><span class="o">=</span><span class="s2">&quot;pval.png&quot;</span><span class="p">,</span>
        <span class="n">filter_theta</span><span class="o">=</span><span class="s2">&quot;theta.png&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">15</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">condition_dict</span><span class="o">=</span><span class="p">{</span>
            <span class="s2">&quot;s897&quot;</span><span class="p">:</span> <span class="s2">&quot;Control_Week2&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s929&quot;</span><span class="p">:</span> <span class="s2">&quot;Control_Week2&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s921&quot;</span><span class="p">:</span> <span class="s2">&quot;DOX_Week2&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s931&quot;</span><span class="p">:</span> <span class="s2">&quot;DOX_Week2&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s933&quot;</span><span class="p">:</span> <span class="s2">&quot;DOX_Week2&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s943&quot;</span><span class="p">:</span> <span class="s2">&quot;Control_Month5&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s948&quot;</span><span class="p">:</span> <span class="s2">&quot;Control_Month5&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s954&quot;</span><span class="p">:</span> <span class="s2">&quot;Control_Month5&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s967&quot;</span><span class="p">:</span> <span class="s2">&quot;DOX_Month5&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s971&quot;</span><span class="p">:</span> <span class="s2">&quot;DOX_Month5&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s974&quot;</span><span class="p">:</span> <span class="s2">&quot;DOX_Month5&quot;</span><span class="p">,</span>
            <span class="s2">&quot;s980&quot;</span><span class="p">:</span> <span class="s2">&quot;DOX_Month5&quot;</span>
        <span class="p">},</span>
        <span class="n">gene_list</span><span class="o">=</span><span class="n">gene_list</span><span class="p">,</span>
        <span class="n">gene_plots_prefix</span><span class="o">=</span><span class="s2">&quot;geneplots/&quot;</span><span class="p">,</span>
        <span class="n">comparison</span><span class="o">=</span><span class="s2">&quot;comparing_test_Month5_vs_ref_Week2&quot;</span><span class="p">,</span>
        <span class="n">chromosomes</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">24</span><span class="p">))</span> <span class="o">+</span> <span class="p">[</span><span class="s2">&quot;X&quot;</span><span class="p">,</span> <span class="s2">&quot;Y&quot;</span><span class="p">,</span> <span class="s2">&quot;MT&quot;</span><span class="p">]</span> <span class="o">+</span> <span class="nb">list</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">24</span><span class="p">)))</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/plot_deseq_genes.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;file:///home/tdayris/Documents/Clusters/flamingo/pipelines/snakemake-wrappers/bio/seaborn/plot_deseq_genes/&quot;</span>
</pre></div>
</div>
<p>Note that input, output and log file paths can be chosen freely.</p>
<p>When running with</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake --use-conda
</pre></div>
</div>
<p>the software dependencies will be automatically deployed into an isolated environment before execution.</p>
</section>
<section id="software-dependencies">
<h2>Software dependencies<a class="headerlink" href="#software-dependencies" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:python==3.9.6</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:pandas==1.3.1</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:numpy==1.21.1</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:matplotlib==3.4.2</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:pillow==8.3.1</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:zlib==1.2.11</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:pyqt==5.12.3</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:libxcb==1.13</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:xorg-libxcb==1.12</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:xorg-libxinerama==1.1.4</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge:seaborn==0.11.1</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">onda-forge:fastcluster==1.1.26</span></code></p></li>
</ul>
</section>
<section id="input-output">
<h2>Input/Output<a class="headerlink" href="#input-output" title="Permalink to this heading"></a></h2>
<p><strong>Input:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">intermediar</span></code>: Path to TSV formatted <cite>mcols(dds)</cite></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">wald</span></code>: Path to TSV formatted <cite>DESeq(dds)</cite></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">mu</span></code>: path to TSV formatted <cite>assays(dds)[[“mu”]]</cite></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">cooks</span></code>: path to TSV formatted <cite>assays(dds)[[“cooks”]]</cite></p></li>
</ul>
<p><strong>Output:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">expression</span></code>: Path to PNG expression plot</p></li>
</ul>
</section>
<section id="params">
<h2>Params<a class="headerlink" href="#params" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">condition_dict</span></code>: This will be used to define color scheme and categories.</p></li>
</ul>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
<section id="code">
<h2>Code<a class="headerlink" href="#code" title="Permalink to this heading"></a></h2>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/usr/bin/env python3</span>
<span class="c1"># -*- coding: utf-8 -*-</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">Draw several QC plots for DESeq2</span>
<span class="sd">&quot;&quot;&quot;</span>

<span class="kn">import</span> <span class="nn">logging</span>
<span class="kn">import</span> <span class="nn">matplotlib</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span>
<span class="kn">import</span> <span class="nn">numpy</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pandas</span>
<span class="kn">import</span> <span class="nn">seaborn</span>

<span class="kn">from</span> <span class="nn">typing</span> <span class="kn">import</span> <span class="n">Any</span><span class="p">,</span> <span class="n">Dict</span><span class="p">,</span> <span class="n">List</span><span class="p">,</span> <span class="n">Optional</span>

<span class="n">logging</span><span class="o">.</span><span class="n">basicConfig</span><span class="p">(</span>
    <span class="n">filename</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">log</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
    <span class="n">filemode</span><span class="o">=</span><span class="s2">&quot;w&quot;</span><span class="p">,</span>
    <span class="n">level</span><span class="o">=</span><span class="n">logging</span><span class="o">.</span><span class="n">DEBUG</span>
<span class="p">)</span>

<span class="n">logging</span><span class="o">.</span><span class="n">getLogger</span><span class="p">(</span><span class="s1">&#39;matplotlib.font_manager&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">disabled</span> <span class="o">=</span> <span class="kc">True</span>

<span class="k">def</span> <span class="nf">read_deseq</span><span class="p">(</span><span class="n">path</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">:</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Loading </span><span class="si">{</span><span class="n">path</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">tmp</span> <span class="o">=</span> <span class="n">pandas</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span>
        <span class="n">path</span><span class="p">,</span>
        <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
        <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span>
    <span class="p">)</span>
    <span class="n">tmp</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;Ensembl_ID&quot;</span>
    <span class="k">return</span> <span class="n">tmp</span>

<span class="k">def</span> <span class="nf">read_gene2gene</span><span class="p">(</span><span class="n">path</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">:</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Loading </span><span class="si">{</span><span class="n">path</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">tmp</span> <span class="o">=</span> <span class="n">pandas</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span>
        <span class="n">path</span><span class="p">,</span>
        <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
        <span class="n">index_col</span><span class="o">=</span><span class="mi">3</span>
    <span class="p">)</span>
    <span class="n">tmp</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;Ensembl_ID&quot;</span>
    <span class="k">return</span> <span class="n">tmp</span>


<span class="k">def</span> <span class="nf">load_all_data</span><span class="p">(</span><span class="n">deseq_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                  <span class="n">intermediar_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                  <span class="n">assays_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                  <span class="n">dst_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                  <span class="n">filter_theta_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                  <span class="n">metadata_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                  <span class="n">gene2gene_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                  <span class="n">padj_threshold</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mf">0.05</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">List</span><span class="p">[</span><span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">]:</span>
    <span class="sd">&quot;&quot;&quot;Load table in memory&quot;&quot;&quot;</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Starting loading data in memory...&quot;</span><span class="p">)</span>
    <span class="n">deseq</span> <span class="o">=</span> <span class="n">pandas</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span>
        <span class="n">left</span><span class="o">=</span><span class="n">read_deseq</span><span class="p">(</span><span class="n">deseq_path</span><span class="p">),</span>
        <span class="n">right</span><span class="o">=</span><span class="n">read_deseq</span><span class="p">(</span><span class="n">intermediar_path</span><span class="p">),</span>
        <span class="n">left_index</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
        <span class="n">right_index</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
        <span class="n">how</span><span class="o">=</span><span class="s2">&quot;outer&quot;</span><span class="p">,</span>
        <span class="n">validate</span><span class="o">=</span><span class="s2">&quot;1:1&quot;</span><span class="p">,</span>
        <span class="n">suffixes</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;_results&quot;</span><span class="p">,</span> <span class="s2">&quot;_mcols&quot;</span><span class="p">]</span>
    <span class="p">)</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Loading </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">gene2gene_path</span><span class="p">)</span>
    <span class="n">deseq</span> <span class="o">=</span> <span class="n">pandas</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span>
        <span class="n">left</span><span class="o">=</span><span class="n">deseq</span><span class="p">,</span>
        <span class="n">right</span><span class="o">=</span><span class="n">read_gene2gene</span><span class="p">(</span><span class="n">gene2gene_path</span><span class="p">),</span>
        <span class="n">left_index</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
        <span class="n">right_index</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
        <span class="n">how</span><span class="o">=</span><span class="s2">&quot;outer&quot;</span><span class="p">,</span>
        <span class="n">validate</span><span class="o">=</span><span class="s2">&quot;1:1&quot;</span><span class="p">,</span>
    <span class="p">)</span>
    <span class="n">deseq</span><span class="p">[</span><span class="s2">&quot;Status&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
        <span class="s2">&quot;Differentially Expressed&quot;</span>
        <span class="k">if</span> <span class="n">padj</span> <span class="o">&lt;=</span> <span class="n">padj_threshold</span> <span class="k">else</span> <span class="s2">&quot;Not Significative&quot;</span>
        <span class="k">for</span> <span class="n">padj</span> <span class="ow">in</span> <span class="n">deseq</span><span class="p">[</span><span class="s2">&quot;padj&quot;</span><span class="p">]</span>
    <span class="p">]</span>
    <span class="n">deseq</span><span class="p">[</span><span class="s2">&quot;log2FC_eq_lfcSE&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
        <span class="p">(</span><span class="n">fc</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="ow">is</span> <span class="p">(</span><span class="n">shr</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">fc</span><span class="p">,</span> <span class="n">shr</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">deseq</span><span class="p">[</span><span class="s2">&quot;log2FoldChange&quot;</span><span class="p">],</span> <span class="n">deseq</span><span class="p">[</span><span class="s2">&quot;lfcSE&quot;</span><span class="p">])</span>
    <span class="p">]</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">deseq</span><span class="o">.</span><span class="n">columns</span><span class="p">)</span>

    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Loading counts: </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">dst_path</span><span class="p">)</span>
    <span class="n">counts</span> <span class="o">=</span> <span class="n">read_deseq</span><span class="p">(</span><span class="n">dst_path</span><span class="p">)</span>

    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Loading assays: </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">assays_path</span><span class="p">)</span>
    <span class="n">mu</span> <span class="o">=</span> <span class="n">read_deseq</span><span class="p">(</span><span class="n">assays_path</span><span class="p">)</span>

    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Loading metadata: </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">metadata_path</span><span class="p">)</span>
    <span class="n">metadata</span> <span class="o">=</span> <span class="n">read_deseq</span><span class="p">(</span><span class="n">metadata_path</span><span class="p">)</span>

    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Loading theta filter data: </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">filter_theta_path</span><span class="p">)</span>
    <span class="n">theta</span> <span class="o">=</span> <span class="n">read_deseq</span><span class="p">(</span><span class="n">filter_theta_path</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">deseq</span><span class="p">,</span> <span class="n">counts</span><span class="p">,</span> <span class="n">mu</span><span class="p">,</span> <span class="n">theta</span><span class="p">,</span> <span class="n">metadata</span>


<span class="k">def</span> <span class="nf">plot_counts</span><span class="p">(</span><span class="n">dst</span><span class="p">:</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
                <span class="n">colors</span><span class="p">:</span> <span class="n">Dict</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="nb">str</span><span class="p">],</span>
                <span class="n">png_out</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                <span class="n">is_mu</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Boxplot of log10 counts of each sample,</span>
<span class="sd">    colored by conditions</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">count</span> <span class="o">=</span> <span class="s2">&quot;Counts&quot;</span> <span class="k">if</span> <span class="n">is_mu</span> <span class="ow">is</span> <span class="kc">False</span> <span class="k">else</span> <span class="s2">&quot;µ&quot;</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Plotting log10 </span><span class="si">{</span><span class="n">count</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">box</span> <span class="o">=</span> <span class="n">dst</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="o">~</span><span class="p">(</span><span class="n">dst</span><span class="o">==</span><span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">all</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)]</span>
    <span class="n">box</span> <span class="o">=</span> <span class="n">box</span><span class="o">.</span><span class="n">melt</span><span class="p">(</span>
        <span class="n">ignore_index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="c1"># Counts are NOT log10, but they will be in final</span>
        <span class="c1"># graph !</span>
        <span class="n">value_name</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;log10(Normalised </span><span class="si">{</span><span class="n">count</span><span class="si">}</span><span class="s2">)&quot;</span><span class="p">,</span>
        <span class="n">var_name</span><span class="o">=</span><span class="s2">&quot;Sample ID&quot;</span>
    <span class="p">)</span>

    <span class="c1"># g, ax = matplotlib.pyplot.subplots(1, 1)</span>
    <span class="n">g</span> <span class="o">=</span> <span class="n">seaborn</span><span class="o">.</span><span class="n">catplot</span><span class="p">(</span>
        <span class="c1"># Counts are NOT log10, but they will be in final</span>
        <span class="c1"># graph !</span>
        <span class="n">x</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;log10(Normalised </span><span class="si">{</span><span class="n">count</span><span class="si">}</span><span class="s2">)&quot;</span><span class="p">,</span>
        <span class="n">y</span><span class="o">=</span><span class="s2">&quot;Sample ID&quot;</span><span class="p">,</span>
        <span class="n">data</span><span class="o">=</span><span class="n">box</span><span class="p">,</span>
        <span class="n">palette</span><span class="o">=</span><span class="n">colors</span><span class="p">,</span>
        <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;box&quot;</span><span class="p">,</span>
        <span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
        <span class="n">legend_out</span><span class="o">=</span><span class="kc">True</span>
    <span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">xscale</span><span class="o">=</span><span class="s2">&quot;log&quot;</span><span class="p">)</span>

    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Counts plot saved to </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">png_out</span><span class="p">)</span>
    <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span>
        <span class="n">png_out</span><span class="p">,</span>
        <span class="n">bbox_inches</span><span class="o">=</span><span class="s2">&quot;tight&quot;</span>
    <span class="p">)</span>


<span class="k">def</span> <span class="nf">plot_single_gene</span><span class="p">(</span><span class="n">dst</span><span class="p">:</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
                     <span class="n">mu</span><span class="p">:</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
                     <span class="n">deseq</span><span class="p">:</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
                     <span class="n">condition_dict</span><span class="p">:</span> <span class="n">Dict</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="nb">str</span><span class="p">],</span>
                     <span class="n">genes</span><span class="p">:</span> <span class="n">List</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
                     <span class="n">png_prefix</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                     <span class="n">comparison</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Plot gene expression, its geometric mean and dispersion value</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">dstbox</span> <span class="o">=</span> <span class="n">dst</span><span class="o">.</span><span class="n">melt</span><span class="p">(</span>
        <span class="n">ignore_index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="n">value_name</span><span class="o">=</span><span class="s2">&quot;Normalised Counts&quot;</span><span class="p">,</span>
        <span class="n">var_name</span><span class="o">=</span><span class="s2">&quot;Sample ID&quot;</span>
    <span class="p">)</span>
    <span class="n">dstbox</span><span class="p">[</span><span class="s2">&quot;Condition&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">condition_dict</span><span class="p">[</span><span class="n">s</span><span class="p">]</span> <span class="k">for</span> <span class="n">s</span> <span class="ow">in</span> <span class="n">dstbox</span><span class="p">[</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">]]</span>

    <span class="n">mubox</span> <span class="o">=</span> <span class="n">mu</span><span class="o">.</span><span class="n">melt</span><span class="p">(</span>
        <span class="n">ignore_index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="n">value_name</span><span class="o">=</span><span class="s2">&quot;µ&quot;</span><span class="p">,</span>
        <span class="n">var_name</span><span class="o">=</span><span class="s2">&quot;Sample ID&quot;</span>
    <span class="p">)</span>
    <span class="n">mubox</span><span class="p">[</span><span class="s2">&quot;Condition&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">condition_dict</span><span class="p">[</span><span class="n">s</span><span class="p">]</span> <span class="k">for</span> <span class="n">s</span> <span class="ow">in</span> <span class="n">mubox</span><span class="p">[</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">]]</span>

    <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">genes</span><span class="p">:</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Plotting gene information on </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">gene</span><span class="p">)</span>
        <span class="n">gene_count</span> <span class="o">=</span> <span class="n">dstbox</span><span class="p">[</span><span class="n">dstbox</span><span class="o">.</span><span class="n">index</span> <span class="o">==</span> <span class="n">gene</span><span class="p">]</span>
        <span class="n">gene_mu</span> <span class="o">=</span> <span class="n">mubox</span><span class="p">[</span><span class="n">mubox</span><span class="o">.</span><span class="n">index</span> <span class="o">==</span> <span class="n">gene</span><span class="p">]</span>
        <span class="n">gene_dge</span> <span class="o">=</span> <span class="n">deseq</span><span class="p">[</span><span class="n">deseq</span><span class="o">.</span><span class="n">index</span> <span class="o">==</span> <span class="n">gene</span><span class="p">]</span>
        <span class="n">status</span> <span class="o">=</span> <span class="n">gene_dge</span><span class="p">[</span><span class="s2">&quot;Status&quot;</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>

        <span class="n">padj</span> <span class="o">=</span> <span class="nb">round</span><span class="p">(</span><span class="n">gene_dge</span><span class="p">[</span><span class="s2">&quot;padj&quot;</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span> <span class="mi">4</span><span class="p">)</span>

        <span class="k">try</span><span class="p">:</span>
            <span class="n">gene_name</span> <span class="o">=</span> <span class="n">gene_dge</span><span class="p">[</span><span class="s2">&quot;Gene_Name&quot;</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>
            <span class="n">name</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">gene_name</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="n">gene</span><span class="si">}</span><span class="s2">)&quot;</span>
            <span class="n">png_out</span> <span class="o">=</span> <span class="n">png_prefix</span> <span class="o">+</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">gene_name</span><span class="si">}</span><span class="s2">.</span><span class="si">{</span><span class="n">gene</span><span class="si">}</span><span class="s2">.png&quot;</span>
        <span class="k">except</span> <span class="ne">KeyError</span><span class="p">:</span>
            <span class="n">name</span> <span class="o">=</span> <span class="n">gene</span>
            <span class="n">gene_name</span> <span class="o">=</span> <span class="n">gene</span>
            <span class="n">png_out</span> <span class="o">=</span> <span class="n">png_prefix</span> <span class="o">+</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">gene</span><span class="si">}</span><span class="s2">.png&quot;</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Plotting gene information on </span><span class="si">{</span><span class="n">name</span><span class="si">}</span><span class="s2"> to </span><span class="si">{</span><span class="n">png_out</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

        <span class="n">fig</span> <span class="o">=</span> <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span><span class="mi">10</span><span class="p">))</span>

        <span class="n">axes</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="n">axes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">1</span><span class="p">))</span>
        <span class="n">axes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">sharey</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
        <span class="n">axes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="n">sharey</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
        <span class="n">axes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

        <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span>
            <span class="sa">f</span><span class="s2">&quot;Expression of </span><span class="si">{</span><span class="n">name</span><span class="si">}</span><span class="s2"> among samples</span><span class="se">\n</span><span class="s2">&quot;</span>
            <span class="s2">&quot;Differential gene expression on &quot;</span>
            <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">comparison</span><span class="si">}</span><span class="s2"> </span><span class="si">{</span><span class="n">status</span><span class="si">}</span><span class="s2"> with P-Adj of </span><span class="si">{</span><span class="n">padj</span><span class="si">}</span><span class="s2">&quot;</span>
        <span class="p">)</span>
        <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Gene VST accross conditions&quot;</span><span class="p">)</span>
        <span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Gene µ accross conditions&quot;</span><span class="p">)</span>
        <span class="n">axes</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Geometric Mean and dispersion&quot;</span><span class="p">)</span>
        <span class="n">axes</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Expression change with standard Error&quot;</span><span class="p">)</span>
        <span class="n">seaborn</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span>
            <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
            <span class="n">y</span><span class="o">=</span><span class="s2">&quot;Normalised Counts&quot;</span><span class="p">,</span>
            <span class="n">x</span><span class="o">=</span><span class="s2">&quot;Condition&quot;</span><span class="p">,</span>
            <span class="n">data</span><span class="o">=</span><span class="n">gene_count</span>
        <span class="p">)</span>
        <span class="n">seaborn</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span>
            <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span>
            <span class="n">y</span><span class="o">=</span><span class="s2">&quot;µ&quot;</span><span class="p">,</span>
            <span class="n">x</span><span class="o">=</span><span class="s2">&quot;Condition&quot;</span><span class="p">,</span>
            <span class="n">data</span><span class="o">=</span><span class="n">gene_mu</span>
        <span class="p">)</span>
        <span class="n">axes</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">bar</span><span class="p">(</span>
            <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span>
            <span class="n">height</span><span class="o">=</span><span class="n">gene_dge</span><span class="p">[</span><span class="s2">&quot;baseMean_results&quot;</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span>
            <span class="n">yerr</span><span class="o">=</span><span class="n">gene_dge</span><span class="p">[</span><span class="s2">&quot;baseMean_results&quot;</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="o">*</span> <span class="n">gene_dge</span><span class="p">[</span><span class="s2">&quot;dispersion&quot;</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span>
            <span class="n">tick_label</span><span class="o">=</span><span class="n">gene</span>
        <span class="p">)</span>
        <span class="n">axes</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span><span class="o">.</span><span class="n">bar</span><span class="p">(</span>
            <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span>
            <span class="n">height</span><span class="o">=</span><span class="n">gene_dge</span><span class="p">[</span><span class="s2">&quot;log2FoldChange&quot;</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span>
            <span class="n">tick_label</span><span class="o">=</span><span class="s2">&quot;Log2(FoldChange)&quot;</span><span class="p">,</span>
            <span class="n">yerr</span><span class="o">=</span><span class="n">gene_dge</span><span class="p">[</span><span class="s2">&quot;lfcSE&quot;</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>
        <span class="p">)</span>

        <span class="c1"># Rotate condition texts</span>
        <span class="k">for</span> <span class="n">ax</span> <span class="ow">in</span> <span class="n">axes</span><span class="p">:</span>
            <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">sca</span><span class="p">(</span><span class="n">ax</span><span class="p">)</span>
            <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">xticks</span><span class="p">(</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">)</span>

        <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Gene plot saved to </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">png_out</span><span class="p">)</span>
        <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span>
            <span class="n">png_out</span><span class="p">,</span>
            <span class="n">bbox_inches</span><span class="o">=</span><span class="s2">&quot;tight&quot;</span>
        <span class="p">)</span>


<span class="k">def</span> <span class="nf">independent_filtering</span><span class="p">(</span><span class="n">deseq</span><span class="p">:</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
                          <span class="n">comparison</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                          <span class="n">png_out</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
    <span class="sd">&quot;&quot;&quot;Plot number of observations filtered by independent filtering method&quot;&quot;&quot;</span>
    <span class="n">tmp</span> <span class="o">=</span> <span class="n">deseq</span><span class="p">[</span>
        <span class="p">[</span><span class="s2">&quot;padj&quot;</span><span class="p">,</span> <span class="s2">&quot;Status&quot;</span><span class="p">,</span> <span class="s2">&quot;filterThreshold&quot;</span><span class="p">,</span> <span class="s2">&quot;baseMean_results&quot;</span><span class="p">,</span> <span class="s2">&quot;maxCooks&quot;</span><span class="p">]</span>
    <span class="p">]</span>
    <span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;padj&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="o">-</span><span class="n">numpy</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;padj&quot;</span><span class="p">])</span>
    <span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;log_baseMean_results&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">numpy</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;baseMean_results&quot;</span><span class="p">])</span>
    <span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;filterThreshold&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
        <span class="s2">&quot;Pass&quot;</span> <span class="k">if</span> <span class="nb">filter</span> <span class="ow">is</span> <span class="kc">True</span> <span class="k">else</span> <span class="s2">&quot;Do not pass&quot;</span>
        <span class="k">for</span> <span class="nb">filter</span> <span class="ow">in</span> <span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;filterThreshold&quot;</span><span class="p">]</span>
    <span class="p">]</span>
    <span class="n">tmp</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span>
        <span class="s2">&quot;-Log10(P-Adjusted)&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Status&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Independent Filtering&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Mean Expression&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Maximum Cook&#39;s distance&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Log10(Mean Expression)&quot;</span><span class="p">,</span>
    <span class="p">]</span>

    <span class="c1"># Build plot</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span>
        <span class="mi">1</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span><span class="mi">10</span><span class="p">)</span><span class="c1">#, sharey=True</span>
    <span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Independent filtering, </span><span class="si">{</span><span class="n">comparison</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;# observations filtered out&quot;</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Relation between expression and adjusted pvalue&quot;</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Relation between expression and cooks distance&quot;</span><span class="p">)</span>

    <span class="c1"># seaborn.countplot(</span>
    <span class="c1">#     ax=axes[0],</span>
    <span class="c1">#     x=&quot;Status&quot;,</span>
    <span class="c1">#     data=tmp</span>
    <span class="c1"># )</span>
    <span class="n">seaborn</span><span class="o">.</span><span class="n">histplot</span><span class="p">(</span>
        <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
        <span class="n">x</span><span class="o">=</span><span class="s2">&quot;Mean Expression&quot;</span><span class="p">,</span>
        <span class="n">data</span><span class="o">=</span><span class="n">tmp</span><span class="p">,</span>
        <span class="n">bins</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span>
        <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;Independent Filtering&quot;</span><span class="p">,</span>
        <span class="n">multiple</span><span class="o">=</span><span class="s2">&quot;stack&quot;</span>
    <span class="p">)</span>
    <span class="n">seaborn</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span>
        <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span>
        <span class="n">data</span><span class="o">=</span><span class="n">tmp</span><span class="p">,</span>
        <span class="n">y</span><span class="o">=</span><span class="s2">&quot;-Log10(P-Adjusted)&quot;</span><span class="p">,</span>
        <span class="n">x</span><span class="o">=</span><span class="s2">&quot;Log10(Mean Expression)&quot;</span><span class="p">,</span>
        <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;Independent Filtering&quot;</span><span class="p">,</span>
        <span class="n">style</span><span class="o">=</span><span class="s2">&quot;Status&quot;</span>
    <span class="p">)</span>
    <span class="n">seaborn</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span>
        <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">2</span><span class="p">],</span>
        <span class="n">data</span><span class="o">=</span><span class="n">tmp</span><span class="p">,</span>
        <span class="n">y</span><span class="o">=</span><span class="s2">&quot;Maximum Cook&#39;s distance&quot;</span><span class="p">,</span>
        <span class="n">x</span><span class="o">=</span><span class="s2">&quot;Log10(Mean Expression)&quot;</span><span class="p">,</span>
        <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;Independent Filtering&quot;</span><span class="p">,</span>
        <span class="n">style</span><span class="o">=</span><span class="s2">&quot;Status&quot;</span>
    <span class="p">)</span>

    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Independent filtering results saved to </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">png_out</span><span class="p">)</span>
    <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span>
        <span class="n">png_out</span><span class="p">,</span>
        <span class="n">bbox_inches</span><span class="o">=</span><span class="s2">&quot;tight&quot;</span>
    <span class="p">)</span>

    <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>


<span class="k">def</span> <span class="nf">pval_distribution</span><span class="p">(</span><span class="n">deseq</span><span class="p">:</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
                      <span class="n">comparison</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                      <span class="n">png_out</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
                      <span class="n">chromosomes</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="n">List</span><span class="p">[</span><span class="nb">str</span><span class="p">]]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
    <span class="sd">&quot;&quot;&quot;Plot number of differentially expressed genes and their Chromosomes&quot;&quot;&quot;</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">deseq</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
    <span class="n">tmp</span> <span class="o">=</span> <span class="n">deseq</span><span class="p">[</span>
        <span class="p">[</span><span class="s2">&quot;padj&quot;</span><span class="p">,</span> <span class="s2">&quot;Status&quot;</span><span class="p">,</span> <span class="s2">&quot;filterThreshold&quot;</span><span class="p">,</span> <span class="s2">&quot;Chromosome&quot;</span><span class="p">]</span>
    <span class="p">]</span>
    <span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;padj&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="o">-</span><span class="n">numpy</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;padj&quot;</span><span class="p">])</span>
    <span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;filterThreshold&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
        <span class="s2">&quot;Pass&quot;</span> <span class="k">if</span> <span class="nb">filter</span> <span class="ow">is</span> <span class="kc">True</span> <span class="k">else</span> <span class="s2">&quot;Do not pass&quot;</span>
        <span class="k">for</span> <span class="nb">filter</span> <span class="ow">in</span> <span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;filterThreshold&quot;</span><span class="p">]</span>
    <span class="p">]</span>

    <span class="k">if</span> <span class="n">chromosomes</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">tmp</span> <span class="o">=</span> <span class="n">tmp</span><span class="p">[</span><span class="n">tmp</span><span class="p">[</span><span class="s2">&quot;Chromosome&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">chromosomes</span><span class="p">)]</span>

    <span class="n">tmp</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s2">&quot;Chromosome&quot;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

    <span class="n">tmp</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span>
        <span class="s2">&quot;-Log10(P-Adjusted)&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Status&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Independent Filtering&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Chromosomes&quot;</span>
    <span class="p">]</span>

    <span class="c1"># Build plot</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span>
        <span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span><span class="mi">10</span><span class="p">)</span><span class="c1">#, sharey=True</span>
    <span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Differential gene expression </span><span class="si">{</span><span class="n">comparison</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;# differentially expressed genes&quot;</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;# differentially expressed genes per chromosomes&quot;</span><span class="p">)</span>

    <span class="n">seaborn</span><span class="o">.</span><span class="n">countplot</span><span class="p">(</span>
        <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
        <span class="n">x</span><span class="o">=</span><span class="s2">&quot;Status&quot;</span><span class="p">,</span>
        <span class="n">data</span><span class="o">=</span><span class="n">tmp</span>
    <span class="p">)</span>

    <span class="n">seaborn</span><span class="o">.</span><span class="n">countplot</span><span class="p">(</span>
        <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span>
        <span class="n">x</span><span class="o">=</span><span class="s2">&quot;Status&quot;</span><span class="p">,</span>
        <span class="n">data</span><span class="o">=</span><span class="n">tmp</span><span class="p">,</span>
        <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;Chromosomes&quot;</span>
    <span class="p">)</span>

    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;P-Value distribution saved to </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">png_out</span><span class="p">)</span>
    <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span>
        <span class="n">png_out</span><span class="p">,</span>
        <span class="n">bbox_inches</span><span class="o">=</span><span class="s2">&quot;tight&quot;</span>
    <span class="p">)</span>
    <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>


<span class="k">def</span> <span class="nf">filter_theta</span><span class="p">(</span><span class="n">theta</span><span class="p">:</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
                 <span class="n">metadata</span><span class="p">:</span> <span class="n">pandas</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
                 <span class="n">png_out</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
    <span class="sd">&quot;&quot;&quot;Plot number of rejected observations due to theta filer&quot;&quot;&quot;</span>
    <span class="n">theta</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;Quantiles of filter&quot;</span><span class="p">,</span> <span class="s2">&quot;# of rejections&quot;</span><span class="p">]</span>
    <span class="n">fig</span> <span class="o">=</span> <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">10</span><span class="p">))</span>
    <span class="n">seaborn</span><span class="o">.</span><span class="n">lineplot</span><span class="p">(</span>
        <span class="n">data</span><span class="o">=</span><span class="n">theta</span><span class="p">,</span>
        <span class="n">x</span><span class="o">=</span><span class="s2">&quot;Quantiles of filter&quot;</span><span class="p">,</span>
        <span class="n">y</span><span class="o">=</span><span class="s2">&quot;# of rejections&quot;</span>
    <span class="p">)</span>
    <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span>
        <span class="n">x</span><span class="o">=</span><span class="n">metadata</span><span class="p">[</span><span class="s2">&quot;filterTheta&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span>
        <span class="n">color</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span>
        <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;--&quot;</span>
    <span class="p">)</span>

    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Theta filter saved to </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">png_out</span><span class="p">)</span>
    <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span>
        <span class="n">png_out</span><span class="p">,</span>
        <span class="n">bbox_inches</span><span class="o">=</span><span class="s2">&quot;tight&quot;</span>
    <span class="p">)</span>

    <span class="n">matplotlib</span><span class="o">.</span><span class="n">pyplot</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>


<span class="c1"># Setting style and colors</span>
<span class="n">seaborn</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">)</span>
<span class="n">condition_list</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;condition_dict&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">())</span>
<span class="n">colors</span> <span class="o">=</span> <span class="n">seaborn</span><span class="o">.</span><span class="n">husl_palette</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">condition_list</span><span class="p">),</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.45</span><span class="p">)</span>
<span class="n">cond_colors</span> <span class="o">=</span> <span class="p">{</span>
    <span class="nb">str</span><span class="p">(</span><span class="n">cond</span><span class="p">):</span> <span class="n">color</span> <span class="k">for</span> <span class="n">cond</span><span class="p">,</span> <span class="n">color</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">condition_list</span><span class="p">,</span> <span class="nb">list</span><span class="p">(</span><span class="n">colors</span><span class="p">))</span>
<span class="p">}</span>
<span class="n">sample_colors</span> <span class="o">=</span> <span class="p">{</span>
    <span class="n">s</span><span class="p">:</span> <span class="n">cond_colors</span><span class="p">[</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;condition_dict&quot;</span><span class="p">][</span><span class="n">s</span><span class="p">]]</span>
    <span class="k">for</span> <span class="n">s</span> <span class="ow">in</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;condition_dict&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">keys</span><span class="p">()</span>
<span class="p">}</span>

<span class="c1"># Loading data</span>
<span class="n">deseq</span><span class="p">,</span> <span class="n">counts</span><span class="p">,</span> <span class="n">mu</span><span class="p">,</span> <span class="n">theta</span><span class="p">,</span> <span class="n">metadata</span> <span class="o">=</span> <span class="n">load_all_data</span><span class="p">(</span>
    <span class="n">deseq_path</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="p">[</span><span class="s2">&quot;deseq&quot;</span><span class="p">],</span>
    <span class="n">intermediar_path</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="p">[</span><span class="s2">&quot;intermediar&quot;</span><span class="p">],</span>
    <span class="n">dst_path</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="p">[</span><span class="s2">&quot;dst&quot;</span><span class="p">],</span>
    <span class="n">assays_path</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="p">[</span><span class="s2">&quot;assays&quot;</span><span class="p">],</span>
    <span class="n">filter_theta_path</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="p">[</span><span class="s2">&quot;filter_theta&quot;</span><span class="p">],</span>
    <span class="n">metadata_path</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="p">[</span><span class="s2">&quot;metadata&quot;</span><span class="p">],</span>
    <span class="n">gene2gene_path</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="p">[</span><span class="s2">&quot;gene2gene&quot;</span><span class="p">],</span>
    <span class="n">padj_threshold</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;padj_threshold&quot;</span><span class="p">,</span> <span class="mf">0.05</span><span class="p">)</span>
<span class="p">)</span>

<span class="c1"># Plotting</span>
<span class="k">if</span> <span class="s2">&quot;log_counts&quot;</span> <span class="ow">in</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Log counts plot...&quot;</span><span class="p">)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">plot_counts</span><span class="p">(</span>
            <span class="n">dst</span><span class="o">=</span><span class="n">counts</span><span class="p">,</span>
            <span class="n">colors</span><span class="o">=</span><span class="n">sample_colors</span><span class="p">,</span>
            <span class="n">png_out</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;log_counts&quot;</span><span class="p">]</span>
        <span class="p">)</span>
    <span class="k">except</span> <span class="n">Error</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">error</span><span class="p">(</span><span class="n">e</span><span class="p">)</span>
        <span class="k">raise</span>

<span class="k">if</span> <span class="s2">&quot;log_mu&quot;</span> <span class="ow">in</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Log µ count plot ...&quot;</span><span class="p">)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">plot_counts</span><span class="p">(</span>
            <span class="n">dst</span><span class="o">=</span><span class="n">mu</span><span class="p">,</span>
            <span class="n">colors</span><span class="o">=</span><span class="n">sample_colors</span><span class="p">,</span>
            <span class="n">png_out</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;log_mu&quot;</span><span class="p">],</span>
            <span class="n">is_mu</span><span class="o">=</span><span class="kc">True</span>
        <span class="p">)</span>
    <span class="k">except</span> <span class="n">Error</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">error</span><span class="p">(</span><span class="n">e</span><span class="p">)</span>
        <span class="k">raise</span>

<span class="k">if</span> <span class="s2">&quot;gene_plots&quot;</span> <span class="ow">in</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Performing gene plots ...&quot;</span><span class="p">)</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;gene_plots&quot;</span><span class="p">]):</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span>
            <span class="s2">&quot;Building output directory: </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span>
            <span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;gene_plots&quot;</span><span class="p">]</span>
        <span class="p">)</span>
        <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;gene_plots&quot;</span><span class="p">]))</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span>
            <span class="s2">&quot;Output directory &#39;</span><span class="si">%s</span><span class="s2">&#39; already exists &quot;</span><span class="p">,</span>
            <span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;gene_plots&quot;</span><span class="p">]</span>
        <span class="p">)</span>

    <span class="k">try</span><span class="p">:</span>
        <span class="n">gene_list</span> <span class="o">=</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">wildcards</span><span class="p">[</span><span class="s2">&quot;gene&quot;</span><span class="p">]</span>
    <span class="k">except</span> <span class="ne">AttributeError</span><span class="p">:</span>
        <span class="n">gene_list</span> <span class="o">=</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;gene_list&quot;</span><span class="p">]</span>

    <span class="k">try</span><span class="p">:</span>
        <span class="n">plot_single_gene</span><span class="p">(</span>
            <span class="n">dst</span><span class="o">=</span><span class="n">counts</span><span class="p">,</span>
            <span class="n">deseq</span><span class="o">=</span><span class="n">deseq</span><span class="p">,</span>
            <span class="n">mu</span><span class="o">=</span><span class="n">mu</span><span class="p">,</span>
            <span class="n">condition_dict</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;condition_dict&quot;</span><span class="p">],</span>
            <span class="n">genes</span><span class="o">=</span><span class="n">gene_list</span><span class="p">,</span>
            <span class="n">png_prefix</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;gene_plots_prefix&quot;</span><span class="p">],</span>
            <span class="n">comparison</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;comparison&quot;</span><span class="p">]</span>
        <span class="p">)</span>
    <span class="k">except</span> <span class="n">Error</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">error</span><span class="p">(</span><span class="n">e</span><span class="p">)</span>
        <span class="k">raise</span>

<span class="k">if</span> <span class="s2">&quot;pval&quot;</span> <span class="ow">in</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;P-Value distribution plot ...&quot;</span><span class="p">)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">pval_distribution</span><span class="p">(</span>
            <span class="n">deseq</span><span class="o">=</span><span class="n">deseq</span><span class="p">,</span>
            <span class="n">comparison</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;comparison&quot;</span><span class="p">],</span>
            <span class="n">png_out</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;pval&quot;</span><span class="p">],</span>
            <span class="n">chromosomes</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;chromosomes&quot;</span><span class="p">,</span> <span class="kc">None</span><span class="p">)</span>
        <span class="p">)</span>
    <span class="k">except</span> <span class="n">Error</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">error</span><span class="p">(</span><span class="n">e</span><span class="p">)</span>
        <span class="k">raise</span>

<span class="k">if</span> <span class="s2">&quot;independent_filtering&quot;</span> <span class="ow">in</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Independent filtering plot ...&quot;</span><span class="p">)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">independent_filtering</span><span class="p">(</span>
            <span class="n">deseq</span><span class="o">=</span><span class="n">deseq</span><span class="p">,</span>
            <span class="n">comparison</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;comparison&quot;</span><span class="p">],</span>
            <span class="n">png_out</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;independent_filtering&quot;</span><span class="p">]</span>
        <span class="p">)</span>
    <span class="k">except</span> <span class="n">Error</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">error</span><span class="p">(</span><span class="n">e</span><span class="p">)</span>
        <span class="k">raise</span>

<span class="k">if</span> <span class="s2">&quot;filter_theta&quot;</span> <span class="ow">in</span> <span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Theta filter plot ...&quot;</span><span class="p">)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">filter_theta</span><span class="p">(</span>
            <span class="n">theta</span><span class="o">=</span><span class="n">theta</span><span class="p">,</span>
            <span class="n">metadata</span><span class="o">=</span><span class="n">metadata</span><span class="p">,</span>
            <span class="n">png_out</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="p">[</span><span class="s2">&quot;filter_theta&quot;</span><span class="p">]</span>
        <span class="p">)</span>
    <span class="k">except</span> <span class="n">Error</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
        <span class="n">logging</span><span class="o">.</span><span class="n">error</span><span class="p">(</span><span class="n">e</span><span class="p">)</span>
        <span class="k">raise</span>

<span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;Process over&quot;</span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="pca.html" class="btn btn-neutral float-left" title="PCA" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="plot_sigprofiler_single_sample.html" class="btn btn-neutral float-right" title="PLOT_SIGPROFILER_SINGLE_SAMPLE" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>