<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>BiGR Unofficial &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
        <script src="_static/clipboard.min.js"></script>
        <script src="_static/copybutton.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Unofficial API" href="apidocs/api.html" />
    <link rel="prev" title="Contributing" href="contributing.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="wrappers.html">Wrappers</a></li>
<li class="toctree-l1"><a class="reference internal" href="meta-wrappers.html">Meta-Wrappers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="pipelines.html">BiGR Pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributing</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">BiGR Unofficial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#why-bigr-unofficial-snakemake-wrappers">Why BiGR Unofficial snakemake Wrappers ?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#bigr-pipeline-template">BiGR pipeline template</a></li>
<li class="toctree-l2"><a class="reference internal" href="#snakemake-basics-with-a-simple-rule">Snakemake basics with a simple rule</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#the-input-block">The input block</a></li>
<li class="toctree-l3"><a class="reference internal" href="#the-output-block">The output block</a></li>
<li class="toctree-l3"><a class="reference internal" href="#threads-and-resources">Threads and Resources</a></li>
<li class="toctree-l3"><a class="reference internal" href="#parameters-and-logging">Parameters and logging</a></li>
<li class="toctree-l3"><a class="reference internal" href="#link-to-the-wrapper">Link to the wrapper</a></li>
<li class="toctree-l3"><a class="reference internal" href="#conclusion">Conclusion</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#snakemake-level-up-with-unofficial-environment">Snakemake level up with Unofficial environment</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#constrain-wildcards">Constrain Wildcards</a></li>
<li class="toctree-l3"><a class="reference internal" href="#resources-reservation-made-easier">Resources reservation made easier</a></li>
<li class="toctree-l3"><a class="reference internal" href="#configuration-and-user-defined-parameters">Configuration and user-defined parameters</a></li>
<li class="toctree-l3"><a class="reference internal" href="#fastq-screen-the-final-example">Fastq Screen the final example</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>BiGR Unofficial</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/bigr_unofficial.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="bigr-unofficial">
<span id="id1"></span><h1>BiGR Unofficial<a class="headerlink" href="#bigr-unofficial" title="Permalink to this heading"></a></h1>
<section id="why-bigr-unofficial-snakemake-wrappers">
<h2>Why BiGR Unofficial snakemake Wrappers ?<a class="headerlink" href="#why-bigr-unofficial-snakemake-wrappers" title="Permalink to this heading"></a></h2>
<p>Well, we all use the same cluster at BiGR. We all follow the same project template, all use the same gitlab
instance, all use the same pipelines. We all share knowledge and code snippets. Wouldn’t it be nice if we also
has a single entry point for our developments projects ?</p>
<p>Sure.</p>
<p>Now let’s create a new pipeline together. We need a simple fastp + fastq screen pipeline. Let us list the
requirements of such pipeline:</p>
<ol class="arabic simple">
<li><p>Single-command launch</p></li>
<li><p>This pipeline expects a R1 and a R2 file.</p></li>
<li><p>Default parameters must be fine with Human genome, and oncology studies.</p></li>
<li><p>If user provided fastq files in working directories, search them, use them.</p></li>
<li><p>If user provided a list of fastq files in a file called <code class="docutils literal notranslate"><span class="pre">design.tsv</span></code>, use this file instead of searching for fastq.</p></li>
<li><p>If user provided optional parameters through a file called <code class="docutils literal notranslate"><span class="pre">config.yaml</span></code>, use this file. Else, create one with default parameters.</p></li>
<li><p>Input datasets must be in a directory called “data_input” and should be deleted once they are not useful anymore.</p></li>
<li><p>Output reports and datasets must be in a directory called “data_output”.</p></li>
<li><p>Temporary files must be in a directory called “tmp” or deleted once they are not useful anymore.</p></li>
<li><p>Pipeline must tell user wether it failed, it is still running, or it has finished.</p></li>
<li><p>Pipeline must install itself and must be re-runnable with same results.</p></li>
</ol>
<p>All of these points are made easier on Flamingo with BiGR Unofficial snakemake wrappers, let’s see how and why.</p>
</section>
<section id="bigr-pipeline-template">
<h2>BiGR pipeline template<a class="headerlink" href="#bigr-pipeline-template" title="Permalink to this heading"></a></h2>
<p>We need a working environment. Let’s create a working branch:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">userdata</span><span class="o">/</span><span class="n">t_dayris</span><span class="o">/</span>
<span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">tdayris</span><span class="o">/</span><span class="n">snakemake</span><span class="o">-</span><span class="n">wrappers</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="n">snakemake</span><span class="o">-</span><span class="n">wrappers</span>
<span class="n">git</span> <span class="n">checkout</span> <span class="n">Unofficial</span>
<span class="n">git</span> <span class="n">checkout</span> <span class="o">-</span><span class="n">B</span> <span class="n">generic_fastp_fastq_screen_pipeline</span>
<span class="n">source</span> <span class="n">bigr_pipelines</span><span class="o">/</span><span class="n">bash_aliases</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
<p>Let’s now create our working environment:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="o">-</span><span class="n">pv</span> <span class="n">bigr_pipelines</span><span class="o">/</span><span class="n">fastp_fastqscreen</span><span class="o">/</span><span class="p">{</span><span class="n">rules</span><span class="p">,</span><span class="n">script</span><span class="p">,</span><span class="n">config</span><span class="p">}</span>
<span class="n">touch</span> <span class="n">bigr_pipelines</span><span class="o">/</span><span class="n">fastp_fastqscreen</span><span class="o">/</span><span class="p">{</span><span class="n">Snakefile</span><span class="p">,</span><span class="n">meta</span><span class="o">.</span><span class="n">yaml</span><span class="p">,</span><span class="n">readme</span><span class="o">.</span><span class="n">md</span><span class="p">,</span><span class="n">run</span><span class="o">.</span><span class="n">sh</span><span class="p">}</span>
<span class="n">tree</span> <span class="n">bigr_pipelines</span><span class="o">/</span><span class="n">fastp_fastqscreen</span>
</pre></div>
</div>
<p>This gives:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>bigr_pipelines/fastp_fastqscreen/
├── Snakefile      # &lt;- This is the Snakemake entry point
├── meta.yaml      # &lt;- Short codified help
├── readme.md      # &lt;- The long help. Feel free to write anything you want
├── run.sh         # &lt;- This is our bash entry point
├── rules          # &lt;- There will be our snakemake rules
├── script         # &lt;- Your in-house scripts
└── config         # &lt;- Default configuration for human/mouse/...
</pre></div>
</div>
</section>
<section id="snakemake-basics-with-a-simple-rule">
<span id="snakemake-basics"></span><h2>Snakemake basics with a simple rule<a class="headerlink" href="#snakemake-basics-with-a-simple-rule" title="Permalink to this heading"></a></h2>
<p>We need a fastp / fastq_screen pipeline. Let us start with fastp, the easier rule to write today. We need to
create a file in which to store the snakemake rule:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>touch bigr_pipelines/fastp_fastqscreen/rules/001.fastp.smk
tree bigr_pipelines/fastp_fastqscreen/

bigr_pipelines/fastp_fastqscreen/
├── Snakefile
├── meta.yaml
├── readme.md
├── run.sh
├── rules
│   └── 001.fastp.smk
├── script
└── config
</pre></div>
</div>
<p>I called it <code class="docutils literal notranslate"><span class="pre">001.fastp.smk</span></code> because it’s the first rule in out pipeline, and the file will contain fastp only.
I feel like it is a good practice to name your snakefile based on their content, thus, for future developers,
it will help them to find the snakemake rules “in oder of execution”. At least, coworkers asked for that.</p>
<p>Our snakefile <cite>bigr_pipelines/fastp_fastqscreen/rules/001.fastp.smk</cite> contains the following:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rule</span> <span class="n">fastp</span><span class="p">:</span>          <span class="c1"># The name of the rule, do not start with number. Keep it clear.</span>
    <span class="nb">input</span><span class="p">:</span>           <span class="c1"># The block that contains Input files</span>
        <span class="n">sample</span><span class="o">=</span>      <span class="c1"># List of strings. Paths to fastq files</span>
    <span class="n">output</span><span class="p">:</span>          <span class="c1"># The block that contains Output files</span>
        <span class="n">trimmed</span><span class="o">=</span>     <span class="c1"># List of strings. Paths to trimmed fastq files</span>
        <span class="n">failed</span><span class="o">=</span>      <span class="c1"># List of strings. Paths to discarded reads</span>
        <span class="n">html</span><span class="o">=</span>        <span class="c1"># String. Path to html formatted trimming report.</span>
        <span class="n">json</span><span class="o">=</span>        <span class="c1"># String. Path to json formatted trimming report.</span>
    <span class="n">threads</span><span class="p">:</span>         <span class="c1"># Number of threads used</span>
    <span class="n">resources</span><span class="p">:</span>       <span class="c1"># The block that contains cluster resources</span>
        <span class="n">mem_mb</span><span class="o">=</span>      <span class="c1"># Integer or Function, the number of MB or ram needed.</span>
        <span class="n">time_min</span><span class="o">=</span>    <span class="c1"># Integer or Function, the time (minutes) required to run fastp</span>
        <span class="n">tmp</span><span class="o">=</span>         <span class="c1"># Path to overloaded temporary directory</span>
    <span class="n">log</span><span class="p">:</span>             <span class="c1"># Path to logging file</span>
    <span class="n">params</span><span class="p">:</span>          <span class="c1"># The block that contains non-file parameters</span>
        <span class="n">extra</span><span class="o">=</span>       <span class="c1"># Optional parameters for fastp</span>
        <span class="n">adapters</span><span class="o">=</span>    <span class="c1"># Optional adapters list for fastp</span>
    <span class="n">wrapper</span><span class="p">:</span>         <span class="c1"># Path/url to the used wrapper</span>
</pre></div>
</div>
<p>How in the hell did I come up with all of that ?</p>
<p>Well, I’m using <a class="reference external" href="https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/fastp.html">fastp</a> and there is already
a documentation, installation, and running examples of Fastp with the Snakemake wrappers. No need for me to search
for installation paths, nor deployments scheme. Everything is shipped with the wrappers. I copied the example, and
now we will fill it together.</p>
<section id="the-input-block">
<h3>The input block<a class="headerlink" href="#the-input-block" title="Permalink to this heading"></a></h3>
<p>Let’s fill the input block:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sample</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
    <span class="s2">&quot;data_input/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{stream}</span><span class="s2">.fq.gz&quot;</span><span class="p">,</span>   <span class="c1"># Input are in data_input</span>
    <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>                      <span class="c1"># Regular expression for sample names</span>
    <span class="n">stream</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>                       <span class="c1"># List of sample names</span>
<span class="p">)</span>
</pre></div>
</div>
<p>Snakemake has a function called <a class="reference external" href="https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#the-expand-function">expand</a>. To make it simple, it returns an iterable (like a list) wich contains all
possible combinations of strings between “{}”. If I write:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">expand</span><span class="p">(</span><span class="s2">&quot;My example is </span><span class="si">{adjective}</span><span class="s2"> </span><span class="si">{status}</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">adjective</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;very&#39;</span><span class="p">,</span> <span class="s1">&#39;truly&#39;</span><span class="p">],</span> <span class="n">status</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;awesome&#39;</span><span class="p">,</span> <span class="s1">&#39;cool&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>Then it will result in:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="s2">&quot;My example is very awesome&quot;</span><span class="p">,</span>
 <span class="s2">&quot;My example is truly awesome&quot;</span><span class="p">,</span>
 <span class="s2">&quot;My example is very cool&quot;</span><span class="p">,</span>
 <span class="s2">&quot;My example is truly cool&quot;</span><span class="p">]</span>
</pre></div>
</div>
<p>Got it ? Good.</p>
<p>In our case, the expand function contains a regular expression: <code class="docutils literal notranslate"><span class="pre">{sample}</span></code>. In snakemake, this is called a
<a class="reference external" href="https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#wildcards">wildcard</a> . We would like this
regular expression to match sample names, and sample names only. We can constrain it. We will dot that later,
be confident, and let a TODO marker on your desktop.</p>
<p>For now, the expand function we wrote in input of our rule fastp will result in:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="sa">r</span><span class="s2">&quot;data_input/.+.1.fq.gz&quot;</span><span class="p">,</span> <span class="sa">r</span><span class="s2">&quot;data_input/.+.2.fq.gz&quot;</span><span class="p">]</span>
</pre></div>
</div>
<p>These are two regular expressions, not that, for now, “{sample}” is solved as “.+”, but we shall precise it later.</p>
</section>
<section id="the-output-block">
<h3>The output block<a class="headerlink" href="#the-output-block" title="Permalink to this heading"></a></h3>
<p>Let’s fill the output block:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output</span><span class="p">:</span>
    <span class="n">trimmed</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>  <span class="c1"># We expect two trimmed fastq files, 1 for R1, 1 for R2.</span>
        <span class="s2">&quot;001.fastp/trimmed/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{stream}</span><span class="s2">.fastq&quot;</span><span class="p">,</span>
        <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">stream</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>
    <span class="p">),</span>
    <span class="c1"># failed= We are not interested in discarded reads, this parameter is flagged optional in wrapper doc.</span>
    <span class="n">html</span><span class="o">=</span><span class="s2">&quot;001.fastp/html/</span><span class="si">{sample}</span><span class="s2">.html&quot;</span><span class="p">,</span>  <span class="c1"># We want only one report per sample</span>
    <span class="n">json</span><span class="o">=</span><span class="s2">&quot;001.fastp/json/</span><span class="si">{sample}</span><span class="s2">.json&quot;</span><span class="p">,</span>  <span class="c1"># We want only one report per sample</span>
</pre></div>
</div>
<p>It is very important that all output share the same wildcards as input. Otherwise, Snakemake won’t be able to solve
the dependency graph and won’t be able to know what to do. This is the same with GNUMake, CMake, BioMake, etc.</p>
<p>That one was easy, wasn’t it ?</p>
</section>
<section id="threads-and-resources">
<h3>Threads and Resources<a class="headerlink" href="#threads-and-resources" title="Permalink to this heading"></a></h3>
<p>These two blocks are here for cluster execution. The name and values available in it are <em>fixed and required</em>. Please
always fill them. Otherwise default values will be provided, but my not match with your needs.</p>
<p>The wrapper documentation says Fastp allows multiple threads.</p>
<p>Usually, Fastp requires quite few RAM and does not take long to run.</p>
<p>Let’s fill these blocks:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">threads</span><span class="p">:</span> <span class="mi">4</span>
<span class="n">resources</span><span class="p">:</span>
    <span class="n">mem_mb</span><span class="o">=</span><span class="mi">1024</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span> <span class="c1"># Two GB should be enough</span>
    <span class="n">time_min</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span>     <span class="c1"># 20 minutes should be enough</span>
    <span class="n">tmp</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span><span class="p">,</span>       <span class="c1"># &quot;tmp&quot; not &quot;/tmp&quot; ! This is very important.</span>
</pre></div>
</div>
<p>If you ever analyse a very big pair of fastq files, with hundreads of milions of reads, then Fastp fill take longer
than 20 minutes to run and the cluster will delete your job. This is not cool. We would like to automatically
re-launch the rule with more time. Or even better, reserve time according to the fastq file size !</p>
<p>Both are possible, but we will come to that a bit later. This is still Snakemake basics ! Don’t ask such interesting
questions so soon !</p>
</section>
<section id="parameters-and-logging">
<h3>Parameters and logging<a class="headerlink" href="#parameters-and-logging" title="Permalink to this heading"></a></h3>
<p>Let’s fill these blocks:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">log</span><span class="p">:</span>
    <span class="s2">&quot;logs/fastp/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
<span class="n">params</span><span class="p">:</span>
    <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;&quot;</span>      <span class="c1"># Optional parameters, see fastp CLI documentation</span>
    <span class="c1"># adapters=&quot;&quot; # Optional according to the fastp wrapper documentation.</span>
</pre></div>
</div>
<p>It is very important that all output and lgos share the same wildcards as input. Otherwise, Snakemake won’t be
able to solve the dependency graph and won’t be able to know what to do. This is the same with GNUMake,
CMake, BioMake, etc.</p>
<p>Parameters are almost unique for each wrapper, please refer to the wrapper documentation.</p>
</section>
<section id="link-to-the-wrapper">
<h3>Link to the wrapper<a class="headerlink" href="#link-to-the-wrapper" title="Permalink to this heading"></a></h3>
<p>The easiest way to link to the wrapper is to put:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">wrapper</span><span class="p">:</span>
    <span class="s2">&quot;bio/fastp&quot;</span>
</pre></div>
</div>
<p>How do I come up with that ? Well, in the repository <cite>snakemake-wrappers</cite>, fastp is under: <code class="docutils literal notranslate"><span class="pre">bio/fastp</span></code>. Yes, it’s
that simple.</p>
<p>This will tell Snakemake how to install and deploy Fastp, how to build the command line and how to launch this tool.</p>
<p>Thanks to the Unofficial environment, Snakemake won’t search on the web but on Flamingo, and if the environment is
already available, then Snakemake won’t re-install it. Environments are sha-signed, so you cannot destroy other
people’s environments, you can only share them ! Enjoy !</p>
</section>
<section id="conclusion">
<h3>Conclusion<a class="headerlink" href="#conclusion" title="Permalink to this heading"></a></h3>
<p>Our file <code class="docutils literal notranslate"><span class="pre">rules/001.fastp.smk</span></code> contains:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rule</span> <span class="n">fastp</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">sample</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;data_input/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{stream}</span><span class="s2">.fq.gz&quot;</span><span class="p">,</span>   <span class="c1"># Input are in data_input</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>                      <span class="c1"># Regular expression for sample names</span>
            <span class="n">stream</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>                       <span class="c1"># List of sample names</span>
        <span class="p">)</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">trimmed</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>  <span class="c1"># We expect two trimmed fastq files, 1 for R1, 1 for R2.</span>
            <span class="s2">&quot;001.fastp/trimmed/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{stream}</span><span class="s2">.fastq&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>
            <span class="n">stream</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="c1"># failed= We are not interested in discarded reads, this parameter is flagged optional in wrapper doc.</span>
        <span class="n">html</span><span class="o">=</span><span class="s2">&quot;001.fastp/html/</span><span class="si">{sample}</span><span class="s2">.html&quot;</span><span class="p">,</span>  <span class="c1"># We want only one report per sample</span>
        <span class="n">json</span><span class="o">=</span><span class="s2">&quot;001.fastp/json/</span><span class="si">{sample}</span><span class="s2">.json&quot;</span><span class="p">,</span>  <span class="c1"># We want only one report per sample</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">4</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="mi">1024</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span> <span class="c1"># Two GB should be enough</span>
        <span class="n">time_min</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span>     <span class="c1"># 20 minutes should be enough</span>
        <span class="n">tmp</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span><span class="p">,</span>       <span class="c1"># &quot;tmp&quot; not &quot;/tmp&quot; ! This is very important.</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/fastp/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;&quot;</span>      <span class="c1"># Optional parameters, see fastp CLI documentation</span>
        <span class="c1"># adapters=&quot;&quot; # Optional according to the fastp wrapper documentation.</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/fastp&quot;</span>
</pre></div>
</div>
</section>
</section>
<section id="snakemake-level-up-with-unofficial-environment">
<span id="snakemake-advanced"></span><h2>Snakemake level up with Unofficial environment<a class="headerlink" href="#snakemake-level-up-with-unofficial-environment" title="Permalink to this heading"></a></h2>
<p>We saw earlier that wildcards were equal to “.+” by default and might be constrained. We saw earlier that resources
could be set upon input file size. We saw earlier that Snakemake can re-run rules that failed due to out-of-memory
error, or out-of-time error.</p>
<p>We said we would talk about it in the future.</p>
<p>The. Future. Is. Now.</p>
<p>Also, the future is written in: <code class="docutils literal notranslate"><span class="pre">000.commons.smk</span></code>. It’s the first file included in the main Snakefile,
and it contains only pure python functions and intructions.</p>
<section id="constrain-wildcards">
<h3>Constrain Wildcards<a class="headerlink" href="#constrain-wildcards" title="Permalink to this heading"></a></h3>
<p>Remember, <code class="docutils literal notranslate"><span class="pre">&quot;{samlple}&quot;</span></code> is solved as <code class="docutils literal notranslate"><span class="pre">&quot;.+&quot;</span></code> (any character, at least once). We would like to have the list of samples
instead.</p>
<p>So, we need to acquire the list of samples, either from user input, of though disc search.</p>
<p>Unofficial provides the functions for that. Here, we search for pairs fastq files. The snakemake-unofficial API lists a
function named <code class="docutils literal notranslate"><span class="pre">search_fastq_pairs</span></code> in the <code class="docutils literal notranslate"><span class="pre">file_manager</span></code> module. This function returns a dictionary formatted
as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">Sample1</span><span class="p">:</span> <span class="p">{</span><span class="n">Upstream_file</span><span class="p">:</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">R1</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span><span class="p">,</span> <span class="n">Downstream_file</span><span class="p">:</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">R2</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span><span class="p">},</span> <span class="n">Sample2</span> <span class="o">...</span><span class="p">}</span>
</pre></div>
</div>
<p>This can be passed to the function <code class="docutils literal notranslate"><span class="pre">get_design</span></code> in the module <code class="docutils literal notranslate"><span class="pre">file_manager</span></code> to produce a design file.</p>
<p>As easy as is sound, we just need to import a single module, use two function and <em>BAM</em>! A design file and a
list of available pairs of fastq file per sample is available !</p>
<p>So, in <cite>000.commons.smk</cite>, let’s write the following:</p>
<blockquote>
<div><p># Manage paths easily in Python
from pathlib import Path</p>
<p># Get workflow path (the one you’re working on!)
workflow_source_dir = Path(snakemake.workflow.srcdir(“..”))</p>
<p># Get path to shared libraries in Unofficial Snakemake wrappers
common = str(workflow_source_dir / “..” / “common” / “python”)</p>
<p># Add this to the list of available libraries in Python
import sys
sys.path.append(common)</p>
<p># Finally ! Add the module file_manager
from file_manager import *</p>
</div></blockquote>
<p>The search of all fastq files and the build of the design file is made by the following:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">os</span> <span class="c1"># To deal with operating system in Python</span>
<span class="n">design</span> <span class="o">=</span> <span class="n">get_design</span><span class="p">(</span>
    <span class="n">dirpath</span><span class="o">=</span><span class="n">os</span><span class="o">.</span><span class="n">getcwd</span><span class="p">(),</span>           <span class="c1"># Where to search for fastq files</span>
    <span class="n">search_func</span><span class="o">=</span><span class="n">search_fastq_pairs</span> <span class="c1"># What function to use in order to search for fastq files</span>
<span class="p">)</span>
</pre></div>
</div>
<p>This saves on disk the result, if and only if another design <em>does not</em> exist. No file will be deleted.</p>
<p>The list of available sample files is available with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">design</span><span class="o">.</span><span class="n">Sample_id</span><span class="p">)</span>
</pre></div>
</div>
<p>We want to constrain our wildcards, this is easily done with the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">wildcard_constraints</span><span class="p">:</span>
    <span class="n">sample</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;|&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">design</span><span class="o">.</span><span class="n">Sample_id</span><span class="p">),</span>
    <span class="n">stream</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;1|2&quot;</span><span class="p">,</span>
</pre></div>
</div>
<p>Tadaaaam ! Our pipeline now finds fastq files by itself. By the way, if your sample name ends
with a number, Snakemake won’t confuse it with sequencing strands.</p>
<p>Let’s imagine we have two samples called: <cite>Sample1</cite> and <cite>Sample2</cite>. The <code class="docutils literal notranslate"><span class="pre">expand</span></code> in <cite>001.fastp.smk</cite>
contains:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sample</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
    <span class="s2">&quot;data_input/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{stream}</span><span class="s2">.fq.gz&quot;</span><span class="p">,</span>   <span class="c1"># Input are in data_input</span>
    <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>                      <span class="c1"># Regular expression for sample names</span>
    <span class="n">stream</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>                       <span class="c1"># List of sample names</span>
<span class="p">)</span>
</pre></div>
</div>
<p>and solves itself as:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="s2">&quot;data_input/Sample1|Sample2.1.fq.gz&quot;</span><span class="p">,</span> <span class="s2">&quot;data_input/Sample1|Sample2.2.fq.gz&quot;</span><span class="p">]</span>
</pre></div>
</div>
<p>No ambiguity is allowed in Snakemake.</p>
</section>
<section id="resources-reservation-made-easier">
<h3>Resources reservation made easier<a class="headerlink" href="#resources-reservation-made-easier" title="Permalink to this heading"></a></h3>
<p>We said earlier that resources could be set upon input file size. These resources should be adjusted
in case we need more memory and/or time for our job on the cluster.</p>
<p>This can be done with functions respecting the <a class="reference external" href="https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#resources">signature</a>
described in official snakemake documentation. Unofficial Snakemake Wrapper has a set of function designed
to make it easy and human readable. These functions are in <code class="docutils literal notranslate"><span class="pre">reservation</span></code> module. Just like before, they
will be imported in your <cite>000.commons.smk</cite>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Manage paths easily in Python</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>

<span class="c1"># Get workflow path (the one you&#39;re working on!)</span>
<span class="n">workflow_source_dir</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">workflow</span><span class="o">.</span><span class="n">srcdir</span><span class="p">(</span><span class="s2">&quot;..&quot;</span><span class="p">))</span>

<span class="c1"># Get path to shared libraries in Unofficial Snakemake wrappers</span>
<span class="n">common</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">workflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;..&quot;</span> <span class="o">/</span> <span class="s2">&quot;common&quot;</span> <span class="o">/</span> <span class="s2">&quot;python&quot;</span><span class="p">)</span>

<span class="c1"># Add this to the list of available libraries in Python</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">common</span><span class="p">)</span>

<span class="c1"># Finally ! Add the module file_manager</span>
<span class="kn">from</span> <span class="nn">file_manager</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Import the functions for easy time/memory reservation</span>
<span class="kn">from</span> <span class="nn">reservation</span> <span class="kn">import</span> <span class="o">*</span>

<span class="kn">import</span> <span class="nn">os</span> <span class="c1"># To deal with operating system in Python</span>
<span class="n">design</span> <span class="o">=</span> <span class="n">get_design</span><span class="p">(</span>
    <span class="n">dirpath</span><span class="o">=</span><span class="n">os</span><span class="o">.</span><span class="n">getcwd</span><span class="p">(),</span>           <span class="c1"># Where to search for fastq files</span>
    <span class="n">search_func</span><span class="o">=</span><span class="n">search_fastq_pairs</span> <span class="c1"># What function to use in order to search for fastq files</span>
<span class="p">)</span>
</pre></div>
</div>
<p>And your <code class="docutils literal notranslate"><span class="pre">fastp</span></code> rule goes from:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">resources</span><span class="p">:</span>
    <span class="n">mem_mb</span><span class="o">=</span><span class="mi">1024</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span> <span class="c1"># Two GB should be enough</span>
    <span class="n">time_min</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span>     <span class="c1"># 20 minutes should be enough</span>
    <span class="n">tmp</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span><span class="p">,</span>       <span class="c1"># &quot;tmp&quot; not &quot;/tmp&quot; ! This is very important.</span>
</pre></div>
</div>
<p>to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">resources</span><span class="p">:</span>
    <span class="n">mem_mb</span><span class="o">=</span><span class="n">get_2gb_per_attempt</span><span class="p">,</span>   <span class="c1"># 2gb on first try, then 4gb, then 6gb, ...</span>
    <span class="n">time_min</span><span class="o">=</span><span class="n">get_20m_per_attempt</span><span class="p">,</span> <span class="c1"># 20min on first try, then 40min, 1h, ...</span>
    <span class="n">tmp</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span><span class="p">,</span>
</pre></div>
</div>
<p>Unofficial Snakemake Wrappers will handle the re-submission.</p>
</section>
<section id="configuration-and-user-defined-parameters">
<h3>Configuration and user-defined parameters<a class="headerlink" href="#configuration-and-user-defined-parameters" title="Permalink to this heading"></a></h3>
<p>Depending on a project content, on the biological question, or simply based on the organism, many
tools and command line may differ. For instance, read adapters may be provided depending on the
sequencing library that was designed. Trimming parameters differ from bulk-rnaseq and whole exome
sequencing.</p>
<p>These information will be stored in a configuration (yaml formatted) that your fellow bioinformatician
may change (or not). These parameters must be forewarded while executing the pipeline instructions.</p>
<p>Let us create the first configuration file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>touch bigr_pipelines/fastp_fastqscreen/config/config.yaml
tree bigr_pipelines/fastp_fastqscreen/

bigr_pipelines/fastp_fastqscreen/
├── Snakefile
├── meta.yaml
├── readme.md
├── run.sh
├── rules
│   ├── 000.common.smk
│   └── 001.fastp.smk
├── script
└── config
    └── config.yaml
</pre></div>
</div>
<p>Let us define some variables in this configuragion file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Maximum number of threads</span>
<span class="n">max_threads</span><span class="p">:</span> <span class="mi">20</span>

<span class="n">fastp</span><span class="p">:</span>
    <span class="n">extra</span><span class="p">:</span> <span class="s2">&quot;&quot;</span> <span class="c1"># Optional parameters</span>
    <span class="n">adapters</span><span class="p">:</span> <span class="n">null</span> <span class="c1"># Optional adapters</span>
</pre></div>
</div>
<p>This file is stored in the pipeline source. We cannot allow the user to modify it. We must provide
a copy of the default parameters for the user to change them.</p>
<p>Once again, Unofficial Snakemake Wrappers provide a function for that. In the file <cite>000.common.smk</cite>
let’s write:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Manage paths easily in Python</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>

<span class="c1"># Get workflow path (the one you&#39;re working on!)</span>
<span class="n">workflow_source_dir</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">workflow</span><span class="o">.</span><span class="n">srcdir</span><span class="p">(</span><span class="s2">&quot;..&quot;</span><span class="p">))</span>

<span class="c1"># Get path to shared libraries in Unofficial Snakemake wrappers</span>
<span class="n">common</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">workflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;..&quot;</span> <span class="o">/</span> <span class="s2">&quot;common&quot;</span> <span class="o">/</span> <span class="s2">&quot;python&quot;</span><span class="p">)</span>

<span class="c1"># Add this to the list of available libraries in Python</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">common</span><span class="p">)</span>

<span class="c1"># Finally ! Add the module file_manager</span>
<span class="kn">from</span> <span class="nn">file_manager</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Import the functions for easy time/memory reservation</span>
<span class="kn">from</span> <span class="nn">reservation</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Import Config file if needed, use user-defined one if present.</span>
<span class="n">default_config</span> <span class="o">=</span> <span class="n">read_yaml</span><span class="p">(</span><span class="n">workflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;config&quot;</span> <span class="o">/</span> <span class="s2">&quot;config.yaml&quot;</span><span class="p">)</span>
<span class="n">configfile</span><span class="p">:</span> <span class="n">get_config</span><span class="p">(</span><span class="n">default_config</span><span class="o">=</span><span class="n">default_config</span><span class="p">)</span>

<span class="kn">import</span> <span class="nn">os</span> <span class="c1"># To deal with operating system in Python</span>
<span class="n">design</span> <span class="o">=</span> <span class="n">get_design</span><span class="p">(</span>
    <span class="n">dirpath</span><span class="o">=</span><span class="n">os</span><span class="o">.</span><span class="n">getcwd</span><span class="p">(),</span>           <span class="c1"># Where to search for fastq files</span>
    <span class="n">search_func</span><span class="o">=</span><span class="n">search_fastq_pairs</span> <span class="c1"># What function to use in order to search for fastq files</span>
<span class="p">)</span>
</pre></div>
</div>
<p>We can now modify our file <cite>001.fastp.smk</cite> in order to include configurations:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rule</span> <span class="n">fastp</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">sample</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;data_input/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{stream}</span><span class="s2">.fq.gz&quot;</span><span class="p">,</span>   <span class="c1"># Input are in data_input</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>                      <span class="c1"># Regular expression for sample names</span>
            <span class="n">stream</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>                       <span class="c1"># List of sample names</span>
        <span class="p">)</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">trimmed</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>  <span class="c1"># We expect two trimmed fastq files, 1 for R1, 1 for R2.</span>
            <span class="s2">&quot;001.fastp/trimmed/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{stream}</span><span class="s2">.fastq&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>
            <span class="n">stream</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="c1"># failed= We are not interested in discarded reads, this parameter is flagged optional in wrapper doc.</span>
        <span class="n">html</span><span class="o">=</span><span class="s2">&quot;001.fastp/html/</span><span class="si">{sample}</span><span class="s2">.html&quot;</span><span class="p">,</span>  <span class="c1"># We want only one report per sample</span>
        <span class="n">json</span><span class="o">=</span><span class="s2">&quot;001.fastp/json/</span><span class="si">{sample}</span><span class="s2">.json&quot;</span><span class="p">,</span>  <span class="c1"># We want only one report per sample</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">4</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="mi">1024</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span> <span class="c1"># Two GB should be enough</span>
        <span class="n">time_min</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span>     <span class="c1"># 20 minutes should be enough</span>
        <span class="n">tmp</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span><span class="p">,</span>       <span class="c1"># &quot;tmp&quot; not &quot;/tmp&quot; ! This is very important.</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/fastp/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="c1"># If user deleted &#39;extra&#39; in fastp, then leave it empty and do not raise error</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;fastp&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s1">&#39;extra&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span>
        <span class="c1"># If user deleted &#39;adapters&#39; in fastp, then leave it to `None` and do not raise error</span>
        <span class="n">adapters</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;fastp&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s1">&#39;extra&#39;</span><span class="p">,</span> <span class="kc">None</span><span class="p">)</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/fastp&quot;</span>
</pre></div>
</div>
</section>
<section id="fastq-screen-the-final-example">
<h3>Fastq Screen the final example<a class="headerlink" href="#fastq-screen-the-final-example" title="Permalink to this heading"></a></h3>
<p>Fastq screen requires a new file in the repository <code class="docutils literal notranslate"><span class="pre">rules</span></code>, we name it <code class="docutils literal notranslate"><span class="pre">002.fastq_screen.smk</span></code>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>touch bigr_pipelines/fastp_fastqscreen/rules/002.fastq_screen.smk
tree bigr_pipelines/fastp_fastqscreen/

bigr_pipelines/fastp_fastqscreen/
├── Snakefile
├── meta.yaml
├── readme.md
├── run.sh
├── rules
│   ├── 000.common.smk
│   ├── 001.fastp.smk
│   └── 002.fastq_screen.smk
├── script
└── config
    └── config.yaml
</pre></div>
</div>
<p>Thanks to the <a class="reference external" href="https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/fastq_screen.html">wrapper documentation</a>,
we know that the file <code class="docutils literal notranslate"><span class="pre">rules/002.fastq_screen.smk</span></code> contains:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rule</span> <span class="n">fastq_screen</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;data_input/</span><span class="si">{sample}</span><span class="s2">.</span><span class="si">{stream}</span><span class="s2">.fq.gz&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">txt</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;fastq_screen/</span><span class="si">{sample}</span><span class="s2">.fastq_screen.txt&quot;</span><span class="p">),</span>
        <span class="n">png</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;fastq_screen/</span><span class="si">{sample}</span><span class="s2">.fastq_screen.png&quot;</span><span class="p">)</span>
    <span class="n">threads</span><span class="p">:</span> <span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;threads&quot;</span><span class="p">,</span> <span class="mi">20</span><span class="p">)</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="n">get_15gb_per_attempt</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="n">get_1h_per_attempt</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">fastq_screen_config</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;fastq_screen&quot;</span><span class="p">],</span>
        <span class="n">subset</span><span class="o">=</span><span class="mi">100000</span><span class="p">,</span>     <span class="c1"># We do not allow user to change the default number of mapped reads</span>
        <span class="n">aligner</span><span class="o">=</span><span class="s1">&#39;bowtie2&#39;</span>  <span class="c1"># We do not allow user to change default mapper</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/fastq_screen/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/fastq_screen&quot;</span>
</pre></div>
</div>
<p>In the file <code class="docutils literal notranslate"><span class="pre">config/config.yaml</span></code> we add the databases:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Maximum number of threads</span>
<span class="n">max_threads</span><span class="p">:</span> <span class="mi">20</span>

<span class="c1"># Fastp parameters</span>
<span class="n">fastp</span><span class="p">:</span>
    <span class="n">extra</span><span class="p">:</span> <span class="s2">&quot;&quot;</span> <span class="c1"># Optional parameters</span>
    <span class="n">adapters</span><span class="p">:</span> <span class="n">null</span> <span class="c1"># Optional adapters</span>

<span class="c1"># Fastq Screen info</span>
<span class="n">fastq_screen</span><span class="p">:</span>
    <span class="n">database</span><span class="p">:</span>
        <span class="n">Contaminants</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Adapters</span><span class="o">/</span><span class="n">Contaminants</span>
        <span class="n">AThaliana</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Arabidopsis</span><span class="o">/</span><span class="n">Arabidopsis_thaliana</span><span class="o">.</span><span class="n">TAIR10</span>
        <span class="n">Drosophila</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Drosophila</span><span class="o">/</span><span class="n">BDGP6</span>
        <span class="n">Ecoli</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">E_coli</span><span class="o">/</span><span class="n">Ecoli</span>
        <span class="n">Human</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Human</span><span class="o">/</span><span class="n">Homo_sapiens</span><span class="o">.</span><span class="n">GRCh38</span>
        <span class="n">Lambda</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Lambda</span><span class="o">/</span><span class="n">Lambda</span>
        <span class="n">Mitochondria</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Mitochondria</span><span class="o">/</span><span class="n">mitochondria</span>
        <span class="n">Mouse</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Mouse</span><span class="o">/</span><span class="n">Mus_musculus</span><span class="o">.</span><span class="n">GRCm38</span>
        <span class="n">PhiX</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">PhiX</span><span class="o">/</span><span class="n">phi_plus_SNPs</span>
        <span class="n">Rat</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Rat</span><span class="o">/</span><span class="n">Rnor_6</span><span class="mf">.0</span>
        <span class="n">rRNA</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">rRNA</span><span class="o">/</span><span class="n">GRCm38_rRNA</span>
        <span class="n">Vectors</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Vectors</span><span class="o">/</span><span class="n">Vectors</span>
        <span class="n">Worm</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Worm</span><span class="o">/</span><span class="n">Caenorhabditis_elegans</span><span class="o">.</span><span class="n">WBcel235</span>
        <span class="n">Yeast</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">Yeast</span><span class="o">/</span><span class="n">Saccharomyces_cerevisiae</span><span class="o">.</span><span class="n">R64</span><span class="o">-</span><span class="mi">1</span><span class="o">-</span><span class="mi">1</span>
        <span class="n">SalmoSalar</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">beegfs</span><span class="o">/</span><span class="n">database</span><span class="o">/</span><span class="n">bioinfo</span><span class="o">/</span><span class="n">Index_DB</span><span class="o">/</span><span class="n">Fastq_Screen</span><span class="o">/</span><span class="mf">0.14.0</span><span class="o">/</span><span class="n">SalmoSalar</span><span class="o">/</span><span class="n">Salmo_salar</span><span class="o">.</span><span class="n">ICSASG_v2</span>
    <span class="n">aligner_paths</span><span class="p">:</span>
        <span class="n">bowtie2</span><span class="p">:</span> <span class="n">bowtie2</span>
</pre></div>
</div>
<p>The file <cite>000.commons.smk</cite> does not need to be changed.</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="contributing.html" class="btn btn-neutral float-left" title="Contributing" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="apidocs/api.html" class="btn btn-neutral float-right" title="Unofficial API" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>