<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>MIRASEQ &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="NFCORE-CHIPSEQ" href="nfcore_chipseq.html" />
    <link rel="prev" title="LOW_PASS_WGS_COPY_NUMBER" href="low_pass_wgs_copy_number.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers and Pipelines</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="CTC_Variant_Calling_Integragen_Legacy.html">CTC_VARIANT_CALLING_INTEGRAGEN_LEGACY</a></li>
<li class="toctree-l2"><a class="reference internal" href="WES_germline.html">WES_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="annot_sv.html">ANNO_SV</a></li>
<li class="toctree-l2"><a class="reference internal" href="bcl2fastq.html">BCL2FASTQ (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="celltype_deconvolution_rnaseq.html">CELLTYPE_DECONVOLUTION_RNASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="chipseq.html">CHIPSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="cytoscan_eacon.html">EACON_CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="design_builder.html">DESIGN_BUILDER</a></li>
<li class="toctree-l2"><a class="reference internal" href="dge_deseq2.html">DGE_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastqc_multiqc.html">FASTQC_MULTIQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_set_enrichment.html">GENE_SET_ENRICHMENT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="low_pass_wgs_copy_number.html">LOW_PASS_WGS_COPY_NUMBER</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">MIRASEQ</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l3"><a class="reference internal" href="#input-output">Input/Output</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-meta-wrappers">Used meta-wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#snakefile">Snakefile</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="nfcore_chipseq.html">NFCORE-CHIPSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="oncoscan_eacon.html">EACON_ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="retrotransposons.html">RETROTRANSPOSONS</a></li>
<li class="toctree-l2"><a class="reference internal" href="rnaseq.html">RNASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon_quant.html">SALMON_QUANT</a></li>
<li class="toctree-l2"><a class="reference internal" href="sarek_enhanced.html">SAREK ENHANCED</a></li>
<li class="toctree-l2"><a class="reference internal" href="seqoia.html">SEQOIA SINGLE-SAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="sigprofiler_signatures.html">SIGPROFILER_SIGNATURES (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_snpsift.html">SNPEFF_SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_ampliseq.html">VARIANT_CALLING_AMPLISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_germline.html">VARIANT_CALLING_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_germline_wes.html">VARIANT_CALLING_GERMLINE_WES (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_somatic.html">VARIANT_CALLING_SOMATIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_callling_germline.html">VARIANT_CALLING_GERMLINE (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_occurence_cohort.html">VARIANT_OCCURENCE_COHORT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="wes_somatic.html">VARIANT_CALLING_SOMATIC</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../pipelines.html">BiGR Pipelines</a> &raquo;</li>
      <li>MIRASEQ</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/bigr-pipelines/miraseq.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="miraseq">
<span id="id1"></span><h1>MIRASEQ<a class="headerlink" href="#miraseq" title="Permalink to this headline"></a></h1>
<p>Analyse methylation enrichment on MIRA-Seq with biskmark</p>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this headline"></a></h2>
<p>In order to run the pipeline, use the following commands</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Go to your working directory</span>

<span class="nb">cd</span> /path/to/my/working/directory

<span class="c1"># Build a design file (see below)</span>

<span class="c1"># Copy/paste the following line for **HG19**</span>

bash /mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/miraseq/run.sh hg19

<span class="c1"># Copy/paste the following line for **HG38**</span>

bash /mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/miraseq/run.sh hg38
</pre></div>
</div>
</section>
<section id="input-output">
<h2>Input/Output<a class="headerlink" href="#input-output" title="Permalink to this headline"></a></h2>
<p><strong>Input:</strong></p>
<ul class="simple">
<li><p>Pair of fastq files</p></li>
<li><p>Fasta-formatted genome sequence</p></li>
</ul>
<p><strong>Output:</strong></p>
<ul class="simple">
<li><p>Bam-formatted mapped reads</p></li>
<li><p>Methylation reports per sample</p></li>
</ul>
</section>
<section id="used-meta-wrappers">
<h2>Used meta-wrappers<a class="headerlink" href="#used-meta-wrappers" title="Permalink to this headline"></a></h2>
<p>The following individual meta-wrappers are used in this pipeline:</p>
<ul class="simple">
<li><p><span class="xref std std-ref">meta/bio/bismark</span></p></li>
</ul>
<p>Please refer to each meta-wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this headline"></a></h2>
<p>The following individual wrappers are used in this pipeline:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/fastp.html#bio-fastp"><span class="std std-ref">FASTP</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/BiGR/copy.html#bio-bigr-copy"><span class="std std-ref">IGR_COPY</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="snakefile">
<h2>Snakefile<a class="headerlink" href="#snakefile" title="Permalink to this headline"></a></h2>
<p>The pipeline contains the following steps:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">snakemake.utils</span> <span class="kn">import</span> <span class="n">min_version</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>
<span class="kn">from</span> <span class="nn">yaml</span> <span class="kn">import</span> <span class="n">dump</span>
<span class="n">min_version</span><span class="p">(</span><span class="s2">&quot;6.0&quot;</span><span class="p">)</span>

<span class="kn">import</span> <span class="nn">sys</span>

<span class="n">worflow_source_dir</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">workflow</span><span class="o">.</span><span class="n">srcdir</span><span class="p">(</span><span class="s2">&quot;.&quot;</span><span class="p">))</span>
<span class="n">common</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">worflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;..&quot;</span> <span class="o">/</span> <span class="s2">&quot;common&quot;</span> <span class="o">/</span> <span class="s2">&quot;python&quot;</span><span class="p">)</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">common</span><span class="p">)</span>

<span class="kn">from</span> <span class="nn">dataframes</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">file_manager</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">files_linker</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">graphics</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">write_yaml</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">messages</span> <span class="kn">import</span> <span class="n">message</span>


<span class="c1">#################</span>
<span class="c1">### Preambule ###</span>
<span class="c1">#################</span>

<span class="n">logging</span><span class="o">.</span><span class="n">basicConfig</span><span class="p">(</span>
    <span class="n">filename</span><span class="o">=</span><span class="s2">&quot;snakemake.miraseq.log&quot;</span><span class="p">,</span>
    <span class="n">filemode</span><span class="o">=</span><span class="s2">&quot;w&quot;</span><span class="p">,</span>
    <span class="n">level</span><span class="o">=</span><span class="n">logging</span><span class="o">.</span><span class="n">DEBUG</span>
<span class="p">)</span>

<span class="n">default_config</span> <span class="o">=</span> <span class="n">read_yaml</span><span class="p">(</span><span class="n">worflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;config.hg38.yaml&quot;</span><span class="p">)</span>
<span class="n">configfile</span><span class="p">:</span> <span class="n">get_config</span><span class="p">(</span><span class="n">default_config</span><span class="p">)</span>
<span class="n">design</span> <span class="o">=</span> <span class="n">get_design</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">getcwd</span><span class="p">(),</span> <span class="n">search_fastq_files_dict</span><span class="p">)</span>

<span class="n">fastq_links</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">if</span> <span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;Paired&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span> <span class="ow">is</span> <span class="kc">True</span><span class="p">:</span>
    <span class="n">fastq_links</span> <span class="o">=</span> <span class="n">link_fq</span><span class="p">(</span>
        <span class="n">design</span><span class="o">.</span><span class="n">Sample_id</span><span class="p">,</span>
        <span class="n">design</span><span class="o">.</span><span class="n">Upstream_file</span><span class="p">,</span>
        <span class="n">design</span><span class="o">.</span><span class="n">Downstream_file</span>
    <span class="p">)</span>
<span class="k">else</span><span class="p">:</span>
        <span class="n">fastq_links</span> <span class="o">=</span> <span class="n">link_fq</span><span class="p">(</span>
        <span class="n">design</span><span class="o">.</span><span class="n">Sample_id</span><span class="p">,</span>
        <span class="n">design</span><span class="o">.</span><span class="n">Upstream_file</span>
    <span class="p">)</span>


<span class="n">ruleorder</span><span class="p">:</span> <span class="n">bismark_methylation_extractor_se_multisample</span> <span class="o">&gt;</span> <span class="n">bismark_methylation_extractor_pe_multisample</span>

<span class="c1">##############</span>
<span class="c1">### Target ###</span>
<span class="c1">##############</span>

<span class="n">rule</span> <span class="n">target</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="c1"># bismark=expand(</span>
        <span class="c1">#     &quot;bismark/report/{sample}.SE.html&quot;,</span>
        <span class="c1">#     sample=design[&quot;Sample_id&quot;].tolist()</span>
        <span class="c1"># ),</span>
        <span class="n">multiqc</span><span class="o">=</span><span class="s2">&quot;results/MiraSeq.html&quot;</span><span class="p">,</span>
        <span class="c1">#multisample_cov=&quot;bismark_multisample/meth_cpg/multisample.bismark.cov.gz&quot;,</span>
        <span class="c1">#multisample_report=&#39;bismark_multisample/meth/multisample_SE_splitting_report.txt&#39;</span>


<span class="c1">####################################</span>
<span class="c1">### FastQ Screen and MultiQC     ###</span>
<span class="c1">####################################</span>


<span class="n">rule</span> <span class="n">multiqc</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">html</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;fastp/html/pe/</span><span class="si">{sample}</span><span class="s2">.fastp.html&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">design</span><span class="p">[</span><span class="s2">&quot;Sample_id&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">json</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;fastp/json/pe/</span><span class="si">{sample}</span><span class="s2">.fastp.json&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">design</span><span class="p">[</span><span class="s2">&quot;Sample_id&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">txt</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;fastq_screen/</span><span class="si">{sample}</span><span class="s2">.fastq_screen.txt&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">design</span><span class="p">[</span><span class="s2">&quot;Sample_id&quot;</span><span class="p">],</span>
            <span class="n">stream</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">png</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;fastq_screen/</span><span class="si">{sample}</span><span class="s2">.fastq_screen.png&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">design</span><span class="p">[</span><span class="s2">&quot;Sample_id&quot;</span><span class="p">],</span>
            <span class="n">stream</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">mapping_qc</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;stats/</span><span class="si">{sample}{ext}</span><span class="s2">&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">design</span><span class="p">[</span><span class="s2">&quot;Sample_id&quot;</span><span class="p">],</span>
            <span class="n">ext</span><span class="o">=</span><span class="p">[</span>
                <span class="s2">&quot;.alignment_summary_metrics&quot;</span><span class="p">,</span>
                <span class="c1">#&quot;.insert_size_metrics&quot;,</span>
                <span class="c1">#&quot;.insert_size_histogram.pdf&quot;,</span>
                <span class="s2">&quot;.quality_distribution_metrics&quot;</span><span class="p">,</span>
                <span class="s2">&quot;.quality_distribution.pdf&quot;</span><span class="p">,</span>
                <span class="c1">#&quot;.quality_by_cycle_metrics&quot;,</span>
                <span class="c1">#&quot;.quality_by_cycle.pdf&quot;,</span>
                <span class="c1">#&quot;.base_distribution_by_cycle_metrics&quot;,</span>
                <span class="c1">#&quot;.base_distribution_by_cycle.pdf&quot;,</span>
                <span class="s2">&quot;.gc_bias.detail_metrics&quot;</span><span class="p">,</span>
                <span class="s2">&quot;.gc_bias.summary_metrics&quot;</span><span class="p">,</span>
                <span class="s2">&quot;.gc_bias.pdf&quot;</span><span class="p">,</span>
                <span class="c1">#&quot;.rna_metrics&quot;,</span>
                <span class="c1">#&quot;.bait_bias_detail_metrics&quot;,</span>
                <span class="c1">#&quot;.bait_bias_summary_metrics&quot;,</span>
                <span class="c1">#&quot;.error_summary_metrics&quot;,</span>
                <span class="c1">#&quot;.pre_adapter_detail_metrics&quot;,</span>
                <span class="c1">#&quot;.pre_adapter_summary_metrics&quot;,</span>
                <span class="c1">#&quot;.quality_yield_metrics&quot;</span>
            <span class="p">]</span>
        <span class="p">)</span>
        <span class="c1"># bismark_alignment_report=expand(</span>
        <span class="c1">#     &quot;bismark/bams/{sample}_SE_report.txt&quot;,</span>
        <span class="c1">#     sample=design[&quot;Sample_id&quot;].tolist()</span>
        <span class="c1"># ),</span>
        <span class="c1"># bismark_nucleotide_report=expand(</span>
        <span class="c1">#     &quot;bismark/bams/{sample}.SE.nucleotide_stats.txt&quot;,</span>
        <span class="c1">#     sample=design[&quot;Sample_id&quot;].tolist()</span>
        <span class="c1"># ),</span>
        <span class="c1"># bismark_dedup_report=expand(</span>
        <span class="c1">#     &quot;bismark/deduplicated/{sample}.SE.deduplication_report.txt&quot;,</span>
        <span class="c1">#     sample=design[&quot;Sample_id&quot;].tolist()</span>
        <span class="c1"># ),</span>
        <span class="c1"># bismark_mbias_report=expand(</span>
        <span class="c1">#     &quot;bismark/meth/{sample}.SE.M-bias.txt&quot;,</span>
        <span class="c1">#     sample=design[&quot;Sample_id&quot;].tolist()</span>
        <span class="c1"># ),</span>
        <span class="c1"># bismark_splitting_report=expand(</span>
        <span class="c1">#     &quot;bismark/meth/{sample}_SE_splitting_report.txt&quot;,</span>
        <span class="c1">#     sample=design[&quot;Sample_id&quot;].tolist()</span>
        <span class="c1"># )</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">report</span><span class="p">(</span>
            <span class="s2">&quot;results/MiraSeq.html&quot;</span><span class="p">,</span>
            <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;../common/reports/multiqc.rst&quot;</span><span class="p">,</span>
            <span class="n">category</span><span class="o">=</span><span class="s2">&quot;Quality Controls&quot;</span>
        <span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Aggregating quality reports from Fastp and Salmon&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">1536</span><span class="p">,</span> <span class="mi">10240</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">35</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/multiqc.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/multiqc&quot;</span>


<span class="n">rule</span> <span class="n">fastq_screen</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;reads/</span><span class="si">{sample}</span><span class="s2">.fq.gz&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">txt</span><span class="o">=</span><span class="s2">&quot;fastq_screen/</span><span class="si">{sample}</span><span class="s2">.fastq_screen.txt&quot;</span><span class="p">,</span>
        <span class="n">png</span><span class="o">=</span><span class="s2">&quot;fastq_screen/</span><span class="si">{sample}</span><span class="s2">.fastq_screen.png&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Assessing quality of </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="n">config</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;threads&quot;</span><span class="p">,</span> <span class="mi">20</span><span class="p">)</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span> <span class="o">*</span> <span class="mi">8</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">50</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">fastq_screen_config</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;fastq_screen&quot;</span><span class="p">],</span>
        <span class="n">subset</span><span class="o">=</span><span class="mi">100000</span><span class="p">,</span>
        <span class="n">aligner</span><span class="o">=</span><span class="s1">&#39;bowtie2&#39;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/fastq_screen/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/fastq_screen&quot;</span>


<span class="c1">#####################</span>
<span class="c1">### Methylation ? ###</span>
<span class="c1">#####################</span>


<span class="n">bismark_config</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;genome&quot;</span><span class="p">:</span> <span class="n">config</span><span class="p">[</span><span class="s2">&quot;ref&quot;</span><span class="p">][</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
    <span class="s2">&quot;paired&quot;</span><span class="p">:</span> <span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;Paired&quot;</span><span class="p">,</span> <span class="kc">True</span><span class="p">),</span>
    <span class="s2">&quot;samples&quot;</span><span class="p">:</span> <span class="n">design</span><span class="p">[</span><span class="s2">&quot;Sample_id&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="p">}</span>

<span class="n">module</span> <span class="n">bismark</span><span class="p">:</span>
    <span class="n">snakefile</span><span class="p">:</span> <span class="s2">&quot;../../meta/bio/bismark/test/Snakefile&quot;</span>
    <span class="n">config</span><span class="p">:</span> <span class="n">bismark_config</span>


<span class="n">use</span> <span class="n">rule</span> <span class="o">*</span> <span class="kn">from</span> <span class="nn">bismark</span> <span class="k">as</span> <span class="o">*</span>


<span class="n">module</span> <span class="n">bismark_multisample</span><span class="p">:</span>
    <span class="n">snakefile</span><span class="p">:</span> <span class="s2">&quot;../../meta/bio/bismark_multisample/test/Snakefile&quot;</span>
    <span class="n">config</span><span class="p">:</span> <span class="n">bismark_config</span>


<span class="n">use</span> <span class="n">rule</span> <span class="o">*</span> <span class="kn">from</span> <span class="nn">bismark_multisample</span>


<span class="c1">################</span>
<span class="c1">###  Mapping ###</span>
<span class="c1">################</span>

<span class="n">rule</span> <span class="n">collect_multiple_metrics</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
         <span class="n">bam</span><span class="o">=</span><span class="s2">&quot;samtools/sort/</span><span class="si">{sample}</span><span class="s2">.bam&quot;</span><span class="p">,</span>
         <span class="n">bam_index</span><span class="o">=</span><span class="s2">&quot;samtools/sort/</span><span class="si">{sample}</span><span class="s2">.bam.bai&quot;</span><span class="p">,</span>
         <span class="n">ref</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;ref&quot;</span><span class="p">][</span><span class="s2">&quot;genome&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">multiext</span><span class="p">(</span><span class="s2">&quot;stats/</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>
                 <span class="s2">&quot;.alignment_summary_metrics&quot;</span><span class="p">,</span>
                 <span class="c1">#&quot;.insert_size_metrics&quot;,</span>
                 <span class="c1">#&quot;.insert_size_histogram.pdf&quot;,</span>
                 <span class="s2">&quot;.quality_distribution_metrics&quot;</span><span class="p">,</span>
                 <span class="s2">&quot;.quality_distribution.pdf&quot;</span><span class="p">,</span>
                 <span class="c1">#&quot;.quality_by_cycle_metrics&quot;,</span>
                 <span class="c1">#&quot;.quality_by_cycle.pdf&quot;,</span>
                 <span class="c1">#&quot;.base_distribution_by_cycle_metrics&quot;,</span>
                 <span class="c1">#&quot;.base_distribution_by_cycle.pdf&quot;,</span>
                 <span class="s2">&quot;.gc_bias.detail_metrics&quot;</span><span class="p">,</span>
                 <span class="s2">&quot;.gc_bias.summary_metrics&quot;</span><span class="p">,</span>
                 <span class="s2">&quot;.gc_bias.pdf&quot;</span><span class="p">,</span>
                 <span class="c1">#&quot;.rna_metrics&quot;,</span>
                 <span class="c1">#&quot;.bait_bias_detail_metrics&quot;,</span>
                 <span class="c1">#&quot;.bait_bias_summary_metrics&quot;,</span>
                 <span class="c1">#&quot;.error_summary_metrics&quot;,</span>
                 <span class="c1">#&quot;.pre_adapter_detail_metrics&quot;,</span>
                 <span class="c1">#&quot;.pre_adapter_summary_metrics&quot;,</span>
                 <span class="c1">#&quot;.quality_yield_metrics&quot;</span>
                 <span class="p">)</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span> <span class="o">*</span> <span class="mi">8</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/picard/multiple_metrics/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="c1">#&quot;VALIDATION_STRINGENCY=LENIENT &quot;</span>
        <span class="c1">#&quot;METRIC_ACCUMULATION_LEVEL=null &quot;</span>
        <span class="c1">#&quot;METRIC_ACCUMULATION_LEVEL=SAMPLE &quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/picard/collectmultiplemetrics&quot;</span>

<span class="n">config_bwa_fixmate</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;threads&quot;</span><span class="p">:</span> <span class="mi">20</span><span class="p">,</span>
    <span class="s2">&quot;genome&quot;</span><span class="p">:</span> <span class="n">config</span><span class="p">[</span><span class="s2">&quot;ref&quot;</span><span class="p">][</span><span class="s2">&quot;genome&quot;</span><span class="p">]</span>
<span class="p">}</span>

<span class="n">module</span> <span class="n">bwa_fixmate</span><span class="p">:</span>
    <span class="n">snakefile</span><span class="p">:</span> <span class="s2">&quot;../../meta/bio/bwa_fixmate/test/Snakefile&quot;</span>
    <span class="n">config</span><span class="p">:</span> <span class="n">config_bwa_fixmate</span>

<span class="n">use</span> <span class="n">rule</span> <span class="o">*</span> <span class="kn">from</span> <span class="nn">bwa_fixmate</span>

<span class="n">use</span> <span class="n">rule</span> <span class="n">bwa_mem</span> <span class="kn">from</span> <span class="nn">bwa_fixmate</span> <span class="k">with</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">reads</span> <span class="o">=</span> <span class="s2">&quot;fastp/trimmed/pe/</span><span class="si">{sample}</span><span class="s2">.fastq&quot;</span><span class="p">,</span>
        <span class="n">index</span><span class="o">=</span><span class="n">multiext</span><span class="p">(</span>
            <span class="s2">&quot;bwa_mem2/index/genome&quot;</span><span class="p">,</span> <span class="s2">&quot;.0123&quot;</span><span class="p">,</span> <span class="s2">&quot;.amb&quot;</span><span class="p">,</span> <span class="s2">&quot;.ann&quot;</span><span class="p">,</span> <span class="s2">&quot;.pac&quot;</span>
        <span class="p">)</span>


<span class="c1">############################</span>
<span class="c1">### FASTP FASTQ CLEANING ###</span>
<span class="c1">############################</span>

<span class="n">rule</span> <span class="n">fastp_clean</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">sample</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;reads/</span><span class="si">{sample}</span><span class="s2">.fq.gz&quot;</span><span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">trimmed</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;fastp/trimmed/pe/</span><span class="si">{sample}</span><span class="s2">.fastq&quot;</span><span class="p">),</span>
        <span class="n">html</span><span class="o">=</span><span class="s2">&quot;fastp/html/pe/</span><span class="si">{sample}</span><span class="s2">.fastp.html&quot;</span><span class="p">,</span>
        <span class="n">json</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;fastp/json/pe/</span><span class="si">{sample}</span><span class="s2">.fastp.json&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Cleaning </span><span class="si">{wildcards.sample}</span><span class="s2"> with Fastp&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">4096</span><span class="p">,</span> <span class="mi">15360</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">adapters</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;fastp_adapters&quot;</span><span class="p">,</span> <span class="kc">None</span><span class="p">),</span>
        <span class="n">extra</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;fastp_extra&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/fastp/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/fastp&quot;</span>


<span class="c1">#################################################</span>
<span class="c1">### Gather files from iRODS or mounting point ###</span>
<span class="c1">#################################################</span>

<span class="n">rule</span> <span class="n">bigr_copy</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;reads/</span><span class="si">{sample}</span><span class="s2">.fq.gz&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Gathering </span><span class="si">{wildcards.sample}</span><span class="s2"> fastq file&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span> <span class="mi">2048</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="nb">input</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">output</span><span class="p">:</span> <span class="n">fastq_links</span><span class="p">[</span><span class="n">output</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bigr_copy/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/BiGR/copy&quot;</span>
</pre></div>
</div>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="low_pass_wgs_copy_number.html" class="btn btn-neutral float-left" title="LOW_PASS_WGS_COPY_NUMBER" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="nfcore_chipseq.html" class="btn btn-neutral float-right" title="NFCORE-CHIPSEQ" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>