<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>EACON_ONCOSCAN &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="RETROTRANSPOSONS" href="retrotransposons.html" />
    <link rel="prev" title="NFCORE-CHIPSEQ" href="nfcore_chipseq.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="CTC_Variant_Calling_Integragen_Legacy.html">CTC_VARIANT_CALLING_INTEGRAGEN_LEGACY</a></li>
<li class="toctree-l2"><a class="reference internal" href="annot_sv.html">ANNOT_SV</a></li>
<li class="toctree-l2"><a class="reference internal" href="bcl2fastq.html">BCL2FASTQ (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="bs_ampliseq.html">BS_AMPLISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="celltype_deconvolution_rnaseq.html">CELLTYPE_DECONVOLUTION_RNASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="chipseq.html">CHIPSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="cytoscan_eacon.html">EACON_CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="design_builder.html">DESIGN_BUILDER</a></li>
<li class="toctree-l2"><a class="reference internal" href="dge_deseq2.html">DGE_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastqc_multiqc.html">FASTQC_MULTIQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_set_enrichment.html">GENE_SET_ENRICHMENT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="low_pass_wgs_copy_number.html">LOW_PASS_WGS_COPY_NUMBER</a></li>
<li class="toctree-l2"><a class="reference internal" href="make_pon.html">MAKE_PON</a></li>
<li class="toctree-l2"><a class="reference internal" href="miraseq.html">MIRASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="nfcore_chipseq.html">NFCORE-CHIPSEQ</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">EACON_ONCOSCAN</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l3"><a class="reference internal" href="#input-output">Input/Output</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-meta-wrappers">Used meta-wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#snakefile">Snakefile</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="retrotransposons.html">RETROTRANSPOSONS</a></li>
<li class="toctree-l2"><a class="reference internal" href="rnaseq.html">RNASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon_quant.html">SALMON_QUANT</a></li>
<li class="toctree-l2"><a class="reference internal" href="seqoia.html">SEQOIA SINGLE-SAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="sigprofiler_signatures.html">SIGPROFILER_SIGNATURES (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_snpsift.html">SNPEFF_SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_ampliseq.html">VARIANT_CALLING_AMPLISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_germline.html">VARIANT_CALLING_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_occurence_cohort.html">VARIANT_OCCURENCE_COHORT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="wes_somatic.html">VARIANT_CALLING_SOMATIC</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../pipelines.html">BiGR Pipelines</a> &raquo;</li>
      <li>EACON_ONCOSCAN</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/bigr-pipelines/oncoscan_eacon.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="eacon-oncoscan">
<span id="id1"></span><h1>EACON_ONCOSCAN<a class="headerlink" href="#eacon-oncoscan" title="Permalink to this heading"></a></h1>
<p>Analyse OncoScans with EaCoN on Flamingo</p>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<p>In order to run the pipeline, use the following commands</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Go to your directory containing CEL files</span>

<span class="nb">cd</span> /path/to/CEL/dir

<span class="c1"># Copy/paster the following line</span>

bash /mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/oncoscan_eacon/run.sh
</pre></div>
</div>
</section>
<section id="input-output">
<h2>Input/Output<a class="headerlink" href="#input-output" title="Permalink to this heading"></a></h2>
<p><strong>Input:</strong></p>
<ul class="simple">
<li><p>CEL files</p></li>
</ul>
<p><strong>Output:</strong></p>
<ul class="simple">
<li><p>Complete EaCoN anaysis directory</p></li>
<li><p>HTML report</p></li>
</ul>
</section>
<section id="used-meta-wrappers">
<h2>Used meta-wrappers<a class="headerlink" href="#used-meta-wrappers" title="Permalink to this heading"></a></h2>
<p>The following individual meta-wrappers are used in this pipeline:</p>
<ul class="simple">
<li><p><span class="xref std std-ref">meta/bio/eacon_oncoscan</span></p></li>
<li><p><span class="xref std std-ref">meta/bio/eacon_post_process</span></p></li>
</ul>
<p>Please refer to each meta-wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="snakefile">
<h2>Snakefile<a class="headerlink" href="#snakefile" title="Permalink to this heading"></a></h2>
<p>The pipeline contains the following steps:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">snakemake.utils</span> <span class="kn">import</span> <span class="n">min_version</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>
<span class="kn">from</span> <span class="nn">yaml</span> <span class="kn">import</span> <span class="n">dump</span>
<span class="n">min_version</span><span class="p">(</span><span class="s2">&quot;6.0&quot;</span><span class="p">)</span>


<span class="c1"># If not config file is provided, a default configfile will be created.</span>
<span class="n">default_oncoscan_process_config</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;samples&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;&quot;</span><span class="p">],</span>
    <span class="s2">&quot;params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;arraytype&quot;</span><span class="p">:</span> <span class="s2">&quot;OncoScan_CNV&quot;</span><span class="p">,</span>
        <span class="s2">&quot;genome&quot;</span><span class="p">:</span> <span class="s2">&quot;hg19&quot;</span><span class="p">,</span>
        <span class="s2">&quot;nar&quot;</span><span class="p">:</span> <span class="s2">&quot;na33.r2&quot;</span><span class="p">,</span>
        <span class="s2">&quot;segmenter&quot;</span><span class="p">:</span> <span class="s2">&quot;ASCAT&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;extra&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;EaCoN_process&quot;</span><span class="p">:</span> <span class="s1">&#39;apt.build = &quot;na33.r2&quot;, force = TRUE&#39;</span><span class="p">,</span>
        <span class="s2">&quot;EaCoN_segment&quot;</span><span class="p">:</span> <span class="p">(</span>
            <span class="s1">&#39;segmenter = &quot;ASCAT&quot;, smooth.k = NULL, BAF.filter = 0.9, &#39;</span>
            <span class="s1">&#39;SER.pen = 40, nrf = 0.5, penalty = 50, force = TRUE&#39;</span>
        <span class="p">),</span>
        <span class="s2">&quot;EaCoN_ascn&quot;</span><span class="p">:</span> <span class="s1">&#39;force = TRUE&#39;</span><span class="p">,</span>
        <span class="s2">&quot;EaCoN_gis&quot;</span><span class="p">:</span> <span class="s2">&quot;&quot;</span>
    <span class="p">}</span>
<span class="p">}</span>

<span class="n">config_path</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s2">&quot;config.yaml&quot;</span><span class="p">)</span>
<span class="k">if</span> <span class="ow">not</span> <span class="n">config_path</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
    <span class="c1"># Search for CEL files in order to get sample names</span>
    <span class="n">sample_identifiers</span> <span class="o">=</span> <span class="p">[</span>
        <span class="n">cel</span><span class="o">.</span><span class="n">name</span><span class="p">[:</span><span class="o">-</span><span class="nb">len</span><span class="p">(</span><span class="s2">&quot;_A.CEL&quot;</span><span class="p">)]</span>
        <span class="k">for</span> <span class="n">cel</span> <span class="ow">in</span> <span class="n">config_path</span><span class="o">.</span><span class="n">parent</span><span class="o">.</span><span class="n">iterdir</span><span class="p">()</span>
        <span class="k">if</span> <span class="n">cel</span><span class="o">.</span><span class="n">name</span><span class="o">.</span><span class="n">endswith</span><span class="p">((</span><span class="s2">&quot;_A.CEL&quot;</span><span class="p">,</span> <span class="s2">&quot;_B.CEL&quot;</span><span class="p">))</span>
    <span class="p">]</span>
    <span class="c1"># Update config file with sample names</span>
    <span class="n">default_oncoscan_process_config</span><span class="p">[</span><span class="s2">&quot;samples&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">sample_identifiers</span>

    <span class="c1"># Save config file</span>
    <span class="k">with</span> <span class="n">config_path</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="s2">&quot;w&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">config_stream</span><span class="p">:</span>
        <span class="n">config_stream</span><span class="o">.</span><span class="n">write</span><span class="p">(</span>
            <span class="n">dump</span><span class="p">(</span><span class="n">default_oncoscan_process_config</span><span class="p">,</span> <span class="n">default_flow_style</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
        <span class="p">)</span>

<span class="c1"># The configfile name is hard coded. It should be: &quot;config.yaml&quot;</span>
<span class="n">configfile</span><span class="p">:</span> <span class="nb">str</span><span class="p">(</span><span class="n">config_path</span><span class="p">)</span>
<span class="c1"># ruleorder: eacon_annotate &gt; post_process_eacon_annotate</span>

<span class="n">module</span> <span class="n">post_process_eacon</span><span class="p">:</span>
    <span class="n">snakefile</span><span class="p">:</span> <span class="s2">&quot;../../meta/bio/eacon_post_process/test/Snakefile&quot;</span>
    <span class="n">config</span><span class="p">:</span> <span class="n">config</span>


<span class="nb">print</span><span class="p">(</span><span class="n">config</span><span class="p">)</span>


<span class="n">wildcard_constraints</span><span class="p">:</span>
    <span class="n">ATchannel</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;A|B&quot;</span><span class="p">,</span>
    <span class="n">GCchannel</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;C|D&quot;</span><span class="p">,</span>
    <span class="n">sample</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;|&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;samples&quot;</span><span class="p">])</span>


<span class="n">rule</span> <span class="n">default_oncoscan_process_all</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="c1"># EaCoN models</span>
        <span class="c1">#ascn = expand(</span>
        <span class="c1">#    os.sep.join([&quot;{sample}&quot;, config[&quot;params&quot;][&quot;segmenter&quot;],</span>
        <span class="c1">#                 &quot;ASCN&quot;, &quot;{sample}.gammaEval.png&quot;]),</span>
        <span class="c1">#    sample=config[&quot;samples&quot;]</span>
        <span class="c1">#),</span>
        <span class="c1"># EaCoN annotate</span>
        <span class="n">html</span> <span class="o">=</span> <span class="n">expand</span><span class="p">(</span>
            <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">sep</span><span class="o">.</span><span class="n">join</span><span class="p">([</span>
                <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">][</span><span class="s2">&quot;segmenter&quot;</span><span class="p">],</span> <span class="s2">&quot;L2R&quot;</span><span class="p">,</span>
                <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">.REPORT.html&quot;</span>
            <span class="p">]),</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;samples&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="c1"># EaCoN new instability scoring feature</span>
        <span class="n">instability</span> <span class="o">=</span> <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">_GIS_from_best_gamma.txt&quot;</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;samples&quot;</span><span class="p">]</span>
        <span class="p">)</span>

<span class="c1"># Import all rules from the eacon_post_process meta wrapper</span>
<span class="n">use</span> <span class="n">rule</span> <span class="o">*</span> <span class="kn">from</span> <span class="nn">post_process_eacon</span>


<span class="n">use</span> <span class="n">rule</span> <span class="n">eacon_annotate</span> <span class="kn">from</span> <span class="nn">post_process_eacon</span> <span class="k">with</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">rds</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">/ASCAT/L2R/</span><span class="si">{sample}</span><span class="s2">.SEG.ASCAT.RDS&quot;</span><span class="p">,</span>
        <span class="n">grd</span> <span class="o">=</span> <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/scripts/grd&quot;</span><span class="p">,</span>
        <span class="n">ldb</span> <span class="o">=</span> <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/databases&quot;</span>



<span class="n">rule</span> <span class="n">eacon_oncoscan_process</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="c1">#install = &#39;install.ok&#39;,</span>
        <span class="n">ATChannelCel</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">_A.CEL&quot;</span><span class="p">,</span>
        <span class="n">GCChannelCel</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">_C.CEL&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">qc_txt</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">_2.4.0_</span><span class="si">{nar}</span><span class="s2">.qc.txt&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">nar</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">][</span><span class="s2">&quot;nar&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">log</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">_2.4.0_</span><span class="si">{nar}</span><span class="s2">.log&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">nar</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">][</span><span class="s2">&quot;nar&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">txt</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">_pairs.txt&quot;</span><span class="p">,</span>
        <span class="n">png</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">_</span><span class="si">{arraytype}</span><span class="s2">_</span><span class="si">{genome}</span><span class="s2">_rawplot.png&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>
            <span class="n">arraytype</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">][</span><span class="s2">&quot;arraytype&quot;</span><span class="p">],</span>
            <span class="n">genome</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">][</span><span class="s2">&quot;genome&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">rds</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">_</span><span class="si">{arraytype}</span><span class="s2">_</span><span class="si">{genome}</span><span class="s2">_processed.RDS&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span>
            <span class="n">genome</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">][</span><span class="s2">&quot;genome&quot;</span><span class="p">],</span>
            <span class="n">arraytype</span> <span class="o">=</span> <span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">][</span><span class="s2">&quot;arraytype&quot;</span><span class="p">]</span>
        <span class="p">),</span>
        <span class="n">pairs</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">_2.4.0_</span><span class="si">{nar}</span><span class="s2">.paircheck.txt&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">nar</span><span class="o">=</span><span class="n">config</span><span class="p">[</span><span class="s2">&quot;params&quot;</span><span class="p">][</span><span class="s2">&quot;nar&quot;</span><span class="p">]</span>
        <span class="p">)</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">50</span><span class="p">,</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">5</span> <span class="o">*</span> <span class="mi">1024</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">extra</span> <span class="o">=</span> <span class="n">config</span><span class="p">[</span><span class="s2">&quot;extra&quot;</span><span class="p">][</span><span class="s2">&quot;EaCoN_process&quot;</span><span class="p">]</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/EaCoN/</span><span class="si">{sample}</span><span class="s2">/oncoscan_process.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/eacon/oncoscan_process&quot;</span>

<span class="n">use</span> <span class="n">rule</span> <span class="n">eacon_oncoscan_process</span> <span class="k">as</span> <span class="n">eacon_oncoscan_process_bd</span> <span class="k">with</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">ATChannelCel</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">_B.CEL&quot;</span><span class="p">,</span>
        <span class="n">GCChannelCel</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">_D.CEL&quot;</span>



<span class="n">rule</span> <span class="n">eacon_install</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">r_packages</span> <span class="o">=</span> <span class="p">[</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/affy.CN.norm.data_0.1.2.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/CytoScan750K.Array.na33.r4_0.1.0.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/CytoScan750K.Array.na36.r1_0.1.0.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/CytoScanHD.Array.na33.r4_0.1.0.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/CytoScanHD.Array.na36.r1_0.1.0.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/OncoScanCNV.na33.r2_0.1.0.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/OncoScanCNV.na36.r1_0.1.0.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/OncoScan.na33.r4_0.1.0.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/OncoScan.na36.r1_0.1.0.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/rcnorm_0.1.5.tar.gz&quot;</span>
        <span class="p">],</span>
        <span class="n">git_packages</span> <span class="o">=</span> <span class="p">[</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/EaCoN_0.50.4-2418-g8fefa1e53.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/EaCoN_Chromosomes.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/apt.cytoscan.2.4.0.tar.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;/mnt/beegfs/database/bioinfo/Index_DB/EaCoN/packages/apt.oncoscan.2.4.0.tar.gz&quot;</span>
        <span class="p">]</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="n">touch</span><span class="p">(</span><span class="s1">&#39;install.ok&#39;</span><span class="p">))</span>
    <span class="n">cache</span><span class="p">:</span> <span class="kc">True</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">time_min</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">480</span><span class="p">,</span>
        <span class="n">mem_mb</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">4096</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/EaCoN/install.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/eacon/install_local&quot;</span>
</pre></div>
</div>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
<li><p>Bastien Job</p></li>
<li><p>Gérôme Jules-Clément</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="nfcore_chipseq.html" class="btn btn-neutral float-left" title="NFCORE-CHIPSEQ" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="retrotransposons.html" class="btn btn-neutral float-right" title="RETROTRANSPOSONS" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>