<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>SALMON_QUANT &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="SEQOIA SINGLE-SAMPLE" href="seqoia.html" />
    <link rel="prev" title="RNASEQ" href="rnaseq.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="CTC_Variant_Calling_Integragen_Legacy.html">CTC_VARIANT_CALLING_INTEGRAGEN_LEGACY</a></li>
<li class="toctree-l2"><a class="reference internal" href="annot_sv.html">ANNOT_SV</a></li>
<li class="toctree-l2"><a class="reference internal" href="bcl2fastq.html">BCL2FASTQ (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="bs_ampliseq.html">BS_AMPLISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="celltype_deconvolution_rnaseq.html">CELLTYPE_DECONVOLUTION_RNASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="chipseq.html">CHIPSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="cytoscan_eacon.html">EACON_CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="design_builder.html">DESIGN_BUILDER</a></li>
<li class="toctree-l2"><a class="reference internal" href="dge_deseq2.html">DGE_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastqc_multiqc.html">FASTQC_MULTIQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_set_enrichment.html">GENE_SET_ENRICHMENT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="low_pass_wgs_copy_number.html">LOW_PASS_WGS_COPY_NUMBER</a></li>
<li class="toctree-l2"><a class="reference internal" href="make_pon.html">MAKE_PON</a></li>
<li class="toctree-l2"><a class="reference internal" href="miraseq.html">MIRASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="nfcore_chipseq.html">NFCORE-CHIPSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="oncoscan_eacon.html">EACON_ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="retrotransposons.html">RETROTRANSPOSONS</a></li>
<li class="toctree-l2"><a class="reference internal" href="rnaseq.html">RNASEQ</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">SALMON_QUANT</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l3"><a class="reference internal" href="#input-output">Input/Output</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#snakefile">Snakefile</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="seqoia.html">SEQOIA SINGLE-SAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="sigprofiler_signatures.html">SIGPROFILER_SIGNATURES (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_snpsift.html">SNPEFF_SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_ampliseq.html">VARIANT_CALLING_AMPLISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_germline.html">VARIANT_CALLING_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_occurence_cohort.html">VARIANT_OCCURENCE_COHORT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="wes_somatic.html">VARIANT_CALLING_SOMATIC</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../pipelines.html">BiGR Pipelines</a> &raquo;</li>
      <li>SALMON_QUANT</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/bigr-pipelines/salmon_quant.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="salmon-quant">
<span id="id1"></span><h1>SALMON_QUANT<a class="headerlink" href="#salmon-quant" title="Permalink to this heading"></a></h1>
<p>Perform trimming and quantification on RNASeq</p>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<p>In order to run the pipeline, use the following commands</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Go to your working directory</span>

<span class="nb">cd</span> /path/to/my/working/directory

<span class="c1"># Build a design file (see below)</span>

<span class="c1"># Copy/paste the following line for **HG19**</span>

bash /mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/salmon_quant/run.sh hg19

<span class="c1"># Copy/paste the following line for **HG38**</span>

bash /mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/salmon_quant/run.sh hg38
</pre></div>
</div>
</section>
<section id="input-output">
<h2>Input/Output<a class="headerlink" href="#input-output" title="Permalink to this heading"></a></h2>
<p><strong>Input:</strong></p>
<ul class="simple">
<li><p>Fastq files</p></li>
<li><p>Fasta-formatted Genome sequence</p></li>
<li><p>Fasta-formatted transcriptome sequence</p></li>
<li><p>GTF formatted genome annotation</p></li>
</ul>
<p><strong>Output:</strong></p>
<ul class="simple">
<li><p>Salmon quantification (Raw.genes.tsv + TMP.genes.tsv + TPM.transcripts.tsv)</p></li>
<li><p>MultiQC report handling Quality controls</p></li>
<li><p>Trimmed fastq files</p></li>
</ul>
</section>
<section id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this heading"></a></h2>
<p>Prerequisites:</p>
<ul class="simple">
<li><p>A TSV formatted design file, <em>named ‘design.tsv’</em> with the following columns:</p></li>
</ul>
<table class="colwidths-given docutils align-default" id="id2">
<caption><span class="caption-text">Desgin file format</span><a class="headerlink" href="#id2" title="Permalink to this table"></a></caption>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Sample_id</p></th>
<th class="head"><p>Upstream_fastq</p></th>
<th class="head"><p>Downstream_fastq</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Name of the Sample1</p></td>
<td><p>Path to upstream fastq file</p></td>
<td><p>Path to downstream fastq file</p></td>
</tr>
<tr class="row-odd"><td><p>Name of the Sample2</p></td>
<td><p>Path to upstream fastq file</p></td>
<td><p>Path to downstream fastq file</p></td>
</tr>
<tr class="row-even"><td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<p>Warning: If available, the salmon index passed in configuration will be used, wather you changed the command line arguments or not.</p>
<p>How does it work ?</p>
<ol class="arabic simple">
<li><p>Gathering Fastq files</p></li>
</ol>
<p>This pipeline copies files from iRODS, or symlinks them. In your design file,
the <cite>Sample_id</cite> is used to rename your fastq file and make them all easily
recognisable. In case of iRODS copy, the checksum is automatically computed
and verified. In cas of a copy from cold to hot storage, then a checksum is
automatically computed and verified. Elsewise, a simple sylmink is done and
no control needs to be performed.</p>
<p>The column Upstream_file/Downstream_file identifies reads’ streams.
If the sequencing was not oriented, then order does not matter.
Otherwise, make sure R1 reads are under Upstream_file, and R2 reads under
Downstream_file.</p>
<p>You may need to concatenate several fastq files into one single fastq file
for a given sample: in case of lane splitting, run splitting, and/or
resequencing. This may be done automatically! Under the corresponding column,
separate the multiple files by a comma (<cite>,</cite>).</p>
<ol class="arabic simple" start="2">
<li><p>Cleaning fastq files</p></li>
</ol>
<p>These (concatenated?) fastq files are trimmed with fastp. By default, the
cleaning is a bit more relaxed than default fastp parameters. We’re woring on
RNA-Seq with Salmon in this pipeline. No need to be very strict.</p>
<p>The running window has been increased to 6 nucleotides, the minimum mean read
quality was raised to 10. The unqualified percent limit was raised to 50% and
the maximum of N bases was raised to 7. The minimum length was lowered to 15.
Because we are trimming RNA-Seq, we analyse overrepresented sequences.</p>
<p>Nothing dramatic! Bad reads will be filtered later if needed.</p>
<p>By the way, FastQ Screen is used over the raw reads, since possible sequencing
artifacts are interesting in this step. Many genomes are tested, see section
5 to look at the results.</p>
<ol class="arabic simple" start="3">
<li><p>Genome indexation</p></li>
</ol>
<p>In parallel of this trimming, a genome indexation is done for your analysis.
Taking your read length into account, we build an ideal index using decoy
sequences and best hash seed size for the selective alignment. Yes, you read
it, no more pseudo-mapping on transcriptome. We are doing selective alignment
over the decoy-aware gentrome! Welcome in 2019!</p>
<p>During this indexation step, duplicates are kept, since fellow biologists
always want all the targets to be tested.</p>
<ol class="arabic simple" start="4">
<li><p>Selective Alignment and Quantification</p></li>
</ol>
<p>Salmon is used to run the selctive alignment. It modelise the possible GC
bias, initial sequence bias and positional bias, since we empirically always
see them on our quality controls.</p>
<p>Mappings are validated, and 100 bootstraps are done over the quantification.</p>
<ol class="arabic simple" start="5">
<li><p>Quality report</p></li>
</ol>
<p>MultiQC is used to aggregate quality reports from Fastp and Salmon. Quite
handy, isn’t it?</p>
</section>
<section id="snakefile">
<h2>Snakefile<a class="headerlink" href="#snakefile" title="Permalink to this heading"></a></h2>
<p>The pipeline contains the following steps:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1">#####################################</span>
<span class="c1"># Check Snakemake version           #</span>
<span class="c1"># Import search-and-buil functions  #</span>
<span class="c1"># for config and design             #</span>
<span class="c1">#####################################</span>

<span class="kn">from</span> <span class="nn">snakemake.utils</span> <span class="kn">import</span> <span class="n">min_version</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>
<span class="kn">from</span> <span class="nn">yaml</span> <span class="kn">import</span> <span class="n">dump</span>

<span class="n">min_version</span><span class="p">(</span><span class="s2">&quot;7.5&quot;</span><span class="p">)</span>

<span class="kn">import</span> <span class="nn">sys</span>

<span class="n">worflow_source_dir</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">workflow</span><span class="o">.</span><span class="n">srcdir</span><span class="p">(</span><span class="s2">&quot;.&quot;</span><span class="p">))</span>
<span class="n">common</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">worflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;..&quot;</span> <span class="o">/</span> <span class="s2">&quot;common&quot;</span> <span class="o">/</span> <span class="s2">&quot;python&quot;</span><span class="p">)</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">common</span><span class="p">)</span>

<span class="kn">from</span> <span class="nn">file_manager</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">files_linker</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">write_yaml</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">messages</span> <span class="kn">import</span> <span class="n">message</span>

<span class="c1">#####################</span>
<span class="c1"># Setup environment #</span>
<span class="c1">#####################</span>

<span class="n">logging</span><span class="o">.</span><span class="n">basicConfig</span><span class="p">(</span>
    <span class="n">filename</span><span class="o">=</span><span class="s2">&quot;snakemake.salmon_quant.log&quot;</span><span class="p">,</span> <span class="n">filemode</span><span class="o">=</span><span class="s2">&quot;w&quot;</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="n">logging</span><span class="o">.</span><span class="n">DEBUG</span>
<span class="p">)</span>


<span class="n">default_config</span> <span class="o">=</span> <span class="n">read_yaml</span><span class="p">(</span><span class="n">worflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;config.hg38.yaml&quot;</span><span class="p">)</span>
<span class="n">configfile</span><span class="p">:</span> <span class="n">get_config</span><span class="p">(</span><span class="n">default_config</span><span class="p">)</span>
<span class="n">design</span> <span class="o">=</span> <span class="n">get_design</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">getcwd</span><span class="p">(),</span> <span class="n">search_fastq_pairs</span><span class="p">)</span>

<span class="c1">##################################</span>
<span class="c1"># Setup globals and fix wilcards #</span>
<span class="c1">##################################</span>

<span class="n">fastq_links</span> <span class="o">=</span> <span class="n">link_fq</span><span class="p">(</span><span class="n">design</span><span class="o">.</span><span class="n">Sample_id</span><span class="p">,</span> <span class="n">design</span><span class="o">.</span><span class="n">Upstream_file</span><span class="p">,</span> <span class="n">design</span><span class="o">.</span><span class="n">Downstream_file</span><span class="p">)</span>


<span class="n">wildcard_constraints</span><span class="p">:</span>
    <span class="n">sample</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;|&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">design</span><span class="o">.</span><span class="n">Sample_id</span><span class="o">.</span><span class="n">to_list</span><span class="p">()),</span>
    <span class="n">stream</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;|&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="nb">range</span><span class="p">(</span><span class="mi">3</span><span class="p">))),</span>


<span class="c1">#################################################</span>
<span class="c1">### Gather files from iRODS or mounting point ###</span>
<span class="c1">#################################################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/001.bigr_copy.smk&quot;</span>


<span class="c1">############################</span>
<span class="c1">### FASTP FASTQ CLEANING ###</span>
<span class="c1">############################</span>


<span class="c1"># See quality controls &amp; trimming:</span>
<span class="c1"># at: rules/001.qc.smk</span>


<span class="c1">########################</span>
<span class="c1">### Quality Controls ###</span>
<span class="c1">########################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/002.qc.smk&quot;</span>



<span class="c1">#############################</span>
<span class="c1">### Salmon quantification ###</span>
<span class="c1">#############################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/003.salmon.smk&quot;</span>

<span class="c1">########################</span>
<span class="c1">### Aggregate counts ###</span>
<span class="c1">########################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/004.aggregate.smk&quot;</span>


<span class="c1">#################</span>
<span class="c1">### Main rule ###</span>
<span class="c1">#################</span>


<span class="n">rule</span> <span class="n">target_salmon_quant</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;multiqc/MultiQC.html&quot;</span><span class="p">,</span>
        <span class="s2">&quot;salmon/TPM.genes.tsv&quot;</span><span class="p">,</span>
        <span class="s2">&quot;salmon/TPM.transcripts.tsv&quot;</span><span class="p">,</span>
        <span class="s2">&quot;salmon/Raw.genes.tsv&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">directory</span><span class="p">(</span><span class="s2">&quot;results_to_upload&quot;</span><span class="p">)</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">512</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">25</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/results_to_upload.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="s2">&quot;--verbose --checksum --human-readable&quot;</span>
    <span class="n">shell</span><span class="p">:</span>
        <span class="s2">&quot;rsync </span><span class="si">{params}</span><span class="s2"> </span><span class="si">{input}</span><span class="s2"> </span><span class="si">{output}</span><span class="s2"> &gt; </span><span class="si">{log}</span><span class="s2"> 2&gt;&amp;1&quot;</span>
</pre></div>
</div>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
<li><p>M boyba Diop</p></li>
<li><p>Marc Deloger</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="rnaseq.html" class="btn btn-neutral float-left" title="RNASEQ" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="seqoia.html" class="btn btn-neutral float-right" title="SEQOIA SINGLE-SAMPLE" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>