<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>SEQOIA SINGLE-SAMPLE &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="SIGPROFILER_SIGNATURES (UNDER DEVELOPMENT)" href="sigprofiler_signatures.html" />
    <link rel="prev" title="SALMON_QUANT" href="salmon_quant.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="CTC_Variant_Calling_Integragen_Legacy.html">CTC_VARIANT_CALLING_INTEGRAGEN_LEGACY</a></li>
<li class="toctree-l2"><a class="reference internal" href="annot_sv.html">ANNOT_SV</a></li>
<li class="toctree-l2"><a class="reference internal" href="bcl2fastq.html">BCL2FASTQ (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="bs_ampliseq.html">BS_AMPLISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="celltype_deconvolution_rnaseq.html">CELLTYPE_DECONVOLUTION_RNASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="chipseq.html">CHIPSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="cytoscan_eacon.html">EACON_CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="design_builder.html">DESIGN_BUILDER</a></li>
<li class="toctree-l2"><a class="reference internal" href="dge_deseq2.html">DGE_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastqc_multiqc.html">FASTQC_MULTIQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_set_enrichment.html">GENE_SET_ENRICHMENT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="low_pass_wgs_copy_number.html">LOW_PASS_WGS_COPY_NUMBER</a></li>
<li class="toctree-l2"><a class="reference internal" href="make_pon.html">MAKE_PON</a></li>
<li class="toctree-l2"><a class="reference internal" href="miraseq.html">MIRASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="nfcore_chipseq.html">NFCORE-CHIPSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="oncoscan_eacon.html">EACON_ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="retrotransposons.html">RETROTRANSPOSONS</a></li>
<li class="toctree-l2"><a class="reference internal" href="rnaseq.html">RNASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon_quant.html">SALMON_QUANT</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">SEQOIA SINGLE-SAMPLE</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l3"><a class="reference internal" href="#input-output">Input/Output</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#snakefile">Snakefile</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="sigprofiler_signatures.html">SIGPROFILER_SIGNATURES (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_snpsift.html">SNPEFF_SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_ampliseq.html">VARIANT_CALLING_AMPLISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_germline.html">VARIANT_CALLING_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_occurence_cohort.html">VARIANT_OCCURENCE_COHORT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="wes_somatic.html">VARIANT_CALLING_SOMATIC</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../pipelines.html">BiGR Pipelines</a> &raquo;</li>
      <li>SEQOIA SINGLE-SAMPLE</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/bigr-pipelines/seqoia.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="seqoia-single-sample">
<span id="id1"></span><h1>SEQOIA SINGLE-SAMPLE<a class="headerlink" href="#seqoia-single-sample" title="Permalink to this heading"></a></h1>
<p>Run seqOIA over single sample WES (tumor/normal) + RNA (tumor).</p>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<p>In order to run the pipeline, use the following commands</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Go to your working directory</span>

<span class="nb">cd</span> /path/to/my/working/directory

<span class="c1"># Build a design file (see below)</span>

<span class="c1"># Copy/paste the following line</span>

bash /mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/seqoia/run.sh
</pre></div>
</div>
</section>
<section id="input-output">
<h2>Input/Output<a class="headerlink" href="#input-output" title="Permalink to this heading"></a></h2>
<p><strong>Input:</strong></p>
<ul class="simple">
<li><p>Path to Fastq WES Tumor (R1 + R2)</p></li>
<li><p>Path to Fastq WES Normal (R1 + R2)</p></li>
<li><p>Path to Fastq RNA Tumor (R1 + R2)</p></li>
</ul>
<p><strong>Output:</strong></p>
<ul class="simple">
<li><p>seqOIA results</p></li>
</ul>
</section>
<section id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this heading"></a></h2>
<p>The following individual wrappers are used in this pipeline:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/BiGR/copy.html#bio-bigr-copy"><span class="std std-ref">IGR_COPY</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this heading"></a></h2>
<p>Prerequisites:</p>
<ul class="simple">
<li><p>A TSV formatted design file, <em>named ‘design.tsv’</em> with the following columns:</p></li>
</ul>
</section>
<section id="snakefile">
<h2>Snakefile<a class="headerlink" href="#snakefile" title="Permalink to this heading"></a></h2>
<p>The pipeline contains the following steps:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/usr/bin/env python3</span>
<span class="c1"># -*- coding: utf-8 -*-</span>

<span class="sd">&quot;&quot;&quot;This snakefile prepares each single sample run of SeqOIA and launch them&quot;&quot;&quot;</span>

<span class="kn">from</span> <span class="nn">snakemake.utils</span> <span class="kn">import</span> <span class="n">min_version</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>
<span class="kn">from</span> <span class="nn">yaml</span> <span class="kn">import</span> <span class="n">dump</span>
<span class="n">min_version</span><span class="p">(</span><span class="s2">&quot;6.0&quot;</span><span class="p">)</span>

<span class="kn">import</span> <span class="nn">sys</span>

<span class="n">worflow_source_dir</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">workflow</span><span class="o">.</span><span class="n">srcdir</span><span class="p">(</span><span class="s2">&quot;.&quot;</span><span class="p">))</span>
<span class="n">common</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">worflow_source_dir</span> <span class="o">/</span> <span class="s2">&quot;..&quot;</span> <span class="o">/</span> <span class="s2">&quot;common&quot;</span> <span class="o">/</span> <span class="s2">&quot;python&quot;</span><span class="p">)</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">common</span><span class="p">)</span>

<span class="kn">from</span> <span class="nn">file_manager</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">files_linker</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">write_yaml</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">messages</span> <span class="kn">import</span> <span class="n">message</span>

<span class="n">logging</span><span class="o">.</span><span class="n">basicConfig</span><span class="p">(</span>
    <span class="n">filename</span><span class="o">=</span><span class="s2">&quot;snakemake.salmon_quant.log&quot;</span><span class="p">,</span>
    <span class="n">filemode</span><span class="o">=</span><span class="s2">&quot;w&quot;</span><span class="p">,</span>
    <span class="n">level</span><span class="o">=</span><span class="n">logging</span><span class="o">.</span><span class="n">DEBUG</span>
<span class="p">)</span>


<span class="k">def</span> <span class="nf">link_seqoia_fq</span><span class="p">(</span>
        <span class="n">sample_names</span><span class="p">:</span> <span class="nb">list</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">r1_paths</span><span class="p">:</span> <span class="nb">list</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">r2_paths</span><span class="p">:</span> <span class="nb">list</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">suffix</span><span class="p">:</span> <span class="nb">str</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">dict</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="nb">str</span><span class="p">]:</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Case r2 are provided:</span>
<span class="sd">    Build a dictionnary containing the following pairs:</span>
<span class="sd">    original_r1_name: reads/{sample}.1.fq.gz</span>
<span class="sd">    original_r2_name: reads/{sample}.2.fq.gz</span>

<span class="sd">    Otherwise:</span>
<span class="sd">    Build a dictionnary containing the following fastq:</span>
<span class="sd">    original_name: reads/{sample}.fq.gz</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">link_dict</span> <span class="o">=</span> <span class="p">{}</span>
    <span class="k">for</span> <span class="n">sample</span><span class="p">,</span> <span class="n">r1</span><span class="p">,</span> <span class="n">r2</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">sample_names</span><span class="p">,</span> <span class="n">r1_paths</span><span class="p">,</span> <span class="n">r2_paths</span><span class="p">):</span>
        <span class="n">link_dict</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;data_</span><span class="si">{</span><span class="n">sample</span><span class="si">}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/fastq/</span><span class="si">{</span><span class="n">sample</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="n">suffix</span><span class="si">}</span><span class="s2">_S1_L1_11_R1_001.fastq.gz&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">r1</span>
        <span class="n">link_dict</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;data_</span><span class="si">{</span><span class="n">sample</span><span class="si">}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/fastq/</span><span class="si">{</span><span class="n">sample</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="n">suffix</span><span class="si">}</span><span class="s2">_S1_L1_11_R2_001.fastq.gz&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">r2</span>
    <span class="k">return</span> <span class="n">link_dict</span>


<span class="n">design</span> <span class="o">=</span> <span class="n">get_design</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">getcwd</span><span class="p">(),</span> <span class="n">search_fastq_pairs</span><span class="p">)</span>
<span class="n">wes_t</span> <span class="o">=</span> <span class="n">link_seqoia_fq</span><span class="p">(</span>
    <span class="n">design</span><span class="o">.</span><span class="n">SAMPLE_ID</span><span class="p">,</span>
    <span class="n">design</span><span class="o">.</span><span class="n">WES_T_U</span><span class="p">,</span>
    <span class="n">design</span><span class="o">.</span><span class="n">WES_T_D</span><span class="p">,</span>
    <span class="n">suffix</span><span class="o">=</span><span class="s2">&quot;WES-T&quot;</span>
<span class="p">)</span>

<span class="n">wes_n</span> <span class="o">=</span> <span class="n">link_seqoia_fq</span><span class="p">(</span>
    <span class="n">design</span><span class="o">.</span><span class="n">SAMPLE_ID</span><span class="p">,</span>
    <span class="n">design</span><span class="o">.</span><span class="n">WES_N_U</span><span class="p">,</span>
    <span class="n">design</span><span class="o">.</span><span class="n">WES_N_D</span><span class="p">,</span>
    <span class="n">suffix</span><span class="o">=</span><span class="s2">&quot;WGS-C&quot;</span>
<span class="p">)</span>

<span class="n">rna_t</span> <span class="o">=</span> <span class="n">link_seqoia_fq</span><span class="p">(</span>
    <span class="n">design</span><span class="o">.</span><span class="n">SAMPLE_ID</span><span class="p">,</span>
    <span class="n">design</span><span class="o">.</span><span class="n">RNA_T_U</span><span class="p">,</span>
    <span class="n">design</span><span class="o">.</span><span class="n">RNA_T_D</span><span class="p">,</span>
    <span class="n">suffix</span><span class="o">=</span><span class="s2">&quot;WTS&quot;</span>
<span class="p">)</span>


<span class="n">sample_list</span> <span class="o">=</span> <span class="n">design</span><span class="o">.</span><span class="n">SAMPLE_ID</span><span class="o">.</span><span class="n">to_list</span><span class="p">()</span>
<span class="n">streams</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;1&quot;</span><span class="p">,</span> <span class="s2">&quot;2&quot;</span><span class="p">]</span>
<span class="n">datatypes</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;WTS&quot;</span><span class="p">,</span> <span class="s2">&quot;WES-T&quot;</span><span class="p">,</span> <span class="s2">&quot;WGS-C&quot;</span><span class="p">]</span>
<span class="n">chromosome_list</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="nb">range</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">23</span><span class="p">)))</span> <span class="o">+</span> <span class="p">[</span><span class="s2">&quot;X&quot;</span><span class="p">,</span> <span class="s2">&quot;Y&quot;</span><span class="p">,</span> <span class="s2">&quot;MT&quot;</span><span class="p">]</span>

<span class="n">wildcard_constraints</span><span class="p">:</span>
    <span class="n">sample</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;|&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">sample_list</span><span class="p">),</span>
    <span class="n">stream</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;|&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">streams</span><span class="p">),</span>
    <span class="n">datatype</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;|&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">datatypes</span><span class="p">)</span>


<span class="n">rule</span> <span class="n">target</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">.flag&quot;</span><span class="p">,</span>
            <span class="c1">#&quot;data_{sample}/{sample}/{sample}_{datatype}_S1_L1_11&quot;,</span>
            <span class="n">sample</span><span class="o">=</span><span class="n">sample_list</span><span class="p">,</span>
            <span class="c1">#datatype=datatypes</span>
        <span class="p">)</span>


<span class="c1">##########################</span>
<span class="c1">### Build config files ###</span>
<span class="c1">##########################</span>


<span class="n">rule</span> <span class="n">prepare_cluster_config</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="s2">&quot;/mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/seqoia/v.2.1.0.GRCh38.cluster.config.json&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">config</span> <span class="o">=</span> <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;v.2.1.0.GRCh38.cluster.config.json&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span> <span class="s2">&quot;Gathering cluster configfile&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">128</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">group</span><span class="p">:</span>
        <span class="s2">&quot;General_config_files&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/prepare_cluster_config.log&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="s2">&quot;--verbose --checksum --human-readable --progress --partial&quot;</span>
    <span class="n">shell</span><span class="p">:</span>
        <span class="s2">&quot;rsync </span><span class="si">{params}</span><span class="s2"> </span><span class="si">{input}</span><span class="s2"> </span><span class="si">{output}</span><span class="s2"> &gt; </span><span class="si">{log}</span><span class="s2"> 2&gt;&amp;1&quot;</span>


<span class="n">rule</span> <span class="n">config_gleaves</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;.config.gleaves&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Building config.gleaves file&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">128</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">group</span><span class="p">:</span>
        <span class="s2">&quot;General_config_files&quot;</span>
    <span class="n">shell</span><span class="p">:</span>
        <span class="s2">&quot;touch </span><span class="si">{output}</span><span class="s2">&quot;</span>


<span class="n">rule</span> <span class="n">prepare_sample_config</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">config</span> <span class="o">=</span> <span class="s2">&quot;v.2.1.0.GRCh38.</span><span class="si">{sample}</span><span class="s2">.pipeline.config.json&quot;</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Building config file for </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">512</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">5</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">group</span><span class="p">:</span> <span class="s2">&quot;sample&quot;</span>
    <span class="c1"># conda:</span>
    <span class="c1">#     &quot;env/seqoia.yaml&quot;</span>
    <span class="n">group</span><span class="p">:</span>
        <span class="s2">&quot;sample&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/prepare_config/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">script</span><span class="p">:</span>
        <span class="s2">&quot;scripts/sample.config.py&quot;</span>


<span class="c1">#################################################</span>
<span class="c1">### Gather files from iRODS or mounting point ###</span>
<span class="c1">#################################################</span>


<span class="n">rule</span> <span class="n">gather_wes_tumor</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/fastq/</span><span class="si">{sample}</span><span class="s2">_WES-T_S1_L1_11_R</span><span class="si">{stream}</span><span class="s2">_001.fastq.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Gathering </span><span class="si">{wildcards.sample}</span><span class="s2"> fastq file (WES-T tumor </span><span class="si">{wildcards.stream}</span><span class="s2">)&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span> <span class="mi">2048</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">group</span><span class="p">:</span> <span class="s2">&quot;sample&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="nb">input</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">output</span><span class="p">:</span> <span class="n">wes_t</span><span class="p">[</span><span class="n">output</span><span class="p">[</span><span class="mi">0</span><span class="p">]],</span>
        <span class="n">datatype</span><span class="o">=</span><span class="s2">&quot;WES-T&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bigr_copy/</span><span class="si">{sample}</span><span class="s2">.WES-T.</span><span class="si">{stream}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/BiGR/copy&quot;</span>


<span class="n">rule</span> <span class="n">gather_wes_normal</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/fastq/</span><span class="si">{sample}</span><span class="s2">_WGS-C_S1_L1_11_R</span><span class="si">{stream}</span><span class="s2">_001.fastq.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Gathering </span><span class="si">{wildcards.sample}</span><span class="s2"> fastq file (WGS-C </span><span class="si">{wildcards.stream}</span><span class="s2">)&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span> <span class="mi">2048</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">group</span><span class="p">:</span> <span class="s2">&quot;sample&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="nb">input</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">output</span><span class="p">:</span> <span class="n">wes_n</span><span class="p">[</span><span class="n">output</span><span class="p">[</span><span class="mi">0</span><span class="p">]],</span>
        <span class="n">datatype</span><span class="o">=</span><span class="s2">&quot;WGS-C&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bigr_copy/</span><span class="si">{sample}</span><span class="s2">.WGS-C.</span><span class="si">{stream}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/BiGR/copy&quot;</span>


<span class="n">rule</span> <span class="n">gather_rna_tumoral</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">temp</span><span class="p">(</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/fastq/</span><span class="si">{sample}</span><span class="s2">_WTS_S1_L1_11_R</span><span class="si">{stream}</span><span class="s2">_001.fastq.gz&quot;</span><span class="p">)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Gathering </span><span class="si">{wildcards.sample}</span><span class="s2"> fastq file (WTS tumor </span><span class="si">{wildcards.stream}</span><span class="s2">)&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="nb">min</span><span class="p">(</span><span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span><span class="p">,</span> <span class="mi">2048</span><span class="p">),</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcard</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">45</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">group</span><span class="p">:</span> <span class="s2">&quot;sample&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="nb">input</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">output</span><span class="p">:</span> <span class="n">rna_t</span><span class="p">[</span><span class="n">output</span><span class="p">[</span><span class="mi">0</span><span class="p">]],</span>
        <span class="n">datatype</span><span class="o">=</span><span class="s2">&quot;WTS&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/bigr_copy/</span><span class="si">{sample}</span><span class="s2">.WTS.</span><span class="si">{stream}</span><span class="s2">.log&quot;</span>
    <span class="n">wrapper</span><span class="p">:</span>
        <span class="s2">&quot;bio/BiGR/copy&quot;</span>


<span class="c1">###########################</span>
<span class="c1">### Run SeqOIA pipeline ###</span>
<span class="c1">###########################</span>


<span class="n">rule</span> <span class="n">create_empty_files</span><span class="p">:</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">ped_genomiser</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/ped_genomiser_generation.done&quot;</span><span class="p">),</span>
        <span class="n">sample_sheet</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/sample_sheet/22NNNN_ANNNNN_NNNN_AXXXXXXXXX.csv&quot;</span><span class="p">),</span>
        <span class="n">json</span><span class="o">=</span><span class="n">temp</span><span class="p">(</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/json/</span><span class="si">{sample}</span><span class="s2">_WGS-C.json&quot;</span><span class="p">),</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Building required empty files for </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">128</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">group</span><span class="p">:</span> <span class="s2">&quot;sample&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/create_empty_files/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">shell</span><span class="p">:</span>
        <span class="s2">&quot;touch </span><span class="si">{output.ped_genomiser}</span><span class="s2"> </span><span class="si">{output.sample_sheet}</span><span class="s2"> </span><span class="si">{output.json}</span><span class="s2"> &gt; </span><span class="si">{log}</span><span class="s2"> 2&gt;&amp;1&quot;</span>


<span class="n">rule</span> <span class="n">seqoia</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">sample_config</span> <span class="o">=</span> <span class="s2">&quot;v.2.1.0.GRCh38.</span><span class="si">{sample}</span><span class="s2">.pipeline.config.json&quot;</span><span class="p">,</span>
        <span class="n">pipeline_config</span> <span class="o">=</span> <span class="s2">&quot;v.2.1.0.GRCh38.cluster.config.json&quot;</span><span class="p">,</span>
        <span class="n">fastq_paths</span> <span class="o">=</span> <span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/fastq/</span><span class="si">{sample}</span><span class="s2">_</span><span class="si">{datatype}</span><span class="s2">_S1_L1_11_R</span><span class="si">{stream}</span><span class="s2">_001.fastq.gz&quot;</span><span class="p">,</span>
            <span class="n">stream</span><span class="o">=</span><span class="n">streams</span><span class="p">,</span>
            <span class="n">datatype</span><span class="o">=</span><span class="n">datatypes</span><span class="p">,</span>
            <span class="n">sample</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">&quot;</span>
        <span class="p">),</span>
        <span class="n">snakefile</span><span class="o">=</span><span class="s2">&quot;/mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/seqoia/scripts/Snakefile_analysis_nowgs_2.1&quot;</span><span class="p">,</span>
        <span class="n">config_gleaves</span><span class="o">=</span><span class="s2">&quot;.config.gleaves&quot;</span><span class="p">,</span>
        <span class="n">ped_genomiser</span><span class="o">=</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/ped_genomiser_generation.done&quot;</span><span class="p">,</span>
        <span class="n">sample_sheet</span><span class="o">=</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/sample_sheet/22NNNN_ANNNNN_NNNN_AXXXXXXXXX.csv&quot;</span><span class="p">,</span>
        <span class="n">json</span><span class="o">=</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/A00000/22NNNN_ANNNNN_NNNN_AXXXXXXXXX/json/</span><span class="si">{sample}</span><span class="s2">_WGS-C.json&quot;</span><span class="p">,</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">touch</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">{sample}</span><span class="s2">.flag&quot;</span><span class="p">)</span>
        <span class="c1">#fusion_inspector=directory(&quot;data_{sample}/{sample}/{sample}_WTS_S1/star_fusion/fusion_inspector/annotator&quot;),</span>
        <span class="c1">#logs=directory(expand(</span>
        <span class="c1">#    &quot;data_{sample}/{sample}/log/{sample}_{datatype}_S1_L1_11&quot;,</span>
        <span class="c1">#    datatype=datatypes,</span>
        <span class="c1">#    allow_missing=True</span>
        <span class="c1">#)),</span>
        <span class="c1">#datatype_dir=directory(expand(</span>
        <span class="c1">#    &quot;data_{sample}/{sample}/{sample}_{datatype}_S1_L1_11&quot;,</span>
        <span class="c1">#    datatype=datatypes,</span>
        <span class="c1">#    allow_missing=True</span>
        <span class="c1">#)),</span>
        <span class="c1">#charge_mut=directory(&quot;data_{sample}/{sample}/Charge_mutationnelle&quot;),</span>
        <span class="c1">#chrom_dirs=directory(expand(</span>
        <span class="c1">#    &quot;data_{sample}/{sample}/chr_{chromosome}&quot;,</span>
        <span class="c1">#    chromosome=chromosome_list,</span>
        <span class="c1">#    allow_missing=True</span>
        <span class="c1">#)),</span>
        <span class="c1">#fusion_dir=directory(&quot;data_{sample}/{sample}/Fusions&quot;),</span>
        <span class="c1">#export_data_status=&quot;data_{sample}/{sample}/export_data_tar.done&quot;,</span>
        <span class="c1">#config_dir=directory(&quot;data_{sample}/{sample}/config&quot;),</span>
        <span class="c1">#mut_sig=directory(&quot;data_{sample}/{sample}/Signature_mutationelle&quot;),</span>
        <span class="c1">#sample_files=multiext(</span>
        <span class="c1">#    &quot;data_{sample}/{sample}/{sample}&quot;,</span>
        <span class="c1">#    &quot;_final_temp.vcf&quot;,</span>
        <span class="c1">#    &quot;_final.vcf&quot;,</span>
        <span class="c1">#    &quot;_importation_gleaves.done&quot;,</span>
        <span class="c1">#    &quot;.json&quot;,</span>
        <span class="c1">#    &quot;_name.vcf&quot;,</span>
        <span class="c1">#    &quot;_qc.csv&quot;,</span>
        <span class="c1">#    &quot;_qc_lab.csv&quot;,</span>
        <span class="c1">#    &quot;_qc_lab_temp.csv&quot;,</span>
        <span class="c1">#    &quot;_qc_validation.done&quot;,</span>
        <span class="c1">#    &quot;.tar.gz&quot;</span>
        <span class="c1">#)</span>
    <span class="n">message</span><span class="p">:</span>
        <span class="s2">&quot;Running SeqOIA single sample no WGS on </span><span class="si">{wildcards.sample}</span><span class="s2">&quot;</span>
    <span class="n">threads</span><span class="p">:</span> <span class="mi">1</span>
    <span class="n">resources</span><span class="p">:</span>
        <span class="n">mem_mb</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">1024</span> <span class="o">*</span> <span class="mi">2</span><span class="p">,</span>
        <span class="n">time_min</span><span class="o">=</span><span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">attempt</span><span class="p">:</span> <span class="n">attempt</span> <span class="o">*</span> <span class="mi">60</span> <span class="o">*</span> <span class="mi">10</span><span class="p">,</span>
        <span class="c1"># mem_mb = 4096,</span>
        <span class="c1"># time_min = 5,</span>
        <span class="n">tmpdir</span><span class="o">=</span><span class="s2">&quot;tmp&quot;</span>
    <span class="n">shadow</span><span class="p">:</span> <span class="s2">&quot;minimal&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">mk</span> <span class="o">=</span> <span class="s2">&quot;--parents --verbose&quot;</span><span class="p">,</span>
        <span class="n">ln</span> <span class="o">=</span> <span class="s2">&quot;--symbolic --force --relative&quot;</span><span class="p">,</span>
        <span class="n">snake_args</span> <span class="o">=</span> <span class="s2">&quot;--jobs 30 --rerun-incomplete --keep-going --max-jobs-per-second 1&quot;</span><span class="p">,</span>
        <span class="n">sbatch</span> <span class="o">=</span> <span class="s2">&quot;&#39;sbatch -p </span><span class="si">{cluster.partition}</span><span class="s2"> -t </span><span class="si">{cluster.time}</span><span class="s2"> --cpus-per-task </span><span class="si">{cluster.cpu}</span><span class="s2"> --mem </span><span class="si">{cluster.mem}</span><span class="s2"> --exclude=n01,n02,n03,n04,n05,n06&#39;&quot;</span><span class="p">,</span>
        <span class="n">fusion_inspector</span><span class="o">=</span><span class="n">directory</span><span class="p">(</span><span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">_WTS_S1/star_fusion/fusion_inspector/annotator&quot;</span><span class="p">),</span>
        <span class="n">logs</span><span class="o">=</span><span class="n">directory</span><span class="p">(</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">/log/</span><span class="si">{sample}</span><span class="s2">_</span><span class="si">{datatype}</span><span class="s2">_S1_L1_11&quot;</span><span class="p">,</span>
            <span class="n">datatype</span><span class="o">=</span><span class="n">datatypes</span><span class="p">,</span>
            <span class="n">allow_missing</span><span class="o">=</span><span class="kc">True</span>
        <span class="p">)),</span>
        <span class="n">datatype_dir</span><span class="o">=</span><span class="n">directory</span><span class="p">(</span><span class="n">expand</span><span class="p">(</span>
            <span class="s2">&quot;data_</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">/</span><span class="si">{sample}</span><span class="s2">_</span><span class="si">{datatype}</span><span class="s2">_S1_L1_11&quot;</span><span class="p">,</span>
            <span class="n">datatype</span><span class="o">=</span><span class="n">datatypes</span><span class="p">,</span>
            <span class="n">allow_missing</span><span class="o">=</span><span class="kc">True</span>
        <span class="p">)),</span>
    <span class="n">conda</span><span class="p">:</span>
        <span class="s2">&quot;env/seqoia.yaml&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="s2">&quot;logs/seqOIA/</span><span class="si">{sample}</span><span class="s2">.log&quot;</span>
    <span class="n">shell</span><span class="p">:</span>
        <span class="s2">&quot;mkdir </span><span class="si">{params.mk}</span><span class="s2"> </span><span class="si">{params.fusion_inspector}</span><span class="s2"> </span><span class="si">{params.logs}</span><span class="s2"> .snakemake &quot;</span>
        <span class="c1">#&quot;mkdir {params.mk} {output.fusion_inspector} {output.logs} .snakemake &quot;</span>
        <span class="s2">&quot;</span><span class="si">{params.datatype_dir}</span><span class="s2"> data_</span><span class="si">{wildcards.sample}</span><span class="s2">/sample_sheet/&quot;</span>
        <span class="c1">#&quot;{output.datatype_dir} data_{wildcards.sample}/sample_sheet/&quot;</span>
        <span class="s2">&quot;data_</span><span class="si">{wildcards.sample}</span><span class="s2">/json/ &gt; </span><span class="si">{log}</span><span class="s2"> 2&gt;&amp;1 &amp;&amp; &quot;</span>
        <span class="s2">&quot;tree -a &gt;&gt; </span><span class="si">{log}</span><span class="s2"> 2&gt;&amp;1 &amp;&amp; &quot;</span>
        <span class="s2">&quot;pwd &gt;&gt; </span><span class="si">{log}</span><span class="s2"> 2&gt;&amp;1 &amp;&amp; &quot;</span>
        <span class="s2">&quot;snakemake --snakefile </span><span class="si">{input.snakefile}</span><span class="s2"> &quot;</span>
        <span class="s2">&quot;--configfile </span><span class="si">{input.sample_config}</span><span class="s2"> &quot;</span>
        <span class="s2">&quot;--cluster-config </span><span class="si">{input.pipeline_config}</span><span class="s2"> &quot;</span>
        <span class="s2">&quot;--cluster </span><span class="si">{params.sbatch}</span><span class="s2"> &quot;</span>
        <span class="s2">&quot;</span><span class="si">{params.snake_args}</span><span class="s2"> --unlock &gt;&gt; </span><span class="si">{log}</span><span class="s2"> 2&gt;&amp;1 &amp;&amp; &quot;</span>
        <span class="s2">&quot;snakemake --snakefile </span><span class="si">{input.snakefile}</span><span class="s2"> &quot;</span>
        <span class="s2">&quot;--configfile </span><span class="si">{input.sample_config}</span><span class="s2"> &quot;</span>
        <span class="s2">&quot;--cluster-config </span><span class="si">{input.pipeline_config}</span><span class="s2"> &quot;</span>
        <span class="s2">&quot;--cluster </span><span class="si">{params.sbatch}</span><span class="s2"> &quot;</span>
        <span class="s2">&quot;</span><span class="si">{params.snake_args}</span><span class="s2"> &gt;&gt; </span><span class="si">{log}</span><span class="s2"> 2&gt;&amp;1 &quot;</span>
</pre></div>
</div>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
<li><p>Marc Deloger</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="salmon_quant.html" class="btn btn-neutral float-left" title="SALMON_QUANT" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="sigprofiler_signatures.html" class="btn btn-neutral float-right" title="SIGPROFILER_SIGNATURES (UNDER DEVELOPMENT)" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>