<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>VARIANT_CALLING_SOMATIC &mdash; Snakemake Wrappers heads/Unofficial documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Long help" href="../readme/readme_long.html" />
    <link rel="prev" title="VARIANT_OCCURENCE_COHORT (UNDER DEVELOPMENT)" href="variant_occurence_cohort.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Snakemake Wrappers
          </a>
              <div class="version">
                heads/Unofficial
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wrappers, Meta-Wrappers</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../wrappers.html">Wrappers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../meta-wrappers.html">Meta-Wrappers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Pipelines</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../pipelines.html">BiGR Pipelines</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="CTC_Variant_Calling_Integragen_Legacy.html">CTC_VARIANT_CALLING_INTEGRAGEN_LEGACY</a></li>
<li class="toctree-l2"><a class="reference internal" href="annot_sv.html">ANNOT_SV</a></li>
<li class="toctree-l2"><a class="reference internal" href="bcl2fastq.html">BCL2FASTQ (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="bs_ampliseq.html">BS_AMPLISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="celltype_deconvolution_rnaseq.html">CELLTYPE_DECONVOLUTION_RNASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="chipseq.html">CHIPSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="cytoscan_eacon.html">EACON_CYTOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="design_builder.html">DESIGN_BUILDER</a></li>
<li class="toctree-l2"><a class="reference internal" href="dge_deseq2.html">DGE_DESEQ2</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastqc_multiqc.html">FASTQC_MULTIQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_set_enrichment.html">GENE_SET_ENRICHMENT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="low_pass_wgs_copy_number.html">LOW_PASS_WGS_COPY_NUMBER</a></li>
<li class="toctree-l2"><a class="reference internal" href="make_pon.html">MAKE_PON</a></li>
<li class="toctree-l2"><a class="reference internal" href="miraseq.html">MIRASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="nfcore_chipseq.html">NFCORE-CHIPSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="oncoscan_eacon.html">EACON_ONCOSCAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="retrotransposons.html">RETROTRANSPOSONS</a></li>
<li class="toctree-l2"><a class="reference internal" href="rnaseq.html">RNASEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon_quant.html">SALMON_QUANT</a></li>
<li class="toctree-l2"><a class="reference internal" href="seqoia.html">SEQOIA SINGLE-SAMPLE</a></li>
<li class="toctree-l2"><a class="reference internal" href="sigprofiler_signatures.html">SIGPROFILER_SIGNATURES (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="snpeff_snpsift.html">SNPEFF_SNPSIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_ampliseq.html">VARIANT_CALLING_AMPLISEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_calling_germline.html">VARIANT_CALLING_GERMLINE</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_occurence_cohort.html">VARIANT_OCCURENCE_COHORT (UNDER DEVELOPMENT)</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">VARIANT_CALLING_SOMATIC</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l3"><a class="reference internal" href="#input-output">Input/Output</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-meta-wrappers">Used meta-wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#used-wrappers">Used wrappers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#notes">Notes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#snakefile">Snakefile</a></li>
<li class="toctree-l3"><a class="reference internal" href="#authors">Authors</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../readme/readme_long.html">Long help</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bigr_unofficial.html">BiGR Unofficial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/api.html">Unofficial API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Snakemake Wrappers</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../pipelines.html">BiGR Pipelines</a> &raquo;</li>
      <li>VARIANT_CALLING_SOMATIC</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/bigr-pipelines/wes_somatic.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="variant-calling-somatic">
<span id="id1"></span><h1>VARIANT_CALLING_SOMATIC<a class="headerlink" href="#variant-calling-somatic" title="Permalink to this heading"></a></h1>
<p>Perform Variant calling on Somatic</p>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<p>In order to run the pipeline, use the following commands</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Go to your working directory</span>

<span class="nb">cd</span> /path/to/my/working/directory

<span class="c1"># Build a design file (see below)</span>

<span class="c1"># Copy/paste the following line for **HG19**</span>

bash /mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/variant_calling_somatic/run.sh hg19

<span class="c1"># Copy/paste the following line for **HG38**</span>

bash /mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/variant_calling_somatic/run.sh hg38

<span class="c1"># Copy/paste the following line for **MM10**</span>

bash /mnt/beegfs/pipelines/snakemake-wrappers/bigr_pipelines/variant_calling_somatic/run.sh mm10
</pre></div>
</div>
</section>
<section id="input-output">
<h2>Input/Output<a class="headerlink" href="#input-output" title="Permalink to this heading"></a></h2>
<p><strong>Input:</strong></p>
<ul class="simple">
<li><p>Pairs of Tumor Fastq-formatted sequence files</p></li>
<li><p>Pairs of corresponding Non-Tumor Fastq-formatted sequence files</p></li>
<li><p>Cosmic database formatted as gzipped vcf and its tbi index (already provided for IGR Flamingo users)</p></li>
<li><p>dbSNP database formatted as gzipped vcf and its tbi index (already provided for IGR Flamingo users)</p></li>
<li><p>MSigDB database formatted as GMT (already provided for IGR Flamingo users)</p></li>
<li><p>GWASCatalog database formatted as TSV (already provided for IGR Flamingo users)</p></li>
<li><p>Kaviar database formatted as gzipped vcf and its tbi index (already provided for IGR Flamingo users)</p></li>
<li><p>SnpEff database downloaded with SnpEff itself (already provided for IGR Flamingo users)</p></li>
<li><p>CaptureKit genomic intervals formatted as BED (already provided for IGR Flamingo users)</p></li>
<li><p>Known variants from dbSNP, with only AF within the INFO field for GATK (already provided for IGR Flamingo users)</p></li>
<li><p>dbNSFP database formatted as TSV (already provided for IGR Flamingo users)</p></li>
<li><p>FastQ Screen databases (already provided for IGR Flamingo users)</p></li>
</ul>
<p><strong>Output:</strong></p>
<ul class="simple">
<li><p>Annotated VCF files</p></li>
<li><p>MultiQC report</p></li>
</ul>
</section>
<section id="used-meta-wrappers">
<h2>Used meta-wrappers<a class="headerlink" href="#used-meta-wrappers" title="Permalink to this heading"></a></h2>
<p>The following individual meta-wrappers are used in this pipeline:</p>
<ul class="simple">
<li><p><span class="xref std std-ref">meta/bio/bwa_fixmate</span></p></li>
<li><p><span class="xref std std-ref">meta/bio/gatk_bqsr</span></p></li>
<li><p><span class="xref std std-ref">meta/bio/varscan2_calling</span></p></li>
<li><p><span class="xref std std-ref">bigr_pipelines/snpeff_snpsift</span></p></li>
</ul>
<p>Please refer to each meta-wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="used-wrappers">
<h2>Used wrappers<a class="headerlink" href="#used-wrappers" title="Permalink to this heading"></a></h2>
<p>The following individual wrappers are used in this pipeline:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../wrappers/BiGR/copy.html#bio-bigr-copy"><span class="std std-ref">IGR_COPY</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/fastp.html#bio-fastp"><span class="std std-ref">FASTP</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/compress/pbgzip.html#bio-compress-pbgzip"><span class="std std-ref">COMPRESS_PBGZIP</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/tabix.html#bio-tabix"><span class="std std-ref">TABIX</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/multiqc.html#bio-multiqc"><span class="std std-ref">MULTIQC</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/picard/collectalignmentsummarymetrics.html#bio-picard-collectalignmentsummarymetrics"><span class="std std-ref">PICARD COLLECTALIGNMENTSUMMARYMETRICS</span></a></p></li>
<li><p><a class="reference internal" href="../wrappers/fastq_screen.html#bio-fastq-screen"><span class="std std-ref">FASTQ_SCREEN</span></a></p></li>
</ul>
<p>Please refer to each wrapper in above list for additional configuration parameters and information about the executed code.</p>
</section>
<section id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this heading"></a></h2>
<p>Prerequisites for piREST:</p>
<ul class="simple">
<li><p>A TSV formatted design file, <em>named ‘design.tsv’</em> with the following columns:</p></li>
</ul>
</section>
<section id="snakefile">
<h2>Snakefile<a class="headerlink" href="#snakefile" title="Permalink to this heading"></a></h2>
<p>The pipeline contains the following steps:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/000.common.smk&quot;</span>


<span class="n">rule</span> <span class="n">variants_only</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">multiqc</span><span class="o">=</span><span class="s2">&quot;data_output/MultiQC/Somatic_Variant_Calling.html&quot;</span><span class="p">,</span>
        <span class="n">vcfs</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span><span class="s2">&quot;data_output/VCF/</span><span class="si">{sample}</span><span class="s2">.vcf.gz&quot;</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample_list</span><span class="p">),</span>
        <span class="n">vcf_tbis</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span><span class="s2">&quot;data_output/VCF/</span><span class="si">{sample}</span><span class="s2">.vcf.gz.tbi&quot;</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample_list</span><span class="p">),</span>
        <span class="n">vcf_xlsx</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span><span class="s2">&quot;data_output/XLSX/</span><span class="si">{sample}</span><span class="s2">.xlsx&quot;</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample_list</span><span class="p">),</span>
        <span class="n">vcf_tsv</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span><span class="s2">&quot;data_output/TSV/</span><span class="si">{sample}</span><span class="s2">.tsv&quot;</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample_list</span><span class="p">),</span>
        <span class="n">tmb</span><span class="o">=</span><span class="s2">&quot;data_output/TMB.tsv&quot;</span><span class="p">,</span>
        <span class="n">msi</span><span class="o">=</span><span class="s2">&quot;data_output/MSI.tsv&quot;</span><span class="p">,</span>
        <span class="n">cnv</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span><span class="s2">&quot;data_output/CNV/</span><span class="si">{sample}</span><span class="s2">.tsv&quot;</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample_list</span><span class="p">),</span>


<span class="n">rule</span> <span class="n">cnv_only</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">cnv</span><span class="o">=</span><span class="n">expand</span><span class="p">(</span><span class="s2">&quot;data_output/CNV/</span><span class="si">{sample}</span><span class="s2">.tsv&quot;</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample_list</span><span class="p">),</span>
        <span class="n">multiqc</span><span class="o">=</span><span class="s2">&quot;data_output/MultiQC/MappingQC.html&quot;</span><span class="p">,</span>


<span class="n">rule</span> <span class="n">tmb_only</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">tmb</span><span class="o">=</span><span class="s2">&quot;data_output/TMB.tsv&quot;</span><span class="p">,</span>
        <span class="n">multiqc</span><span class="o">=</span><span class="s2">&quot;data_output/MultiQC/Somatic_Variant_Calling.html&quot;</span><span class="p">,</span>


<span class="n">rule</span> <span class="n">msi_only</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">msi</span><span class="o">=</span><span class="s2">&quot;data_output/MSI.tsv&quot;</span><span class="p">,</span>
        <span class="n">multiqc</span><span class="o">=</span><span class="s2">&quot;data_output/MultiQC/MappingQC.html&quot;</span><span class="p">,</span>


<span class="n">rule</span> <span class="n">mapping_only</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">multiqc</span><span class="o">=</span><span class="s2">&quot;data_output/MultiQC/MappingQC.html&quot;</span><span class="p">,</span>


<span class="n">rule</span> <span class="n">fusions_only</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">multiqc</span><span class="o">=</span><span class="s2">&quot;data_output/MultiQC/FusionsQC.html&quot;</span><span class="p">,</span>


<span class="c1">#####################</span>
<span class="c1">### Final archive ###</span>
<span class="c1">#####################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/024.archive.smk&quot;</span>


<span class="c1">###############</span>
<span class="c1">### Fusions ###</span>
<span class="c1">###############</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/025.star_chimera.smk&quot;</span>
<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/026.star_fusions.smk&quot;</span>


<span class="c1">###############################</span>
<span class="c1">### TSV and Xlsx formatting ###</span>
<span class="c1">###############################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/018.vcf2tsv.smk&quot;</span>


<span class="c1">#################</span>
<span class="c1">### Gather QC ###</span>
<span class="c1">#################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/016.mapping_qc.smk&quot;</span>
<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/017.multiqc.smk&quot;</span>


<span class="c1">######################</span>
<span class="c1">### MSI sensor pro ###</span>
<span class="c1">######################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/022.msisensor.smk&quot;</span>


<span class="c1">###########</span>
<span class="c1">### TMB ###</span>
<span class="c1">###########</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/021.tmb.smk&quot;</span>


<span class="c1">##################</span>
<span class="c1">### CNV Facets ###</span>
<span class="c1">##################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/020.cnv_facets.smk&quot;</span>
<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/023.annot_sv.smk&quot;</span>


<span class="c1">###########################</span>
<span class="c1">### VCF FILE INDEXATION ###</span>
<span class="c1">###########################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/019.tabix.smk&quot;</span>


<span class="c1">######################</span>
<span class="c1">### VCF annotation ###</span>
<span class="c1">######################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/010.snpeff.smk&quot;</span>
<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/012.snpsift.smk&quot;</span>
<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/015.occurence.smk&quot;</span>
<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/011.spliceai.smk&quot;</span>
<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/013.vcftools.smk&quot;</span>
<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/014.bigr.annot.smk&quot;</span>


<span class="c1">############################################################################</span>
<span class="c1">### Correcting Mutect2 :                                                 ###</span>
<span class="c1">### AS_FilterStatus: Number=1 and not Number=A which violates VCF format ###</span>
<span class="c1">############################################################################</span>

<span class="c1">###############################</span>
<span class="c1">### Variant calling Mutect2 ###</span>
<span class="c1">###############################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/009.gatk.mutect2.smk&quot;</span>


<span class="c1">##############################</span>
<span class="c1">### GATK BAM RECALIBRATION ###</span>
<span class="c1">##############################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/007.gatk.recal.smk&quot;</span>


<span class="c1">#####################</span>
<span class="c1">### Deduplicating ###</span>
<span class="c1">#####################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/006.sambamba.markdup.smk&quot;</span>


<span class="c1"># Filter a bam over the capturekit bed file</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/005.samtools.filter.smk&quot;</span>


<span class="c1">###################</span>
<span class="c1">### BWA MAPPING ###</span>
<span class="c1">###################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/003.bwa.fixmate.smk&quot;</span>


<span class="c1">############################</span>
<span class="c1">### FASTP FASTQ CLEANING ###</span>
<span class="c1">############################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/002.fastp.trimming.smk&quot;</span>


<span class="c1">#################################################</span>
<span class="c1">### Gather files from iRODS or mounting point ###</span>
<span class="c1">#################################################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/001.bigr_copy.smk&quot;</span>


<span class="c1">###########################</span>
<span class="c1">### Datasets indexation ###</span>
<span class="c1">###########################</span>


<span class="n">include</span><span class="p">:</span> <span class="s2">&quot;rules/004.sambamba.index.smk&quot;</span>
</pre></div>
</div>
</section>
<section id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Thibault Dayris</p></li>
<li><p>M boyba Diop</p></li>
<li><p>Marc Deloger</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="variant_occurence_cohort.html" class="btn btn-neutral float-left" title="VARIANT_OCCURENCE_COHORT (UNDER DEVELOPMENT)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../readme/readme_long.html" class="btn btn-neutral float-right" title="Long help" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2016, The Snakemake team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>